

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Module Reference &mdash; MMTK User Guide 2.7.7 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '2.7.7',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="MMTK User Guide 2.7.7 documentation" href="index.html" />
    <link rel="prev" title="Code Examples" href="examples.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="examples.html" title="Code Examples"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">MMTK User Guide 2.7.7 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <span class="target" id="reference"></span><div class="section" id="module-reference">
<h1>Module Reference<a class="headerlink" href="#module-reference" title="Permalink to this headline">¶</a></h1>
<div class="section" id="module-MMTK">
<span id="mmtk"></span><h2>MMTK<a class="headerlink" href="#module-MMTK" title="Permalink to this headline">¶</a></h2>
<p>MMTK is the base module of the Molecular Modelling Toolkit.
It contains the most common objects and all submodules. As a convenience
to the user, it also imports some commonly used objects from other
libraries:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">Vector</span></tt> from <tt class="docutils literal"><span class="pre">Scientific.Geometry</span></tt></li>
<li><tt class="docutils literal"><span class="pre">Translation</span></tt> and <tt class="docutils literal"><span class="pre">Rotation</span></tt> from <tt class="docutils literal"><span class="pre">Scientific.Geometry.Transformation</span></tt></li>
<li><tt class="docutils literal"><span class="pre">copy</span></tt> and <tt class="docutils literal"><span class="pre">deepcopy</span></tt> from <tt class="docutils literal"><span class="pre">copy</span></tt></li>
<li><tt class="docutils literal"><span class="pre">stdin</span></tt>, <tt class="docutils literal"><span class="pre">stdout</span></tt>, and <tt class="docutils literal"><span class="pre">stderr</span></tt> from <tt class="docutils literal"><span class="pre">sys</span></tt></li>
</ul>
</div>
<div class="section" id="module-MMTK.Biopolymers">
<span id="mmtk-biopolymers"></span><h2>MMTK.Biopolymers<a class="headerlink" href="#module-MMTK.Biopolymers" title="Permalink to this headline">¶</a></h2>
<p>Base classes for proteins and nucleic acids</p>
<dl class="class">
<dt id="MMTK.Biopolymers.Residue">
<em class="property">class </em><tt class="descclassname">MMTK.Biopolymers.</tt><tt class="descname">Residue</tt><big>(</big><em>group_spec</em>, <em>_memo=None</em>, <em>**properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/Biopolymers.html#Residue"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Biopolymers.Residue" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.ChemicalObjects.Group" title="MMTK.ChemicalObjects.Group"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ChemicalObjects.Group</span></tt></a></p>
<p>Base class for aminoacid and nucleic acid residues</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>group_spec</strong> (<em>str</em>) &#8211; a string (not case sensitive) that specifies
the group name in the chemical database</li>
<li><strong>position</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the position of the center of mass of the group</li>
<li><strong>name</strong> (<em>str</em>) &#8211; a name given to the group</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.Biopolymers.Residue.nextResidue">
<tt class="descname">nextResidue</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Biopolymers.html#Residue.nextResidue"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Biopolymers.Residue.nextResidue" title="Permalink to this definition">¶</a></dt>
<dd><p>:returns the next residue in the chain</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.Biopolymers.Residue.precedingResidue">
<tt class="descname">precedingResidue</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Biopolymers.html#Residue.precedingResidue"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Biopolymers.Residue.precedingResidue" title="Permalink to this definition">¶</a></dt>
<dd><p>:returns the preceding residue in the chain</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Biopolymers.ResidueChain">
<em class="property">class </em><tt class="descclassname">MMTK.Biopolymers.</tt><tt class="descname">ResidueChain</tt><big>(</big><em>molecule_spec</em>, <em>_memo=None</em>, <em>**properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/Biopolymers.html#ResidueChain"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Biopolymers.ResidueChain" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.ChemicalObjects.Molecule" title="MMTK.ChemicalObjects.Molecule"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ChemicalObjects.Molecule</span></tt></a></p>
<p>Chain of residues</p>
<p>Base class for peptide chains and nucleotide chains</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>molecule_spec</strong> (<em>str</em>) &#8211; a string (not case sensitive) that specifies
the molecule name in the chemical database</li>
<li><strong>position</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the position of the center of mass of the molecule</li>
<li><strong>name</strong> (<em>str</em>) &#8211; a name given to the molecule</li>
<li><strong>configuration</strong> (<em>str</em>) &#8211; the name of a configuration listed in the
database definition of the molecule, which
is used to initialize the atom positions.
If no configuration is specified, the
configuration named &#8220;default&#8221; will be used,
if it exists. Otherwise the atom positions
are undefined.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.Biopolymers.ResidueChain.residues">
<tt class="descname">residues</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Biopolymers.html#ResidueChain.residues"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Biopolymers.ResidueChain.residues" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a collection containing all residues</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Collections.Collection" title="MMTK.Collections.Collection"><tt class="xref py py-class docutils literal"><span class="pre">Collection</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Biopolymers.ResidueChain.residuesOfType">
<tt class="descname">residuesOfType</tt><big>(</big><em>*types</em><big>)</big><a class="reference internal" href="_modules/MMTK/Biopolymers.html#ResidueChain.residuesOfType"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Biopolymers.ResidueChain.residuesOfType" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>types</strong> (<em>str</em>) &#8211; residue type codes</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a collection that contains all residues whose type
(residue code) is contained in types</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Collections.Collection" title="MMTK.Collections.Collection"><tt class="xref py py-class docutils literal"><span class="pre">Collection</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Biopolymers.ResidueChain.sequence">
<tt class="descname">sequence</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Biopolymers.html#ResidueChain.sequence"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Biopolymers.ResidueChain.sequence" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the residue sequence as a list of residue codes</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list of str</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="MMTK.Biopolymers.defineAminoAcidResidue">
<tt class="descclassname">MMTK.Biopolymers.</tt><tt class="descname">defineAminoAcidResidue</tt><big>(</big><em>full_name</em>, <em>code3</em>, <em>code1=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Biopolymers.html#defineAminoAcidResidue"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Biopolymers.defineAminoAcidResidue" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a non-standard amino acid residue to the internal residue table.
Once added to the residue table, the new residue can be used
like any of the standard residues in the creation of peptide chains.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>full_name</strong> (<em>str</em>) &#8211; the name of the group definition in the chemical database</li>
<li><strong>code3</strong> (<em>str</em>) &#8211; the three-letter residue code</li>
<li><strong>code1</strong> (<em>str</em>) &#8211; an optionel one-letter residue code</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.Biopolymers.defineNucleicAcidResidue">
<tt class="descclassname">MMTK.Biopolymers.</tt><tt class="descname">defineNucleicAcidResidue</tt><big>(</big><em>full_name</em>, <em>code</em><big>)</big><a class="reference internal" href="_modules/MMTK/Biopolymers.html#defineNucleicAcidResidue"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Biopolymers.defineNucleicAcidResidue" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a non-standard nucleic acid residue to the internal residue table.
Once added to the residue table, the new residue can be used
like any of the standard residues in the creation of nucleotide chains.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>full_name</strong> (<em>str</em>) &#8211; the name of the group definition in the chemical database</li>
<li><strong>code</strong> &#8211; the residue code</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.Bonds">
<span id="mmtk-bonds"></span><h2>MMTK.Bonds<a class="headerlink" href="#module-MMTK.Bonds" title="Permalink to this headline">¶</a></h2>
<p>Bonds, bond lists, bond angle lists, and dihedral angle lists</p>
<p>The classes in this module are normally not used directly from
client code. They are used by the classes in ChemicalObjects and
ForceField.</p>
<dl class="class">
<dt id="MMTK.Bonds.Bond">
<em class="property">class </em><tt class="descclassname">MMTK.Bonds.</tt><tt class="descname">Bond</tt><big>(</big><em>blueprint</em>, <em>memo=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Bonds.html#Bond"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Bonds.Bond" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Chemical bond</p>
<p>A bond links two atoms (attributes a1 and a2)</p>
<dl class="method">
<dt id="MMTK.Bonds.Bond.hasAtom">
<tt class="descname">hasAtom</tt><big>(</big><em>a</em><big>)</big><a class="reference internal" href="_modules/MMTK/Bonds.html#Bond.hasAtom"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Bonds.Bond.hasAtom" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>a</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; an atom</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">True if a participates in the bond</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Bonds.Bond.otherAtom">
<tt class="descname">otherAtom</tt><big>(</big><em>a</em><big>)</big><a class="reference internal" href="_modules/MMTK/Bonds.html#Bond.otherAtom"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Bonds.Bond.otherAtom" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>a</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; an atom involved in the bond</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the atom at the other end of the bond</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a></td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Raises ValueError:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body">if a does not belong to the bond</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Bonds.BondAngle">
<em class="property">class </em><tt class="descclassname">MMTK.Bonds.</tt><tt class="descname">BondAngle</tt><big>(</big><em>b1</em>, <em>b2</em>, <em>ca</em><big>)</big><a class="reference internal" href="_modules/MMTK/Bonds.html#BondAngle"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Bonds.BondAngle" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Bond angle</p>
<p>A bond angle is the angle between two bonds that share a common atom.
It is defined by two bond objects (attributes b1 and b2) and an atom
object (the common atom, attribute ca).</p>
<dl class="method">
<dt id="MMTK.Bonds.BondAngle.otherBond">
<tt class="descname">otherBond</tt><big>(</big><em>bond</em><big>)</big><a class="reference internal" href="_modules/MMTK/Bonds.html#BondAngle.otherBond"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Bonds.BondAngle.otherBond" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>bond</strong> (<a class="reference internal" href="#MMTK.Bonds.Bond" title="MMTK.Bonds.Bond"><tt class="xref py py-class docutils literal"><span class="pre">Bond</span></tt></a>) &#8211; a bond involved in the angle</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the other bond involved in the angle</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Bonds.Bond" title="MMTK.Bonds.Bond"><tt class="xref py py-class docutils literal"><span class="pre">Bond</span></tt></a></td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Raises ValueError:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body">if bond does not belong to the angle</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Bonds.BondAngleList">
<em class="property">class </em><tt class="descclassname">MMTK.Bonds.</tt><tt class="descname">BondAngleList</tt><big>(</big><em>angles</em><big>)</big><a class="reference internal" href="_modules/MMTK/Bonds.html#BondAngleList"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Bonds.BondAngleList" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Bond angle list</p>
<dl class="method">
<dt id="MMTK.Bonds.BondAngleList.dihedralAngles">
<tt class="descname">dihedralAngles</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Bonds.html#BondAngleList.dihedralAngles"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Bonds.BondAngleList.dihedralAngles" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a list of all dihedral angles that can be formed from the
bond angles in the list</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Bonds.DihedralAngleList" title="MMTK.Bonds.DihedralAngleList"><tt class="xref py py-class docutils literal"><span class="pre">DihedralAngleList</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Bonds.DihedralAngle">
<em class="property">class </em><tt class="descclassname">MMTK.Bonds.</tt><tt class="descname">DihedralAngle</tt><big>(</big><em>ba1</em>, <em>ba2</em>, <em>cb</em><big>)</big><a class="reference internal" href="_modules/MMTK/Bonds.html#DihedralAngle"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Bonds.DihedralAngle" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Dihedral angle</p>
<p>A dihedral angle is the angle between two planes that are defined by
BondAngle objects (attributes ba1 and ba2) and their common bond
(attribute cb).</p>
<p>There are proper dihedrals (four atoms linked by three bonds in
sequence) and improper dihedrals (a central atom linked to three
surrounding atoms by three bonds). The boolean attribute improper
indicates whether a dihedral is an improper one.</p>
</dd></dl>

<dl class="class">
<dt id="MMTK.Bonds.DihedralAngleList">
<em class="property">class </em><tt class="descclassname">MMTK.Bonds.</tt><tt class="descname">DihedralAngleList</tt><big>(</big><em>dihedrals</em><big>)</big><a class="reference internal" href="_modules/MMTK/Bonds.html#DihedralAngleList"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Bonds.DihedralAngleList" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Dihedral angle list</p>
</dd></dl>

</div>
<div class="section" id="module-MMTK.ChargeFit">
<span id="mmtk-chargefit"></span><h2>MMTK.ChargeFit<a class="headerlink" href="#module-MMTK.ChargeFit" title="Permalink to this headline">¶</a></h2>
<p>Fit of point chages to an electrostatic potential surface</p>
<p>This module implements a numerically stable method (based on
Singular Value Decomposition) to fit point charges to values of an
electrostatic potential surface. Two types of constraints are
avaiable: a constraint on the total charge of the system or a subset
of the system, and constraints that force the charges of several atoms
to be equal. There is also a utility function that selects suitable
evaluation points for the electrostatic potential surface. For the
potential evaluation itself, some quantum chemistry program is needed.</p>
<p>The charge fitting method is described in:</p>
<blockquote>
<div><div class="line-block">
<div class="line">K. Hinsen and B. Roux,</div>
<div class="line">An accurate potential for simulating proton transfer in acetylacetone,</div>
<div class="line">J. Comp. Chem. 18, 1997: 368</div>
</div>
</div></blockquote>
<p>See also Examples/Miscellaneous/charge_fit.py.</p>
<dl class="class">
<dt id="MMTK.ChargeFit.ChargeFit">
<em class="property">class </em><tt class="descclassname">MMTK.ChargeFit.</tt><tt class="descname">ChargeFit</tt><big>(</big><em>system</em>, <em>points</em>, <em>constraints=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChargeFit.html#ChargeFit"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChargeFit.ChargeFit" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Fit of point charges to an electrostatic potential surface</p>
<p>A ChargeFit object acts like a dictionary that stores the fitted charge
value for each atom in the system.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>system</strong> &#8211; any chemical object (usually a molecule)</li>
<li><strong>points</strong> &#8211; a list of point/potential pairs (a vector for the
evaluation point, a number for the potential),
or a dictionary whose keys are Configuration objects
and whose values are lists of point/potential pairs.
The latter case permits combined fits for several
conformations of the system.</li>
<li><strong>constraints</strong> &#8211; an optional list of constraint objects
(<a class="reference internal" href="#MMTK.ChargeFit.TotalChargeConstraint" title="MMTK.ChargeFit.TotalChargeConstraint"><tt class="xref py py-class docutils literal"><span class="pre">TotalChargeConstraint</span></tt></a>
and/or
<a class="reference internal" href="#MMTK.ChargeFit.EqualityConstraint" title="MMTK.ChargeFit.EqualityConstraint"><tt class="xref py py-class docutils literal"><span class="pre">EqualityConstraint</span></tt></a> objects). 
If the constraints are inconsistent, a warning is
printed and the result will satisfy the
constraints only in a least-squares sense.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.ChargeFit.EqualityConstraint">
<em class="property">class </em><tt class="descclassname">MMTK.ChargeFit.</tt><tt class="descname">EqualityConstraint</tt><big>(</big><em>atom1</em>, <em>atom2</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChargeFit.html#EqualityConstraint"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChargeFit.EqualityConstraint" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Constraint forcing two charges to be equal</p>
<p>To be used with <a class="reference internal" href="#MMTK.ChargeFit.ChargeFit" title="MMTK.ChargeFit.ChargeFit"><tt class="xref py py-class docutils literal"><span class="pre">ChargeFit</span></tt></a></p>
<p>Any atom may occur in more than one EqualityConstraint object,
in order to keep the charges of more than two atoms equal.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>atom1</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; the first atom in the equality relation</li>
<li><strong>atom2</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; the second atom in the equality relation</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.ChargeFit.TotalChargeConstraint">
<em class="property">class </em><tt class="descclassname">MMTK.ChargeFit.</tt><tt class="descname">TotalChargeConstraint</tt><big>(</big><em>system</em>, <em>charge</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChargeFit.html#TotalChargeConstraint"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChargeFit.TotalChargeConstraint" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Constraint on the total system charge</p>
<p>To be used with <a class="reference internal" href="#MMTK.ChargeFit.ChargeFit" title="MMTK.ChargeFit.ChargeFit"><tt class="xref py py-class docutils literal"><span class="pre">ChargeFit</span></tt></a></p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>system</strong> &#8211; any chamical object whose total charge
is to be constrained</li>
<li><strong>charge</strong> (<em>number</em>) &#8211; the total charge value</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.ChargeFit.evaluationPoints">
<tt class="descclassname">MMTK.ChargeFit.</tt><tt class="descname">evaluationPoints</tt><big>(</big><em>system</em>, <em>n</em>, <em>smallest=0.3</em>, <em>largest=0.5</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChargeFit.html#evaluationPoints"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChargeFit.evaluationPoints" title="Permalink to this definition">¶</a></dt>
<dd><p>Generate points in space around a molecule that are suitable
for potential evaluation in view of a subsequent charge fit.
The points are chosen at random and uniformly in a shell around the system.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>system</strong> &#8211; the chemical object for which the charges
will be fitted</li>
<li><strong>n</strong> &#8211; the number of evaluation points to be generated</li>
<li><strong>smallest</strong> &#8211; the smallest allowed distance of any evaluation
point from any non-hydrogen atom</li>
<li><strong>largest</strong> &#8211; the largest allowed value for the distance
from an evaluation point to the nearest atom</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">a list of evaluation points</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">list of Scientific.Geometry.Vector</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.ChemicalObjects">
<span id="mmtk-chemicalobjects"></span><h2>MMTK.ChemicalObjects<a class="headerlink" href="#module-MMTK.ChemicalObjects" title="Permalink to this headline">¶</a></h2>
<p>Atoms, groups, molecules and similar objects</p>
<dl class="class">
<dt id="MMTK.ChemicalObjects.Atom">
<em class="property">class </em><tt class="descclassname">MMTK.ChemicalObjects.</tt><tt class="descname">Atom</tt><big>(</big><em>atom_spec</em>, <em>_memo=None</em>, <em>**properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#Atom"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.Atom" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.ChemicalObjects.ChemicalObject" title="MMTK.ChemicalObjects.ChemicalObject"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ChemicalObjects.ChemicalObject</span></tt></a></p>
<p>Atom</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>atom_spec</strong> (<em>str</em>) &#8211; a string (not case sensitive) specifying
the chemical element</li>
<li><strong>position</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the position of the atom</li>
<li><strong>name</strong> (<em>str</em>) &#8211; a name given to the atom</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.ChemicalObjects.Atom.bondedTo">
<tt class="descname">bondedTo</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#Atom.bondedTo"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.Atom.bondedTo" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a list of all atoms to which a chemical bond exists.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.Atom.centerOfMass">
<tt class="descname">centerOfMass</tt><big>(</big><em>conf=None</em><big>)</big><a class="headerlink" href="#MMTK.ChemicalObjects.Atom.centerOfMass" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the position in configuration conf. If conf is 
&#8216;None&#8217;, use the current configuration. If the atom has
not been assigned a position, the return value is None.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Vector</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.Atom.position">
<tt class="descname">position</tt><big>(</big><em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#Atom.position"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.Atom.position" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the position in configuration conf. If conf is 
&#8216;None&#8217;, use the current configuration. If the atom has
not been assigned a position, the return value is None.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Vector</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.Atom.setMass">
<tt class="descname">setMass</tt><big>(</big><em>mass</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#Atom.setMass"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.Atom.setMass" title="Permalink to this definition">¶</a></dt>
<dd><p>Changes the mass of the atom.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>mass</strong> (<em>float</em>) &#8211; the mass</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.Atom.setPosition">
<tt class="descname">setPosition</tt><big>(</big><em>position</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#Atom.setPosition"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.Atom.setPosition" title="Permalink to this definition">¶</a></dt>
<dd><p>Changes the position of the atom.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>position</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the new position</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.Atom.translateTo">
<tt class="descname">translateTo</tt><big>(</big><em>position</em><big>)</big><a class="headerlink" href="#MMTK.ChemicalObjects.Atom.translateTo" title="Permalink to this definition">¶</a></dt>
<dd><p>Changes the position of the atom.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>position</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the new position</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.ChemicalObjects.AtomCluster">
<em class="property">class </em><tt class="descclassname">MMTK.ChemicalObjects.</tt><tt class="descname">AtomCluster</tt><big>(</big><em>atoms</em>, <em>**properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#AtomCluster"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.AtomCluster" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.ChemicalObjects.CompositeChemicalObject" title="MMTK.ChemicalObjects.CompositeChemicalObject"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ChemicalObjects.CompositeChemicalObject</span></tt></a>, <a class="reference internal" href="#MMTK.ChemicalObjects.ChemicalObject" title="MMTK.ChemicalObjects.ChemicalObject"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ChemicalObjects.ChemicalObject</span></tt></a></p>
<p>An agglomeration of atoms</p>
<p>An atom cluster acts like a molecule without any bonds or atom
properties. It can be used to represent a group of atoms that
are known to form a chemical unit but whose chemical properties
are not sufficiently known to define a molecule.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>atoms</strong> (<em>list</em>) &#8211; a list of atoms in the cluster</li>
<li><strong>position</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the position of the center of mass of the cluster</li>
<li><strong>name</strong> (<em>str</em>) &#8211; a name given to the cluster</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.ChemicalObjects.ChainMolecule">
<em class="property">class </em><tt class="descclassname">MMTK.ChemicalObjects.</tt><tt class="descname">ChainMolecule</tt><big>(</big><em>molecule_spec</em>, <em>_memo=None</em>, <em>**properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#ChainMolecule"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.ChainMolecule" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.ChemicalObjects.Molecule" title="MMTK.ChemicalObjects.Molecule"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ChemicalObjects.Molecule</span></tt></a></p>
<p>ChainMolecule</p>
<p>ChainMolecules are generated by a MoleculeFactory from
templates that have a sequence attribute.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>molecule_spec</strong> (<em>str</em>) &#8211; a string (not case sensitive) that specifies
the molecule name in the chemical database</li>
<li><strong>position</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the position of the center of mass of the molecule</li>
<li><strong>name</strong> (<em>str</em>) &#8211; a name given to the molecule</li>
<li><strong>configuration</strong> (<em>str</em>) &#8211; the name of a configuration listed in the
database definition of the molecule, which
is used to initialize the atom positions.
If no configuration is specified, the
configuration named &#8220;default&#8221; will be used,
if it exists. Otherwise the atom positions
are undefined.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.ChemicalObjects.ChemicalObject">
<em class="property">class </em><tt class="descclassname">MMTK.ChemicalObjects.</tt><tt class="descname">ChemicalObject</tt><big>(</big><em>blueprint</em>, <em>memo</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#ChemicalObject"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.ChemicalObject" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Collections.GroupOfAtoms</span></tt></a>, <a class="reference internal" href="#MMTK.Visualization.Viewable" title="MMTK.Visualization.Viewable"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Visualization.Viewable</span></tt></a></p>
<p>General chemical object</p>
<p>This is an abstract base class that implements methods which
are applicable to any chemical object (atom, molecule, etc.).</p>
<dl class="method">
<dt id="MMTK.ChemicalObjects.ChemicalObject.atomIterator">
<tt class="descname">atomIterator</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#ChemicalObject.atomIterator"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.ChemicalObject.atomIterator" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">an iterator over all atoms in the object</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">iterator</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.ChemicalObject.atomList">
<tt class="descname">atomList</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#ChemicalObject.atomList"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.ChemicalObject.atomList" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a list containing all atoms in the object</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.ChemicalObject.bondedUnits">
<tt class="descname">bondedUnits</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#ChemicalObject.bondedUnits"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.ChemicalObject.bondedUnits" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the largest subobjects which can contain bonds.
There are no bonds between any of the subobjects
in the list.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.ChemicalObject.degreesOfFreedom">
<tt class="descname">degreesOfFreedom</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#ChemicalObject.degreesOfFreedom"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.ChemicalObject.degreesOfFreedom" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the number of degrees of freedom of the object</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.ChemicalObject.distanceConstraintList">
<tt class="descname">distanceConstraintList</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#ChemicalObject.distanceConstraintList"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.ChemicalObject.distanceConstraintList" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the distance constraints of the object</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.ChemicalObject.fullName">
<tt class="descname">fullName</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#ChemicalObject.fullName"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.ChemicalObject.fullName" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the full name of the object. The full name consists
of the proper name of the object preceded by
the full name of its parent separated by a dot.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.ChemicalObject.getAtomProperty">
<tt class="descname">getAtomProperty</tt><big>(</big><em>atom</em>, <em>property</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#ChemicalObject.getAtomProperty"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.ChemicalObject.getAtomProperty" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieve atom properties from the chemical database.</p>
<p>Note: the property is first looked up in the database entry
for the object on which the method is called. If the lookup
fails, the complete hierarchy from the atom to the top-level
object is constructed and traversed starting from the top-level
object until the property is found. This permits database entries
for higher-level objects to override property definitions in
its constituents.</p>
<p>At the atom level, the property is retrieved from an attribute
with the same name. This means that properties at the atom
level can be defined both in the chemical database and for
each atom individually by assignment to the attribute.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the value of the specified property for the given
atom from the chemical database.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.ChemicalObject.numberOfDistanceConstraints">
<tt class="descname">numberOfDistanceConstraints</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#ChemicalObject.numberOfDistanceConstraints"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.ChemicalObject.numberOfDistanceConstraints" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the number of distance constraints of the object</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.ChemicalObject.removeDistanceConstraints">
<tt class="descname">removeDistanceConstraints</tt><big>(</big><em>universe=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#ChemicalObject.removeDistanceConstraints"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.ChemicalObject.removeDistanceConstraints" title="Permalink to this definition">¶</a></dt>
<dd><p>Removes all distance constraints.</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.ChemicalObject.setBondConstraints">
<tt class="descname">setBondConstraints</tt><big>(</big><em>universe=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#ChemicalObject.setBondConstraints"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.ChemicalObject.setBondConstraints" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets distance constraints for all bonds.</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.ChemicalObject.setRigidBodyConstraints">
<tt class="descname">setRigidBodyConstraints</tt><big>(</big><em>universe=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#ChemicalObject.setRigidBodyConstraints"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.ChemicalObject.setRigidBodyConstraints" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets distance constraints that make the object fully rigid.</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.ChemicalObject.topLevelChemicalObject">
<tt class="descname">topLevelChemicalObject</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#ChemicalObject.topLevelChemicalObject"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.ChemicalObject.topLevelChemicalObject" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the highest-level chemical object of which
the current object is a part.</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.ChemicalObjects.ChemicalObject.universe">
<tt class="descname">universe</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#ChemicalObject.universe"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.ChemicalObject.universe" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the universe to which the object belongs,
or None if the object does not belong to any universe</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.ChemicalObjects.Complex">
<em class="property">class </em><tt class="descclassname">MMTK.ChemicalObjects.</tt><tt class="descname">Complex</tt><big>(</big><em>complex_spec</em>, <em>_memo=None</em>, <em>**properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#Complex"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.Complex" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.ChemicalObjects.CompositeChemicalObject" title="MMTK.ChemicalObjects.CompositeChemicalObject"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ChemicalObjects.CompositeChemicalObject</span></tt></a>, <a class="reference internal" href="#MMTK.ChemicalObjects.ChemicalObject" title="MMTK.ChemicalObjects.ChemicalObject"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ChemicalObjects.ChemicalObject</span></tt></a></p>
<p>Complex</p>
<p>A complex is an assembly of molecules that are not connected by
chemical bonds.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>complex_spec</strong> (<em>str</em>) &#8211; a string (not case sensitive) that specifies
the complex name in the chemical database</li>
<li><strong>position</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the position of the center of mass of the complex</li>
<li><strong>name</strong> (<em>str</em>) &#8211; a name given to the complex</li>
<li><strong>configuration</strong> (<em>str</em>) &#8211; the name of a configuration listed in the
database definition of the complex, which
is used to initialize the atom positions.
If no configuration is specified, the
configuration named &#8220;default&#8221; will be used,
if it exists. Otherwise the atom positions
are undefined.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.ChemicalObjects.CompositeChemicalObject">
<em class="property">class </em><tt class="descclassname">MMTK.ChemicalObjects.</tt><tt class="descname">CompositeChemicalObject</tt><big>(</big><em>properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#CompositeChemicalObject"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.CompositeChemicalObject" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Chemical object containing subobjects</p>
<p>This is an abstract base class that implements methods
which can be used with any composite chemical object,
i.e. any chemical object that is not an atom.</p>
</dd></dl>

<dl class="class">
<dt id="MMTK.ChemicalObjects.Group">
<em class="property">class </em><tt class="descclassname">MMTK.ChemicalObjects.</tt><tt class="descname">Group</tt><big>(</big><em>group_spec</em>, <em>_memo=None</em>, <em>**properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#Group"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.Group" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.ChemicalObjects.CompositeChemicalObject" title="MMTK.ChemicalObjects.CompositeChemicalObject"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ChemicalObjects.CompositeChemicalObject</span></tt></a>, <a class="reference internal" href="#MMTK.ChemicalObjects.ChemicalObject" title="MMTK.ChemicalObjects.ChemicalObject"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ChemicalObjects.ChemicalObject</span></tt></a></p>
<p>Group of bonded atoms</p>
<p>Groups can contain atoms and other groups, and link them by chemical
bonds. They are used to represent functional groups or any other
part of a molecule that has a well-defined identity.</p>
<p>Groups cannot be created in application programs, but only in
database definitions for molecules or through a MoleculeFactory.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>group_spec</strong> (<em>str</em>) &#8211; a string (not case sensitive) that specifies
the group name in the chemical database</li>
<li><strong>position</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the position of the center of mass of the group</li>
<li><strong>name</strong> (<em>str</em>) &#8211; a name given to the group</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.ChemicalObjects.Molecule">
<em class="property">class </em><tt class="descclassname">MMTK.ChemicalObjects.</tt><tt class="descname">Molecule</tt><big>(</big><em>molecule_spec</em>, <em>_memo=None</em>, <em>**properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#Molecule"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.Molecule" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.ChemicalObjects.CompositeChemicalObject" title="MMTK.ChemicalObjects.CompositeChemicalObject"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ChemicalObjects.CompositeChemicalObject</span></tt></a>, <a class="reference internal" href="#MMTK.ChemicalObjects.ChemicalObject" title="MMTK.ChemicalObjects.ChemicalObject"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ChemicalObjects.ChemicalObject</span></tt></a></p>
<p>Molecule</p>
<p>Molecules consist of atoms and groups linked by bonds.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>molecule_spec</strong> (<em>str</em>) &#8211; a string (not case sensitive) that specifies
the molecule name in the chemical database</li>
<li><strong>position</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the position of the center of mass of the molecule</li>
<li><strong>name</strong> (<em>str</em>) &#8211; a name given to the molecule</li>
<li><strong>configuration</strong> (<em>str</em>) &#8211; the name of a configuration listed in the
database definition of the molecule, which
is used to initialize the atom positions.
If no configuration is specified, the
configuration named &#8220;default&#8221; will be used,
if it exists. Otherwise the atom positions
are undefined.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.ChemicalObjects.Molecule.findHydrogenPositions">
<tt class="descname">findHydrogenPositions</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#Molecule.findHydrogenPositions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.Molecule.findHydrogenPositions" title="Permalink to this definition">¶</a></dt>
<dd><p>Find reasonable positions for hydrogen atoms that have no
position assigned.</p>
<p>This method uses a heuristic approach based on standard geometry
data. It was developed for proteins and DNA and may not give
good results for other molecules. It raises an exception
if presented with a topology it cannot handle.</p>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="MMTK.ChemicalObjects.isChemicalObject">
<tt class="descclassname">MMTK.ChemicalObjects.</tt><tt class="descname">isChemicalObject</tt><big>(</big><em>object</em><big>)</big><a class="reference internal" href="_modules/MMTK/ChemicalObjects.html#isChemicalObject"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ChemicalObjects.isChemicalObject" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">True if object is a chemical object</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.Collections">
<span id="mmtk-collections"></span><h2>MMTK.Collections<a class="headerlink" href="#module-MMTK.Collections" title="Permalink to this headline">¶</a></h2>
<p>Collections of chemical objects</p>
<dl class="class">
<dt id="MMTK.Collections.Collection">
<em class="property">class </em><tt class="descclassname">MMTK.Collections.</tt><tt class="descname">Collection</tt><big>(</big><em>*objects</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#Collection"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.Collection" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Collections.GroupOfAtoms</span></tt></a>, <a class="reference internal" href="#MMTK.Visualization.Viewable" title="MMTK.Visualization.Viewable"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Visualization.Viewable</span></tt></a></p>
<p>Collection of chemical objects</p>
<p>Collections permit the grouping of arbitrary chemical objects
(atoms, molecules, etc.) into one object for the purpose of analysis
or manipulation.</p>
<p>Collections permit length inquiry, item extraction by indexing,
and iteration, like any Python sequence object. Two collections
can be added to yield a collection that contains the combined
elements.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>objects</strong> &#8211; a chemical object or a sequence of chemical objects that
define the initial content of the collection.</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.Collections.Collection.addObject">
<tt class="descname">addObject</tt><big>(</big><em>object</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#Collection.addObject"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.Collection.addObject" title="Permalink to this definition">¶</a></dt>
<dd><p>Add objects to the collection.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>object</strong> &#8211; the object(s) to be added. If it is another collection
or a list, all of its elements are added</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.Collection.atomList">
<tt class="descname">atomList</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#Collection.atomList"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.Collection.atomList" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a list containing all atoms of all objects in the collection</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.Collection.distanceConstraintList">
<tt class="descname">distanceConstraintList</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#Collection.distanceConstraintList"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.Collection.distanceConstraintList" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the list of distance constraints</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.Collection.map">
<tt class="descname">map</tt><big>(</big><em>function</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#Collection.map"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.Collection.map" title="Permalink to this definition">¶</a></dt>
<dd><p>Apply a function to all objects in the collection and
return the list of the results. If the results are chemical
objects, a Collection object is returned instead of a list.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>function</strong> (<em>callable</em>) &#8211; the function to be applied</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the list or collection of the results</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.Collection.numberOfAtoms">
<tt class="descname">numberOfAtoms</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#Collection.numberOfAtoms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.Collection.numberOfAtoms" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the total number of atoms in the objects of the collection</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.Collection.numberOfDistanceConstraints">
<tt class="descname">numberOfDistanceConstraints</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#Collection.numberOfDistanceConstraints"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.Collection.numberOfDistanceConstraints" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the number of distance constraints</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.Collection.objectList">
<tt class="descname">objectList</tt><big>(</big><em>type=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#Collection.objectList"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.Collection.objectList" title="Permalink to this definition">¶</a></dt>
<dd><p>Make a list of all objects in the collection that are instances
of a specific type or of one of its subtypes.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>type</strong> &#8211; the type that serves as a filter. If None,
all objects are returned</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the objects that match the given type</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.Collection.removeDistanceConstraints">
<tt class="descname">removeDistanceConstraints</tt><big>(</big><em>universe=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#Collection.removeDistanceConstraints"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.Collection.removeDistanceConstraints" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove all distance constraints</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.Collection.removeObject">
<tt class="descname">removeObject</tt><big>(</big><em>object</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#Collection.removeObject"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.Collection.removeObject" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove an object or a list or collection of objects from the
collection. The object(s) to be removed must be elements of the
collection.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>object</strong> &#8211; the object to be removed, or a list or collection
of objects whose elements are to be removed</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Raises ValueError:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body">if the object is not an element of the collection</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.Collection.selectBox">
<tt class="descname">selectBox</tt><big>(</big><em>p1</em>, <em>p2</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#Collection.selectBox"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.Collection.selectBox" title="Permalink to this definition">¶</a></dt>
<dd><p>Select objects in a rectangular volume</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>p1</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; one corner of the rectangular volume</li>
<li><strong>p2</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the other corner of the rectangular volume</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">a collection of all elements that lie
within the rectangular volume</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last"><a class="reference internal" href="#MMTK.Collections.Collection" title="MMTK.Collections.Collection"><tt class="xref py py-class docutils literal"><span class="pre">Collection</span></tt></a></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.Collection.selectShell">
<tt class="descname">selectShell</tt><big>(</big><em>point</em>, <em>r1</em>, <em>r2=0.0</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#Collection.selectShell"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.Collection.selectShell" title="Permalink to this definition">¶</a></dt>
<dd><p>Select objects in a spherical shell around a central point.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>point</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the center of the spherical shell</li>
<li><strong>r1</strong> (<em>float</em>) &#8211; inner or outer radius of the shell</li>
<li><strong>r2</strong> (<em>float</em>) &#8211; inner or outer radius of the shell (default: 0.)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">a collection of all elements whose
distance from point is between r1 and r2</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last"><a class="reference internal" href="#MMTK.Collections.Collection" title="MMTK.Collections.Collection"><tt class="xref py py-class docutils literal"><span class="pre">Collection</span></tt></a></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.Collection.setBondConstraints">
<tt class="descname">setBondConstraints</tt><big>(</big><em>universe=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#Collection.setBondConstraints"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.Collection.setBondConstraints" title="Permalink to this definition">¶</a></dt>
<dd><p>Set distance constraints for all bonds</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.Collection.universe">
<tt class="descname">universe</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#Collection.universe"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.Collection.universe" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the universe of which all objects in the collection
are part. If no such universe exists, the return value
is None</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Collections.GroupOfAtoms">
<em class="property">class </em><tt class="descclassname">MMTK.Collections.</tt><tt class="descname">GroupOfAtoms</tt><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Anything that consists of atoms</p>
<p>A mix-in class that defines a large set of operations which are
common to all objects that consist of atoms, i.e. any subset of
a chemical system. Examples are atoms, molecules, collections,
or universes.</p>
<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.angularMomentum">
<tt class="descname">angularMomentum</tt><big>(</big><em>velocities=None</em>, <em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.angularMomentum"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.angularMomentum" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>velocities</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>) &#8211; a set of velocities for all atoms, or
None for the current velocities</li>
<li><strong>conf</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a>) &#8211; a configuration object, or None for the
current configuration</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the angluar momentum</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">Scientific.Geometry.Vector</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.angularVelocity">
<tt class="descname">angularVelocity</tt><big>(</big><em>velocities=None</em>, <em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.angularVelocity"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.angularVelocity" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>velocities</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>) &#8211; a set of velocities for all atoms, or
None for the current velocities</li>
<li><strong>conf</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a>) &#8211; a configuration object, or None for the
current configuration</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the angluar velocity</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">Scientific.Geometry.Vector</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.applyTransformation">
<tt class="descname">applyTransformation</tt><big>(</big><em>t</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.applyTransformation"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.applyTransformation" title="Permalink to this definition">¶</a></dt>
<dd><p>Apply a transformation to the object</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>t</strong> (<em>Scientific.Geometry.Transformation</em>) &#8211; the transformation to be applied</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.atomCollection">
<tt class="descname">atomCollection</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.atomCollection"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.atomCollection" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a collection containing all atoms in the object</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.atomsWithDefinedPositions">
<tt class="descname">atomsWithDefinedPositions</tt><big>(</big><em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.atomsWithDefinedPositions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.atomsWithDefinedPositions" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>conf</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a> or NoneType) &#8211; a configuration object, or None for the
current configuration</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a collection of all atoms that have a position in the
given configuration</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.booleanMask">
<tt class="descname">booleanMask</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.booleanMask"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.booleanMask" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a ParticleScalar object that contains a value of 1
for each atom that is in the object and a value of 0 for all
other atoms in the universe</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.boundingBox">
<tt class="descname">boundingBox</tt><big>(</big><em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.boundingBox"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.boundingBox" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>conf</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a> or NoneType) &#8211; a configuration object, or None for the
current configuration</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">two opposite corners of a bounding box around the
object. The bounding box is the smallest rectangular
bounding box with edges parallel to the coordinate axes.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">tuple of two Scientific.Geometry.Vector</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.boundingSphere">
<tt class="descname">boundingSphere</tt><big>(</big><em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.boundingSphere"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.boundingSphere" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>conf</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a> or NoneType) &#8211; a configuration object, or None for the
current configuration</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a sphere that contains all atoms in the object.
This is B{not} the minimal bounding sphere, just B{some}
bounding sphere.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Objects3D.Sphere</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.centerAndMomentOfInertia">
<tt class="descname">centerAndMomentOfInertia</tt><big>(</big><em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.centerAndMomentOfInertia"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.centerAndMomentOfInertia" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>conf</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a> or NoneType) &#8211; a configuration object, or None for the
current configuration</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the center of mass and the moment of inertia tensor
in the given configuration</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.centerOfMass">
<tt class="descname">centerOfMass</tt><big>(</big><em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.centerOfMass"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.centerOfMass" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>conf</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a> or NoneType) &#8211; a configuration object, or None for the
current configuration</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the center of mass in the given configuration</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Vector</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.charge">
<tt class="descname">charge</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.charge"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.charge" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the total charge of the object. This is defined only
for objects that are part of a universe with a force
field that defines charges.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.degreesOfFreedom">
<tt class="descname">degreesOfFreedom</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.degreesOfFreedom"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.degreesOfFreedom" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the number of mechanical degrees of freedom</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.dipole">
<tt class="descname">dipole</tt><big>(</big><em>reference=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.dipole"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.dipole" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the total dipole moment of the object. This is defined only
for objects that are part of a universe with a force field
that defines charges.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Vector</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.displacementUnderTransformation">
<tt class="descname">displacementUnderTransformation</tt><big>(</big><em>t</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.displacementUnderTransformation"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.displacementUnderTransformation" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>t</strong> (<em>Scientific.Geometry.Transformation</em>) &#8211; the transformation to be applied</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the displacement vectors for the atoms in the object
that correspond to the transformation t.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.findTransformation">
<tt class="descname">findTransformation</tt><big>(</big><em>conf1</em>, <em>conf2=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.findTransformation"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.findTransformation" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>conf1</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a>) &#8211; a configuration object</li>
<li><strong>conf2</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a> or NoneType) &#8211; a configuration object, or None for the
current configuration</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">the linear transformation that, when applied to
the object in configuration conf1, minimizes the
RMS distance to the conformation in conf2, and the
minimal RMS distance.
If conf2 is None, returns the transformation from the
current configuration to conf1 and the associated
RMS distance.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.kineticEnergy">
<tt class="descname">kineticEnergy</tt><big>(</big><em>velocities=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.kineticEnergy"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.kineticEnergy" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>velocities</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>) &#8211; a set of velocities for all atoms, or
None for the current velocities</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the kinetic energy</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.mass">
<tt class="descname">mass</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.mass"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.mass" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the total mass</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.momentum">
<tt class="descname">momentum</tt><big>(</big><em>velocities=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.momentum"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.momentum" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>velocities</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>) &#8211; a set of velocities for all atoms, or
None for the current velocities</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the momentum</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Vector</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.normalizeConfiguration">
<tt class="descname">normalizeConfiguration</tt><big>(</big><em>repr=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.normalizeConfiguration"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.normalizeConfiguration" title="Permalink to this definition">¶</a></dt>
<dd><p>Apply a linear transformation such that the center of mass of
the object is translated to the coordinate origin and its
principal axes of inertia become parallel to the three
coordinate axes.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>repr</strong> &#8211; the specific representation for axis alignment:
- Ir    : x y z &lt;&#8211;&gt; b c a
- IIr   : x y z &lt;&#8211;&gt; c a b
- IIIr  : x y z &lt;&#8211;&gt; a b c
- Il    : x y z &lt;&#8211;&gt; c b a
- IIl   : x y z &lt;&#8211;&gt; a c b
- IIIl  : x y z &lt;&#8211;&gt; b a c</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.normalizePosition">
<tt class="descname">normalizePosition</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.normalizePosition"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.normalizePosition" title="Permalink to this definition">¶</a></dt>
<dd><p>Translate the center of mass to the coordinate origin</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.normalizingTransformation">
<tt class="descname">normalizingTransformation</tt><big>(</big><em>repr=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.normalizingTransformation"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.normalizingTransformation" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate a linear transformation that shifts the center of mass
of the object to the coordinate origin and makes its
principal axes of inertia parallel to the three coordinate
axes.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>repr</strong> &#8211; the specific representation for axis alignment:
Ir    : x y z &lt;&#8211;&gt; b c a
IIr   : x y z &lt;&#8211;&gt; c a b
IIIr  : x y z &lt;&#8211;&gt; a b c
Il    : x y z &lt;&#8211;&gt; c b a
IIl   : x y z &lt;&#8211;&gt; a c b
IIIl  : x y z &lt;&#8211;&gt; b a c</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the normalizing transformation</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Transformation.RigidBodyTransformation</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.numberOfAtoms">
<tt class="descname">numberOfAtoms</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.numberOfAtoms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.numberOfAtoms" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the number of atoms</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.numberOfCartesianCoordinates">
<tt class="descname">numberOfCartesianCoordinates</tt><big>(</big><big>)</big><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.numberOfCartesianCoordinates" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the number of geometrical points that define the
object. It is currently equal to the number of atoms,
but could be different e.g. for quantum systems, in which
each atom is described by a wave function or a path integral.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.numberOfFixedAtoms">
<tt class="descname">numberOfFixedAtoms</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.numberOfFixedAtoms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.numberOfFixedAtoms" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the number of atoms that are fixed, i.e. that cannot move</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.numberOfPoints">
<tt class="descname">numberOfPoints</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.numberOfPoints"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.numberOfPoints" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the number of geometrical points that define the
object. It is currently equal to the number of atoms,
but could be different e.g. for quantum systems, in which
each atom is described by a wave function or a path integral.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.position">
<tt class="descname">position</tt><big>(</big><em>conf=None</em><big>)</big><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.position" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>conf</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a> or NoneType) &#8211; a configuration object, or None for the
current configuration</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the center of mass in the given configuration</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Vector</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.rmsDifference">
<tt class="descname">rmsDifference</tt><big>(</big><em>conf1</em>, <em>conf2=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.rmsDifference"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.rmsDifference" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>conf1</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a>) &#8211; a configuration object</li>
<li><strong>conf2</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a> or NoneType) &#8211; a configuration object, or None for the
current configuration</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the RMS (root-mean-square) difference between the
conformations of the object in two universe configurations,
conf1 and conf2</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">float</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.rotateAroundAxis">
<tt class="descname">rotateAroundAxis</tt><big>(</big><em>point1</em>, <em>point2</em>, <em>angle</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.rotateAroundAxis"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.rotateAroundAxis" title="Permalink to this definition">¶</a></dt>
<dd><p>Rotate the object arond an axis specified by two points</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>point1</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the first point</li>
<li><strong>point2</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the second point</li>
<li><strong>angle</strong> (<em>float</em>) &#8211; the rotation angle (in radians)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.rotateAroundCenter">
<tt class="descname">rotateAroundCenter</tt><big>(</big><em>axis_direction</em>, <em>angle</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.rotateAroundCenter"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.rotateAroundCenter" title="Permalink to this definition">¶</a></dt>
<dd><p>Rotate the object around an axis that passes through its center
of mass.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>axis_direction</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the direction of the axis of rotation</li>
<li><strong>angle</strong> (<em>float</em>) &#8211; the rotation angle (in radians)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.rotateAroundOrigin">
<tt class="descname">rotateAroundOrigin</tt><big>(</big><em>axis_direction</em>, <em>angle</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.rotateAroundOrigin"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.rotateAroundOrigin" title="Permalink to this definition">¶</a></dt>
<dd><p>Rotate the object around an axis that passes through the
coordinate origin.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>axis_direction</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the direction of the axis of rotation</li>
<li><strong>angle</strong> (<em>float</em>) &#8211; the rotation angle (in radians)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.rotationalConstants">
<tt class="descname">rotationalConstants</tt><big>(</big><em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.rotationalConstants"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.rotationalConstants" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>conf</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a> or NoneType) &#8211; a configuration object, or None for the
current configuration</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a sorted array of rotational constants A, B, C
in internal units</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.temperature">
<tt class="descname">temperature</tt><big>(</big><em>velocities=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.temperature"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.temperature" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>velocities</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>) &#8211; a set of velocities for all atoms, or
None for the current velocities</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the temperature</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.translateBy">
<tt class="descname">translateBy</tt><big>(</big><em>vector</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.translateBy"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.translateBy" title="Permalink to this definition">¶</a></dt>
<dd><p>Translate the object by a displacement vector</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>vector</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the displacement vector</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.translateTo">
<tt class="descname">translateTo</tt><big>(</big><em>position</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.translateTo"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.translateTo" title="Permalink to this definition">¶</a></dt>
<dd><p>Translate the object such that its center of mass is at position
:param position: the final position
:type position: Scientific.Geometry.Vector</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.universe">
<tt class="descname">universe</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.universe"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.universe" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the universe of which the object is part. For an
object that is not part of a universe, the result is
None</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.view">
<tt class="descname">view</tt><big>(</big><em>configuration=None</em>, <em>format='pdb'</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.view"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.view" title="Permalink to this definition">¶</a></dt>
<dd><p>Start an external viewer for the object in the given
configuration.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>configuration</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a>) &#8211; the configuration to be visualized</li>
<li><strong>format</strong> &#8211; &#8216;pdb&#8217; (for running $PDBVIEWER) or &#8216;vrml&#8217;
(for running $VRMLVIEWER). An optional
subformat specification can be added, see
<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms.writeToFile" title="MMTK.Collections.GroupOfAtoms.writeToFile"><tt class="xref py py-class docutils literal"><span class="pre">writeToFile</span></tt></a> for the details.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.GroupOfAtoms.writeToFile">
<tt class="descname">writeToFile</tt><big>(</big><em>filename</em>, <em>configuration=None</em>, <em>format=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#GroupOfAtoms.writeToFile"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.GroupOfAtoms.writeToFile" title="Permalink to this definition">¶</a></dt>
<dd><p>Write a representation of the object in a given
configuration to a file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>filename</strong> (<em>str</em>) &#8211; the name of the file</li>
<li><strong>configuration</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a> or NoneType) &#8211; a configuration object, or None for the
current configuration</li>
<li><strong>format</strong> (<em>str</em>) &#8211; &#8216;pdb&#8217; or &#8216;vrml&#8217; (default: guess from filename)
A subformat specification can be added, separated
by a dot. Subformats of &#8216;vrml&#8217; are &#8216;wireframe&#8217;
(default), &#8216;ball_and_stick&#8217;, &#8216;highlight&#8217; (like
&#8216;wireframe&#8217;, but with a small sphere for
all atoms that have an attribute &#8216;highlight&#8217; with a
non-zero value), and &#8216;charge&#8217; (wireframe plus small
spheres for the atoms whose color indicates the
charge on a red-to-green color scale)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Collections.PartitionedAtomCollection">
<em class="property">class </em><tt class="descclassname">MMTK.Collections.</tt><tt class="descname">PartitionedAtomCollection</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#PartitionedAtomCollection"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.PartitionedAtomCollection" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Collections.PartitionedCollection" title="MMTK.Collections.PartitionedCollection"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Collections.PartitionedCollection</span></tt></a></p>
<p>Partitioned collection of atoms</p>
<p>PartitionedAtomCollection objects behave like PartitionedCollection
atoms, except that they store only atoms. When a composite chemical
object is added, its atoms are stored instead.</p>
</dd></dl>

<dl class="class">
<dt id="MMTK.Collections.PartitionedCollection">
<em class="property">class </em><tt class="descclassname">MMTK.Collections.</tt><tt class="descname">PartitionedCollection</tt><big>(</big><em>partition_size</em>, <em>*objects</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#PartitionedCollection"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.PartitionedCollection" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Collections.Collection" title="MMTK.Collections.Collection"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Collections.Collection</span></tt></a></p>
<p>Collection with cubic partitions</p>
<p>A PartitionedCollection differs from a plain Collection by
sorting its elements into small cubic cells. This makes adding
objects slower, but geometrical operations like 
selectShell become much faster for a large number of
objects.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>partition_size</strong> &#8211; the edge length of the cubic cells</li>
<li><strong>objects</strong> &#8211; a chemical object or a sequence of chemical objects that
define the initial content of the collection.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.Collections.PartitionedCollection.pairsWithinCutoff">
<tt class="descname">pairsWithinCutoff</tt><big>(</big><em>cutoff</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#PartitionedCollection.pairsWithinCutoff"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.PartitionedCollection.pairsWithinCutoff" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>cutoff</strong> &#8211; a cutoff for pair distances</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a list containing all pairs of objects in the
collection whose center-of-mass distance is less than
the cutoff</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Collections.PartitionedCollection.partitions">
<tt class="descname">partitions</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#PartitionedCollection.partitions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.PartitionedCollection.partitions" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a list of cubic partitions. Each partition is specified
by a tuple containing two vectors (describing the diagonally
opposite corners) and the list of objects in the partition.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="MMTK.Collections.isCollection">
<tt class="descclassname">MMTK.Collections.</tt><tt class="descname">isCollection</tt><big>(</big><em>object</em><big>)</big><a class="reference internal" href="_modules/MMTK/Collections.html#isCollection"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Collections.isCollection" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>object</strong> &#8211; any Python object</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">True if the object is a <a class="reference internal" href="#MMTK.Collections.Collection" title="MMTK.Collections.Collection"><tt class="xref py py-class docutils literal"><span class="pre">Collection</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.ConfigIO">
<span id="mmtk-configio"></span><h2>MMTK.ConfigIO<a class="headerlink" href="#module-MMTK.ConfigIO" title="Permalink to this headline">¶</a></h2>
<p>I/O of molecular configurations</p>
<p>Input: Z-Matrix and Cartesian
Output: VRML
PDB files are handled in <a class="reference internal" href="#module-MMTK.PDB" title="MMTK.PDB"><tt class="xref py py-class docutils literal"><span class="pre">PDB</span></tt></a>.</p>
<dl class="class">
<dt id="MMTK.ConfigIO.Cartesian">
<em class="property">class </em><tt class="descclassname">MMTK.ConfigIO.</tt><tt class="descname">Cartesian</tt><big>(</big><em>data</em><big>)</big><a class="reference internal" href="_modules/MMTK/ConfigIO.html#Cartesian"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ConfigIO.Cartesian" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Cartesian specification of a molecule conformation</p>
<p>Cartesian objects can be used in chemical database entries
to specify molecule conformations by Cartesian coordinates.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>data</strong> &#8211; a dictionary mapping atoms to tuples of length three
that define its Cartesian coordinates</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.ConfigIO.Cartesian.applyTo">
<tt class="descname">applyTo</tt><big>(</big><em>object</em><big>)</big><a class="reference internal" href="_modules/MMTK/ConfigIO.html#Cartesian.applyTo"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ConfigIO.Cartesian.applyTo" title="Permalink to this definition">¶</a></dt>
<dd><p>Define the positions of the atoms in a chemical object by the
stored coordinates.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>object</strong> &#8211; the object to which the coordinates are applied</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.ConfigIO.ZMatrix">
<em class="property">class </em><tt class="descclassname">MMTK.ConfigIO.</tt><tt class="descname">ZMatrix</tt><big>(</big><em>data</em><big>)</big><a class="reference internal" href="_modules/MMTK/ConfigIO.html#ZMatrix"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ConfigIO.ZMatrix" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Z-Matrix specification of a molecule conformation</p>
<p>ZMatrix objects can be used in chemical database entries
to specify molecule conformations by internal coordinates.
With the exception of the three first atoms, each atom is
defined relative to three previously atoms by a distance,
an angle, and a dihedral angle.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>data</strong> &#8211; <p>a list of atom definitions. Each atom definition (except
for the first three ones) is a list containing seven
elements:</p>
<blockquote>
<div><ul class="simple">
<li>the atom to be defined</li>
<li>a previously defined atom and the distance to it</li>
<li>another previously defined atom and the angle to it</li>
<li>a third previously defined atom and the dihedral
angle to it</li>
</ul>
</div></blockquote>
<p>The definition of the first atom contains only the first
element, the second atom needs the first three elements,
and the third atom is defined by the first five elements.</p>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.ConfigIO.ZMatrix.applyTo">
<tt class="descname">applyTo</tt><big>(</big><em>object</em><big>)</big><a class="reference internal" href="_modules/MMTK/ConfigIO.html#ZMatrix.applyTo"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ConfigIO.ZMatrix.applyTo" title="Permalink to this definition">¶</a></dt>
<dd><p>Define the positions of the atoms in a chemical object by the
internal coordinates of the Z-Matrix.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>object</strong> &#8211; the object to which the Z-Matrix is applied</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-MMTK.DCD">
<span id="mmtk-dcd"></span><h2>MMTK.DCD<a class="headerlink" href="#module-MMTK.DCD" title="Permalink to this headline">¶</a></h2>
<p>Reading and writing of DCD trajectory files</p>
<p>The DCD format for trajectories is used by CHARMM, X-Plor,
and NAMD. It can be read by various visualization programs.</p>
<p>The DCD format is defined as a binary (unformatted) Fortran
format and is therefore platform-dependent.</p>
<dl class="class">
<dt id="MMTK.DCD.DCDReader">
<em class="property">class </em><tt class="descclassname">MMTK.DCD.</tt><tt class="descname">DCDReader</tt><big>(</big><em>universe</em>, <em>**options</em><big>)</big><a class="reference internal" href="_modules/MMTK/DCD.html#DCDReader"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.DCD.DCDReader" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Trajectory.TrajectoryGenerator" title="MMTK.Trajectory.TrajectoryGenerator"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Trajectory.TrajectoryGenerator</span></tt></a></p>
<p>Reader for DCD trajectories (CHARMM/X-Plor/NAMD)</p>
<p>A DCDReader reads a DCD trajectory and &#8220;plays back&#8221; the
data as if it were generated directly by an integrator.
The universe for which the DCD file is read must be
perfectly compatible with the data in the file, including
an identical internal atom numbering. This can be guaranteed
only if the universe was created from a PDB file that is
compatible with the DCD file without leaving out any
part of the system.</p>
<p>Reading is started by calling the reader object.
The following data categories and variables are available for
output:</p>
<ul class="simple">
<li>category &#8220;time&#8221;: time</li>
<li>category &#8220;configuration&#8221;: configuration</li>
</ul>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> &#8211; the universe for which the information from the
trajectory file is read</li>
<li><strong>options</strong> &#8211; keyword options</li>
<li><strong>dcd_file</strong> &#8211; the name of the DCD trajecory file to be read</li>
<li><strong>actions</strong> &#8211; a list of actions to be executed periodically
(default is none)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.DCD.writeDCDPDB">
<tt class="descclassname">MMTK.DCD.</tt><tt class="descname">writeDCDPDB</tt><big>(</big><em>conf_list</em>, <em>dcd_file_name</em>, <em>pdb_file_name</em>, <em>delta_t=0.1</em><big>)</big><a class="reference internal" href="_modules/MMTK/DCD.html#writeDCDPDB"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.DCD.writeDCDPDB" title="Permalink to this definition">¶</a></dt>
<dd><p>Write a sequence of configurations to a DCD file and generate
a compatible PDB file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>conf_list</strong> (sequence of <a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a>) &#8211; the sequence of configurations</li>
<li><strong>dcd_file_name</strong> (<em>str</em>) &#8211; the name of the DCD file</li>
<li><strong>pdb_file_name</strong> (<em>str</em>) &#8211; the name of the PDB file</li>
<li><strong>delta_t</strong> (<em>float</em>) &#8211; the time step between two configurations</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.DCD.writeVelocityDCDPDB">
<tt class="descclassname">MMTK.DCD.</tt><tt class="descname">writeVelocityDCDPDB</tt><big>(</big><em>vel_list</em>, <em>dcd_file_name</em>, <em>pdb_file_name</em>, <em>delta_t=0.1</em><big>)</big><a class="reference internal" href="_modules/MMTK/DCD.html#writeVelocityDCDPDB"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.DCD.writeVelocityDCDPDB" title="Permalink to this definition">¶</a></dt>
<dd><p>Write a sequence of velocity particle vectors to a DCD file and generate
a compatible PDB file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>vel_list</strong> (sequence of <a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>) &#8211; the sequence of velocity particle vectors</li>
<li><strong>dcd_file_name</strong> (<em>str</em>) &#8211; the name of the DCD file</li>
<li><strong>pdb_file_name</strong> (<em>str</em>) &#8211; the name of the PDB file</li>
<li><strong>delta_t</strong> (<em>float</em>) &#8211; the time step between two velocity sets</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.Deformation">
<span id="mmtk-deformation"></span><h2>MMTK.Deformation<a class="headerlink" href="#module-MMTK.Deformation" title="Permalink to this headline">¶</a></h2>
<p>Deformation energies in proteins</p>
<p>This module implements deformational energies for use in the analysis
of motions and conformational changes in macromolecules. A description
of the techniques can be found in the following articles:</p>
<blockquote>
<div><div class="line-block">
<div class="line">K. Hinsen</div>
<div class="line">Analysis of domain motions by approximate normal mode calculations</div>
<div class="line">Proteins 33 (1998): 417-429</div>
</div>
<div class="line-block">
<div class="line">K. Hinsen, A. Thomas, M.J. Field</div>
<div class="line">Analysis of domain motions in large proteins</div>
<div class="line">Proteins 34 (1999): 369-382</div>
</div>
</div></blockquote>
<dl class="class">
<dt id="MMTK.Deformation.DeformationEnergyFunction">
<em class="property">class </em><tt class="descclassname">MMTK.Deformation.</tt><tt class="descname">DeformationEnergyFunction</tt><big>(</big><em>universe</em>, <em>fc_length=0.7</em>, <em>cutoff=1.2</em>, <em>factor=46402.0</em>, <em>form='exponential'</em><big>)</big><a class="reference internal" href="_modules/MMTK/Deformation.html#DeformationEnergyFunction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Deformation.DeformationEnergyFunction" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.Deformation.DeformationEvaluationFunction</span></tt></p>
<p>Infinite-displacement deformation energy function</p>
<p>The deformation energy is the sum of the deformation values over
all atoms of a system.</p>
<p>The default values are appropriate for a C<sub>α</sub> model of a protein
with the global scaling described in the reference cited above.</p>
<p>A DeformationEnergyFunction is called with one or two parameters.
The first parameter is a ParticleVector object containing the
infinitesimal displacements of the atoms for which the deformation
energy is to be evaluated. The optional second argument can be
set to a non-zero value to request the gradients of the energy
in addition to the energy itself. In that case there are two
return values (energy and the gradients in a ParticleVector
object), otherwise only the energy is returned.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe for which the deformation function
is defined</li>
<li><strong>fc_length</strong> (<em>float</em>) &#8211; the range parameter r_0 in the pair interaction term</li>
<li><strong>cutoff</strong> (<em>float</em>) &#8211; the cutoff used in the deformation calculation</li>
<li><strong>factor</strong> (<em>float</em>) &#8211; a global scaling factor</li>
<li><strong>form</strong> (<em>str</em>) &#8211; the functional form (&#8216;exponential&#8217; or &#8216;calpha&#8217;)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Deformation.DeformationFunction">
<em class="property">class </em><tt class="descclassname">MMTK.Deformation.</tt><tt class="descname">DeformationFunction</tt><big>(</big><em>universe</em>, <em>fc_length=0.7</em>, <em>cutoff=1.2</em>, <em>factor=46402.0</em>, <em>form='exponential'</em><big>)</big><a class="reference internal" href="_modules/MMTK/Deformation.html#DeformationFunction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Deformation.DeformationFunction" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.Deformation.DeformationEvaluationFunction</span></tt></p>
<p>Infinite-displacement deformation function</p>
<p>The default values are appropriate for a C<sub>α</sub> model of a protein
with the global scaling described in the reference cited above.</p>
<p>A DeformationFunction object must be called with a single parameter,
which is a ParticleVector object containing the infinitesimal displacements
of the atoms for which the deformation is to be evaluated.
The return value is a ParticleScalar object containing the
deformation value for each atom.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe for which the deformation function
is defined</li>
<li><strong>fc_length</strong> (<em>float</em>) &#8211; the range parameter r_0 in the pair interaction term</li>
<li><strong>cutoff</strong> (<em>float</em>) &#8211; the cutoff used in the deformation calculation</li>
<li><strong>factor</strong> (<em>float</em>) &#8211; a global scaling factor</li>
<li><strong>form</strong> (<em>str</em>) &#8211; the functional form (&#8216;exponential&#8217; or &#8216;calpha&#8217;)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Deformation.DeformationReducer">
<em class="property">class </em><tt class="descclassname">MMTK.Deformation.</tt><tt class="descname">DeformationReducer</tt><big>(</big><em>universe</em>, <em>fc_length=0.7</em>, <em>cutoff=1.2</em>, <em>factor=46402.0</em>, <em>form='exponential'</em><big>)</big><a class="reference internal" href="_modules/MMTK/Deformation.html#DeformationReducer"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Deformation.DeformationReducer" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.Deformation.DeformationEvaluationFunction</span></tt></p>
<p>Iterative reduction of the deformation energy</p>
<p>The default values are appropriate for a C<sub>α</sub> model of a protein
with the global scaling described in the reference cited above.</p>
<p>A DeformationReducer is called with two arguments. The first
is a ParticleVector containing the initial infinitesimal displacements
for all atoms. The second is an integer indicating the number of
iterations. The result is a modification of the displacements
by steepest-descent minimization of the deformation energy.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe for which the deformation function
is defined</li>
<li><strong>fc_length</strong> (<em>float</em>) &#8211; the range parameter r_0 in the pair interaction term</li>
<li><strong>cutoff</strong> (<em>float</em>) &#8211; the cutoff used in the deformation calculation</li>
<li><strong>factor</strong> (<em>float</em>) &#8211; a global scaling factor</li>
<li><strong>form</strong> (<em>str</em>) &#8211; the functional form (&#8216;exponential&#8217; or &#8216;calpha&#8217;)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Deformation.FiniteDeformationEnergyFunction">
<em class="property">class </em><tt class="descclassname">MMTK.Deformation.</tt><tt class="descname">FiniteDeformationEnergyFunction</tt><big>(</big><em>universe</em>, <em>fc_length=0.7</em>, <em>cutoff=1.2</em>, <em>factor=46402.0</em>, <em>form='exponential'</em><big>)</big><a class="reference internal" href="_modules/MMTK/Deformation.html#FiniteDeformationEnergyFunction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Deformation.FiniteDeformationEnergyFunction" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.Deformation.DeformationEvaluationFunction</span></tt></p>
<p>Finite-displacement deformation energy function</p>
<p>The deformation energy is the sum of the
deformation values over all atoms of a system.</p>
<p>The default values are appropriate for a C<sub>α</sub> model of a protein
with the global scaling described in the reference cited above.</p>
<p>A FiniteDeformationEnergyFunction is called with one or two parameters.
The first parameter is a ParticleVector object containing the
alternate configuration of the universe for which the deformation
energy is to be evaluated. The optional second argument can be
set to a non-zero value to request the gradients of the energy
in addition to the energy itself. In that case there are two
return values (energy and the gradients in a ParticleVector
object), otherwise only the energy is returned.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe for which the deformation function
is defined</li>
<li><strong>fc_length</strong> (<em>float</em>) &#8211; the range parameter r_0 in the pair interaction term</li>
<li><strong>cutoff</strong> (<em>float</em>) &#8211; the cutoff used in the deformation calculation</li>
<li><strong>factor</strong> (<em>float</em>) &#8211; a global scaling factor</li>
<li><strong>form</strong> (<em>str</em>) &#8211; the functional form (&#8216;exponential&#8217; or &#8216;calpha&#8217;)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Deformation.FiniteDeformationFunction">
<em class="property">class </em><tt class="descclassname">MMTK.Deformation.</tt><tt class="descname">FiniteDeformationFunction</tt><big>(</big><em>universe</em>, <em>fc_length=0.7</em>, <em>cutoff=1.2</em>, <em>factor=46402.0</em>, <em>form='exponential'</em><big>)</big><a class="reference internal" href="_modules/MMTK/Deformation.html#FiniteDeformationFunction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Deformation.FiniteDeformationFunction" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.Deformation.DeformationEvaluationFunction</span></tt></p>
<p>Finite-displacement deformation function</p>
<p>The default values are appropriate for a C<sub>α</sub> model of a protein
with the global scaling described in the reference cited above.</p>
<p>A FiniteDeformationFunction object must be called with a single parameter,
which is a Configuration or a ParticleVector object containing the
alternate configuration of the universe for which the deformation is to be
evaluated. The return value is a ParticleScalar object containing the
deformation value for each atom.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe for which the deformation function
is defined</li>
<li><strong>fc_length</strong> (<em>float</em>) &#8211; the range parameter r_0 in the pair interaction term</li>
<li><strong>cutoff</strong> (<em>float</em>) &#8211; the cutoff used in the deformation calculation</li>
<li><strong>factor</strong> (<em>float</em>) &#8211; a global scaling factor</li>
<li><strong>form</strong> (<em>str</em>) &#8211; the functional form (&#8216;exponential&#8217; or &#8216;calpha&#8217;)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Deformation.FiniteDeformationReducer">
<em class="property">class </em><tt class="descclassname">MMTK.Deformation.</tt><tt class="descname">FiniteDeformationReducer</tt><big>(</big><em>universe</em>, <em>fc_length=0.7</em>, <em>cutoff=1.2</em>, <em>factor=46402.0</em>, <em>form='exponential'</em><big>)</big><a class="reference internal" href="_modules/MMTK/Deformation.html#FiniteDeformationReducer"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Deformation.FiniteDeformationReducer" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.Deformation.DeformationEvaluationFunction</span></tt></p>
<p>Iterative reduction of the finite-displacement deformation energy</p>
<p>The default values are appropriate for a C<sub>α</sub> model of a protein
with the global scaling described in the reference cited above.</p>
<p>A FiniteDeformationReducer is called with two arguments. The first
is a ParticleVector or Configuration containing the alternate
configuration for which the deformation energy is evaluated.
The second is the RMS distance that defines the termination
condition. The return value a configuration that differs from
the input configuration by approximately the specified RMS distance,
and which is obtained by iterative steepest-descent minimization of
the finite-displacement deformation energy.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe for which the deformation function
is defined</li>
<li><strong>fc_length</strong> (<em>float</em>) &#8211; the range parameter r_0 in the pair interaction term</li>
<li><strong>cutoff</strong> (<em>float</em>) &#8211; the cutoff used in the deformation calculation</li>
<li><strong>factor</strong> (<em>float</em>) &#8211; a global scaling factor</li>
<li><strong>form</strong> (<em>str</em>) &#8211; the functional form (&#8216;exponential&#8217; or &#8216;calpha&#8217;)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Deformation.NormalizedDeformationEnergyFunction">
<em class="property">class </em><tt class="descclassname">MMTK.Deformation.</tt><tt class="descname">NormalizedDeformationEnergyFunction</tt><big>(</big><em>universe</em>, <em>fc_length=0.7</em>, <em>cutoff=1.2</em>, <em>factor=46402.0</em>, <em>form='exponential'</em><big>)</big><a class="reference internal" href="_modules/MMTK/Deformation.html#NormalizedDeformationEnergyFunction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Deformation.NormalizedDeformationEnergyFunction" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Deformation.DeformationEnergyFunction" title="MMTK.Deformation.DeformationEnergyFunction"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Deformation.DeformationEnergyFunction</span></tt></a></p>
<p>Normalized infinite-displacement deformation energy function</p>
<p>The normalized deformation energy is the sum of the normalized
deformation values over all atoms of a system.</p>
<p>The default values are appropriate for a C<sub>α</sub> model of a protein
with the global scaling described in the reference cited above.
The normalization is defined by equation 10 of reference 1.</p>
<p>A NormalizedDeformationEnergyFunction is called with one or two parameters.
The first parameter is a ParticleVector object containing the
infinitesimal displacements of the atoms for which the deformation
energy is to be evaluated. The optional second argument can be
set to a non-zero value to request the gradients of the energy
in addition to the energy itself. In that case there are two
return values (energy and the gradients in a ParticleVector
object), otherwise only the energy is returned.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe for which the deformation function
is defined</li>
<li><strong>fc_length</strong> (<em>float</em>) &#8211; the range parameter r_0 in the pair interaction term</li>
<li><strong>cutoff</strong> (<em>float</em>) &#8211; the cutoff used in the deformation calculation</li>
<li><strong>factor</strong> (<em>float</em>) &#8211; a global scaling factor</li>
<li><strong>form</strong> (<em>str</em>) &#8211; the functional form (&#8216;exponential&#8217; or &#8216;calpha&#8217;)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Deformation.NormalizedDeformationFunction">
<em class="property">class </em><tt class="descclassname">MMTK.Deformation.</tt><tt class="descname">NormalizedDeformationFunction</tt><big>(</big><em>*args</em>, <em>**kwargs</em><big>)</big><a class="reference internal" href="_modules/MMTK/Deformation.html#NormalizedDeformationFunction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Deformation.NormalizedDeformationFunction" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Deformation.DeformationFunction" title="MMTK.Deformation.DeformationFunction"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Deformation.DeformationFunction</span></tt></a></p>
<p>Normalized infinite-displacement deformation function</p>
<p>The default values are appropriate for a C<sub>α</sub> model of a protein
with the global scaling described in the reference cited above.
The normalization is defined by equation 10 of reference 1 (see above).</p>
<p>A NormalizedDeformationFunction object must be called with a single
parameter, which is a ParticleVector object containing the infinitesimal
displacements of the atoms for which the deformation is to be evaluated.
The return value is a ParticleScalar object containing the
deformation value for each atom.</p>
</dd></dl>

</div>
<div class="section" id="module-MMTK.Dynamics">
<span id="mmtk-dynamics"></span><h2>MMTK.Dynamics<a class="headerlink" href="#module-MMTK.Dynamics" title="Permalink to this headline">¶</a></h2>
<p>Molecular Dynamics integrators</p>
<dl class="class">
<dt id="MMTK.Dynamics.BarostatReset">
<em class="property">class </em><tt class="descclassname">MMTK.Dynamics.</tt><tt class="descname">BarostatReset</tt><big>(</big><em>first=0</em>, <em>last=None</em>, <em>skip=1</em><big>)</big><a class="reference internal" href="_modules/MMTK/Dynamics.html#BarostatReset"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Dynamics.BarostatReset" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Trajectory.TrajectoryAction" title="MMTK.Trajectory.TrajectoryAction"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Trajectory.TrajectoryAction</span></tt></a></p>
<p>Barostat reset action</p>
<p>A BarostatReset object is used in the action list of a
VelocityVerletIntegrator. It resets the barostat coordinate
to zero.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>first</strong> (<em>int</em>) &#8211; the number of the first step at which the action
is executed</li>
<li><strong>last</strong> (<em>int</em>) &#8211; the number of the first step at which the action
is no longer executed. A value of None indicates
that the action should be executed indefinitely.</li>
<li><strong>skip</strong> (<em>int</em>) &#8211; the number of steps to skip between two applications
of the action</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Dynamics.Heater">
<em class="property">class </em><tt class="descclassname">MMTK.Dynamics.</tt><tt class="descname">Heater</tt><big>(</big><em>temp1</em>, <em>temp2</em>, <em>gradient</em>, <em>first=0</em>, <em>last=None</em>, <em>skip=1</em><big>)</big><a class="reference internal" href="_modules/MMTK/Dynamics.html#Heater"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Dynamics.Heater" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Trajectory.TrajectoryAction" title="MMTK.Trajectory.TrajectoryAction"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Trajectory.TrajectoryAction</span></tt></a></p>
<p>Periodic heating action</p>
<p>A Heater object us used in the action list of a VelocityVerletIntegrator.
It scales the velocities to a temperature that increases over time.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>temp1</strong> (<em>float</em>) &#8211; the temperature value to which the velocities
should be scaled initially</li>
<li><strong>temp2</strong> (<em>float</em>) &#8211; the final temperature value to which the velocities
should be scaled</li>
<li><strong>gradient</strong> (<em>float</em>) &#8211; the temperature gradient (in K/ps)</li>
<li><strong>first</strong> (<em>int</em>) &#8211; the number of the first step at which the action
is executed</li>
<li><strong>last</strong> (<em>int</em>) &#8211; the number of the first step at which the action
is no longer executed. A value of None indicates
that the action should be executed indefinitely.</li>
<li><strong>skip</strong> (<em>int</em>) &#8211; the number of steps to skip between two applications
of the action</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Dynamics.RotationRemover">
<em class="property">class </em><tt class="descclassname">MMTK.Dynamics.</tt><tt class="descname">RotationRemover</tt><big>(</big><em>first=0</em>, <em>last=None</em>, <em>skip=1</em><big>)</big><a class="reference internal" href="_modules/MMTK/Dynamics.html#RotationRemover"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Dynamics.RotationRemover" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Trajectory.TrajectoryAction" title="MMTK.Trajectory.TrajectoryAction"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Trajectory.TrajectoryAction</span></tt></a></p>
<p>Action that eliminates global rotation</p>
<p>A RotationRemover object is used in the action list of a
VelocityVerletIntegrator. It adjusts the atomic velocities
such that the total angular momentum is zero.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>first</strong> (<em>int</em>) &#8211; the number of the first step at which the action
is executed</li>
<li><strong>last</strong> (<em>int</em>) &#8211; the number of the first step at which the action
is no longer executed. A value of None indicates
that the action should be executed indefinitely.</li>
<li><strong>skip</strong> (<em>int</em>) &#8211; the number of steps to skip between two applications
of the action</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Dynamics.TranslationRemover">
<em class="property">class </em><tt class="descclassname">MMTK.Dynamics.</tt><tt class="descname">TranslationRemover</tt><big>(</big><em>first=0</em>, <em>last=None</em>, <em>skip=1</em><big>)</big><a class="reference internal" href="_modules/MMTK/Dynamics.html#TranslationRemover"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Dynamics.TranslationRemover" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Trajectory.TrajectoryAction" title="MMTK.Trajectory.TrajectoryAction"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Trajectory.TrajectoryAction</span></tt></a></p>
<p>Action that eliminates global translation</p>
<p>A TranslationRemover object is used in the action list of a
VelocityVerletIntegrator. It subtracts the total velocity
of the system from each atomic velocity.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>first</strong> (<em>int</em>) &#8211; the number of the first step at which the action
is executed</li>
<li><strong>last</strong> (<em>int</em>) &#8211; the number of the first step at which the action
is no longer executed. A value of None indicates
that the action should be executed indefinitely.</li>
<li><strong>skip</strong> (<em>int</em>) &#8211; the number of steps to skip between two applications
of the action</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Dynamics.VelocityScaler">
<em class="property">class </em><tt class="descclassname">MMTK.Dynamics.</tt><tt class="descname">VelocityScaler</tt><big>(</big><em>temperature</em>, <em>window=0.0</em>, <em>first=0</em>, <em>last=None</em>, <em>skip=1</em><big>)</big><a class="reference internal" href="_modules/MMTK/Dynamics.html#VelocityScaler"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Dynamics.VelocityScaler" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Trajectory.TrajectoryAction" title="MMTK.Trajectory.TrajectoryAction"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Trajectory.TrajectoryAction</span></tt></a></p>
<p>Periodic velocity scaling action</p>
<p>A VelocityScaler object is used in the action list of
a VelocityVerletIntegrator. It rescales all atomic velocities
by a common factor to make the temperature of the system equal
to a predefined value.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>temperature</strong> (<em>float</em>) &#8211; the temperature value to which the velocities
should be scaled</li>
<li><strong>window</strong> (<em>float</em>) &#8211; the deviation from the ideal temperature that
is tolerated in either direction before rescaling
takes place</li>
<li><strong>first</strong> (<em>int</em>) &#8211; the number of the first step at which the action
is executed</li>
<li><strong>last</strong> (<em>int</em>) &#8211; the number of the first step at which the action
is no longer executed. A value of None indicates
that the action should be executed indefinitely.</li>
<li><strong>skip</strong> (<em>int</em>) &#8211; the number of steps to skip between two applications
of the action</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Dynamics.VelocityVerletIntegrator">
<em class="property">class </em><tt class="descclassname">MMTK.Dynamics.</tt><tt class="descname">VelocityVerletIntegrator</tt><big>(</big><em>universe</em>, <em>**options</em><big>)</big><a class="reference internal" href="_modules/MMTK/Dynamics.html#VelocityVerletIntegrator"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Dynamics.VelocityVerletIntegrator" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.Dynamics.Integrator</span></tt></p>
<p>Velocity-Verlet molecular dynamics integrator</p>
<p>The integrator can handle fixed atoms, distance constraints,
a thermostat, and a barostat, as well as any combination.
It is fully thread-safe.</p>
<p>The integration is started by calling the integrator object.
All the keyword options (see documnentation of __init__) can be
specified either when creating the integrator or when calling it.</p>
<p>The following data categories and variables are available for
output:</p>
<blockquote>
<div><ul class="simple">
<li>category &#8220;time&#8221;: time</li>
<li>category &#8220;configuration&#8221;: configuration and box size (for
periodic universes)</li>
<li>category &#8220;velocities&#8221;: atomic velocities</li>
<li>category &#8220;gradients&#8221;: energy gradients for each atom</li>
<li>category &#8220;energy&#8221;: potential and kinetic energy, plus
extended-system energy terms if a thermostat and/or barostat
are used</li>
<li>category &#8220;thermodynamic&#8221;: temperature, volume (if a barostat
is used) and pressure</li>
<li>category &#8220;auxiliary&#8221;: extended-system coordinates if a thermostat
and/or barostat are used</li>
</ul>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe on which the integrator acts</li>
<li><strong>steps</strong> (<em>int</em>) &#8211; the number of integration steps (default is 100)</li>
<li><strong>delta_t</strong> (<em>float</em>) &#8211; the time step (default is 1 fs)</li>
<li><strong>actions</strong> (<em>list</em>) &#8211; a list of actions to be executed periodically
(default is none)</li>
<li><strong>threads</strong> (<em>int</em>) &#8211; the number of threads to use in energy evaluation
(default set by MMTK_ENERGY_THREADS)</li>
<li><strong>background</strong> (<em>bool</em>) &#8211; if True, the integration is executed as a
separate thread (default: False)</li>
<li><strong>mpi_communicator</strong> (<em>Scientific.MPI.MPICommunicator</em>) &#8211; an MPI communicator object, or None,
meaning no parallelization (default: None)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.Environment">
<span id="mmtk-environment"></span><h2>MMTK.Environment<a class="headerlink" href="#module-MMTK.Environment" title="Permalink to this headline">¶</a></h2>
<p>Environment objects</p>
<p>Environment objects are objects that define a simulation system without
being composed of atoms. Examples are thermostats, barostats, external
fields, etc.</p>
<dl class="class">
<dt id="MMTK.Environment.AndersenBarostat">
<em class="property">class </em><tt class="descclassname">MMTK.Environment.</tt><tt class="descname">AndersenBarostat</tt><big>(</big><em>pressure</em>, <em>relaxation_time=1.5</em><big>)</big><a class="reference internal" href="_modules/MMTK/Environment.html#AndersenBarostat"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Environment.AndersenBarostat" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.Environment.EnvironmentObject</span></tt></p>
<p>Andersen barostat for Molecular Dynamics</p>
<p>A barostat object can be added to a universe and will then
together with a thermostat object modify the integration algorithm
to a simulation of an NPT ensemble.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>pressure</strong> (<em>float</em>) &#8211; the pressure set by the barostat</li>
<li><strong>relaxation_time</strong> (<em>float</em>) &#8211; the relaxation time of the
barostat coordinate</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Environment.NoseThermostat">
<em class="property">class </em><tt class="descclassname">MMTK.Environment.</tt><tt class="descname">NoseThermostat</tt><big>(</big><em>temperature</em>, <em>relaxation_time=0.2</em><big>)</big><a class="reference internal" href="_modules/MMTK/Environment.html#NoseThermostat"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Environment.NoseThermostat" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.Environment.EnvironmentObject</span></tt></p>
<p>Nose thermostat for Molecular Dynamics</p>
<p>A thermostat object can be added to a universe and will then
modify the integration algorithm to a simulation of an NVT
ensemble.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>temperature</strong> (<em>float</em>) &#8211; the temperature set by the thermostat</li>
<li><strong>relaxation_time</strong> (<em>float</em>) &#8211; the relaxation time of the
thermostat coordinate</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.Field">
<span id="mmtk-field"></span><h2>MMTK.Field<a class="headerlink" href="#module-MMTK.Field" title="Permalink to this headline">¶</a></h2>
<p>Scalar and vector fields in molecular systems</p>
<p>This module defines field objects that are useful in the analysis
and visualization of collective motions in molecular systems. Atomic
quantities characterizing collective motions vary slowly in space, and
can be considered functions of position instead of values per atom.
Functions of position are called fields, and mathematical techniques
for the analysis of fields have proven useful in many branches of
physics. Fields can be described numerically by values on a
regular grid. In addition to permitting the application of vector
analysis methods to atomic quantities, the introduction of
fields is a valuable visualization aid, because information defined on
a coarse regular grid can be added to a picture of a molecular system
without overloading it.</p>
<dl class="class">
<dt id="MMTK.Field.AtomicField">
<em class="property">class </em><tt class="descclassname">MMTK.Field.</tt><tt class="descname">AtomicField</tt><big>(</big><em>system</em>, <em>grid_size</em>, <em>values</em><big>)</big><a class="reference internal" href="_modules/MMTK/Field.html#AtomicField"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Field.AtomicField" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>A field whose values are determined by atomic quantities</p>
<p>This is an abstract base class. To create field objects,
use one of its subclasses.</p>
<dl class="method">
<dt id="MMTK.Field.AtomicField.particleValues">
<tt class="descname">particleValues</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Field.html#AtomicField.particleValues"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Field.AtomicField.particleValues" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the values of the field at the positions of the atoms</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ParticleProperties.ParticleProperty" title="MMTK.ParticleProperties.ParticleProperty"><tt class="xref py py-class docutils literal"><span class="pre">ParticleProperty</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Field.AtomicField.view">
<tt class="descname">view</tt><big>(</big><em>scale=1.0</em>, <em>length_scale=1.0</em>, <em>color=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Field.html#AtomicField.view"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Field.AtomicField.view" title="Permalink to this definition">¶</a></dt>
<dd><p>Shows a graphical representation of the field using a VRML viewer.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>scale</strong> (<em>float</em>) &#8211; scale factor applied to all field values</li>
<li><strong>color</strong> (<em>Scientific.Visualization.Color</em>) &#8211; the color for all graphics objects</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Field.AtomicField.writeToFile">
<tt class="descname">writeToFile</tt><big>(</big><em>filename</em>, <em>scale=1.0</em>, <em>length_scale=1.0</em>, <em>color=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Field.html#AtomicField.writeToFile"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Field.AtomicField.writeToFile" title="Permalink to this definition">¶</a></dt>
<dd><p>Writes a graphical representation of the field to a VRML file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>filename</strong> (<em>str</em>) &#8211; the name of the destination file</li>
<li><strong>scale</strong> (<em>float</em>) &#8211; scale factor applied to all field values</li>
<li><strong>color</strong> (<em>Scientific.Visualization.Color</em>) &#8211; the color for all graphics objects</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Field.AtomicScalarField">
<em class="property">class </em><tt class="descclassname">MMTK.Field.</tt><tt class="descname">AtomicScalarField</tt><big>(</big><em>system</em>, <em>grid_size</em>, <em>values</em><big>)</big><a class="reference internal" href="_modules/MMTK/Field.html#AtomicScalarField"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Field.AtomicScalarField" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Field.AtomicField" title="MMTK.Field.AtomicField"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Field.AtomicField</span></tt></a>, <a class="reference internal" href="#MMTK.Visualization.Viewable" title="MMTK.Visualization.Viewable"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Visualization.Viewable</span></tt></a></p>
<p>Scalar field defined by atomic quantities</p>
<p>For visualization, scalar fields are represented by a small cube
on each grid point whose color indicates the field&#8217;s value on a symmetric
red-to-green color scale defined by the range of the field values.</p>
<dl class="docutils">
<dt>Additional keyword options exist for graphics object generation:</dt>
<dd><ul class="first last simple">
<li>scale=factor, to multiply all field values by a factor</li>
<li>range=(min, max), to eliminate graphics objects for values
that are smaller than min or larger than max</li>
</ul>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>system</strong> &#8211; any subset of a molecular system</li>
<li><strong>grid_size</strong> (<em>float</em>) &#8211; the spacing of a cubic grid on which the field
values are defined. The value for a point is obtained
by averaging the atomic quantities over all
atoms in a cube centered on the point.</li>
<li><strong>values</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a>) &#8211; the atomic values that define the field</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.Field.AtomicScalarField.gradient">
<tt class="descname">gradient</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Field.html#AtomicScalarField.gradient"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Field.AtomicScalarField.gradient" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the gradient of the field</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Field.AtomicVectorField" title="MMTK.Field.AtomicVectorField"><tt class="xref py py-class docutils literal"><span class="pre">AtomicVectorField</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Field.AtomicScalarField.laplacian">
<tt class="descname">laplacian</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Field.html#AtomicScalarField.laplacian"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Field.AtomicScalarField.laplacian" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the laplacian of the field</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Field.AtomicScalarField" title="MMTK.Field.AtomicScalarField"><tt class="xref py py-class docutils literal"><span class="pre">AtomicScalarField</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Field.AtomicVectorField">
<em class="property">class </em><tt class="descclassname">MMTK.Field.</tt><tt class="descname">AtomicVectorField</tt><big>(</big><em>system</em>, <em>grid_size</em>, <em>values</em><big>)</big><a class="reference internal" href="_modules/MMTK/Field.html#AtomicVectorField"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Field.AtomicVectorField" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Field.AtomicField" title="MMTK.Field.AtomicField"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Field.AtomicField</span></tt></a>, <a class="reference internal" href="#MMTK.Visualization.Viewable" title="MMTK.Visualization.Viewable"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Visualization.Viewable</span></tt></a></p>
<p>Vector field defined by atomic quantities</p>
<p>For visualization, scalar fields are represented by a small arrow
on each grid point. The arrow starts at the grid point and represents
the vector value at that point.</p>
<dl class="docutils">
<dt>Additional keyword options exist for graphics object generation:</dt>
<dd><ul class="first last simple">
<li>scale=factor, to multiply all field values by a factor</li>
<li>diameter=number, to define the diameter of the arrow objects
(default: 1.)</li>
<li>range=(min, max), to eliminate graphics objects for values
that are smaller than min or larger than max</li>
<li>color=string, to define the color of the arrows by a color name</li>
</ul>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>system</strong> &#8211; any subset of a molecular system</li>
<li><strong>grid_size</strong> (<em>float</em>) &#8211; the spacing of a cubic grid on which the field
values are defined. The value for a point is obtained
by averaging the atomic quantities over all
atoms in a cube centered on the point.</li>
<li><strong>values</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>) &#8211; the atomic values that define the field</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.Field.AtomicVectorField.curl">
<tt class="descname">curl</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Field.html#AtomicVectorField.curl"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Field.AtomicVectorField.curl" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the curl of the field</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Field.AtomicVectorField" title="MMTK.Field.AtomicVectorField"><tt class="xref py py-class docutils literal"><span class="pre">AtomicVectorField</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Field.AtomicVectorField.divergence">
<tt class="descname">divergence</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Field.html#AtomicVectorField.divergence"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Field.AtomicVectorField.divergence" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the divergence of the field</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Field.AtomicScalarField" title="MMTK.Field.AtomicScalarField"><tt class="xref py py-class docutils literal"><span class="pre">AtomicScalarField</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Field.AtomicVectorField.laplacian">
<tt class="descname">laplacian</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Field.html#AtomicVectorField.laplacian"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Field.AtomicVectorField.laplacian" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the laplacian of the field</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Field.AtomicVectorField" title="MMTK.Field.AtomicVectorField"><tt class="xref py py-class docutils literal"><span class="pre">AtomicVectorField</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Field.AtomicVectorField.length">
<tt class="descname">length</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Field.html#AtomicVectorField.length"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Field.AtomicVectorField.length" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a field of the length of the field vectors</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Field.AtomicScalarField" title="MMTK.Field.AtomicScalarField"><tt class="xref py py-class docutils literal"><span class="pre">AtomicScalarField</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="mmtk-forcefields">
<h2>MMTK.ForceFields<a class="headerlink" href="#mmtk-forcefields" title="Permalink to this headline">¶</a></h2>
<div class="section" id="module-MMTK.ForceFields.Amber.AmberForceField">
<span id="mmtk-forcefields-amberforcefield"></span><h3>MMTK.ForceFields.AmberForceField<a class="headerlink" href="#module-MMTK.ForceFields.Amber.AmberForceField" title="Permalink to this headline">¶</a></h3>
<p>Amber force field implementation</p>
<p>General comments about parameters for Lennard-Jones and electrostatic
interactions:</p>
<p>Pair interactions in periodic systems are calculated using the
minimum-image convention; the cutoff should therefore never be
larger than half the smallest edge length of the elementary
cell.</p>
<p>For Lennard-Jones interactions, all terms for pairs whose distance
exceeds the cutoff are set to zero, without any form of correction.
For electrostatic interactions, a charge-neutralizing surface charge
density is added around the cutoff sphere in order to reduce
cutoff effects (see Wolf et al., J. Chem. Phys. 17, 8254 (1999)).</p>
<p>For Ewald summation, there are some additional parameters that can
be specified by dictionary entries:</p>
<ul>
<li><p class="first">&#8220;beta&#8221; specifies the Ewald screening parameter</p>
</li>
<li><dl class="first docutils">
<dt>&#8220;real_cutoff&#8221; specifies the cutoff for the real-space sum.</dt>
<dd><p class="first last">It should be significantly larger than 1/beta to ensure that
the neglected terms are small.</p>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>&#8220;reciprocal_cutoff&#8221; specifies the cutoff for the reciprocal-space sum.</dt>
<dd><p class="first last">Note that, like the real-space cutoff, this is a distance; it describes
the smallest wavelength of plane waves to take into account.
Consequently, a smaller value means a more precise (and more expensive)
calculation.</p>
</dd>
</dl>
</li>
</ul>
<p>MMTK provides default values for these parameter which are calculated
as a function of the system size. However, these parameters are
exaggerated in most cases of practical interest and can lead to excessive
calculation times for large systems. It is preferable to determine
suitable values empirically for the specific system to be simulated.</p>
<p>The method &#8220;screened&#8221; uses the real-space part of the Ewald sum
with a charge-neutralizing surface charge density around the
cutoff sphere, and no reciprocal sum (see article cited above).
It requires the specification of the dictionary entries &#8220;cutoff&#8221;
and &#8220;beta&#8221;.</p>
<div class="section" id="mmtk-forcefields-amber94forcefield">
<h4>MMTK.ForceFields.Amber94ForceField<a class="headerlink" href="#mmtk-forcefields-amber94forcefield" title="Permalink to this headline">¶</a></h4>
<dl class="class">
<dt id="MMTK.ForceFields.Amber.Amber94ForceField">
<em class="property">class </em><tt class="descclassname">MMTK.ForceFields.Amber.</tt><tt class="descname">Amber94ForceField</tt><big>(</big><em>lj_options=None</em>, <em>es_options=None</em>, <em>bonded_scale_factor=1.0</em>, <em>**kwargs</em><big>)</big><a class="headerlink" href="#MMTK.ForceFields.Amber.Amber94ForceField" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.ForceFields.MMForceField.MMForceField</span></tt></p>
<p>Amber 94 force field</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>lj_options</strong> &#8211; <p>parameters for Lennard-Jones
interactions; one of:</p>
<ul>
<li>a number, specifying the cutoff</li>
<li>None, meaning the default method
(no cutoff; inclusion of all
pairs, using the minimum-image
conventions for periodic universes)</li>
<li>a dictionary with an entry &#8220;method&#8221;
which specifies the calculation
method as either &#8220;direct&#8221; (all pair
terms) or &#8220;cutoff&#8221;, with the cutoff
specified by the dictionary
entry &#8220;cutoff&#8221;.</li>
</ul>
</li>
<li><strong>es_options</strong> &#8211; <p>parameters for electrostatic
interactions; one of:</p>
<ul>
<li>a number, specifying the cutoff</li>
<li>None, meaning the default method
(all pairs without cutoff for
non-periodic system, Ewald summation
for periodic systems)</li>
<li>a dictionary with an entry &#8220;method&#8221;
which specifies the calculation
method as either &#8220;direct&#8221; (all pair
terms), &#8220;cutoff&#8221; (with the cutoff
specified by the dictionary
entry &#8220;cutoff&#8221;), &#8220;ewald&#8221; (Ewald
summation, only for periodic
universes), or &#8220;screened&#8221;.</li>
</ul>
</li>
<li><strong>mod_files</strong> &#8211; a list of parameter modification files. The file
format is the one defined by AMBER. Each item
in the list can be either a file object
or a filename, filenames are looked up
first relative to the current directory and then
relative to the directory containing MMTK&#8217;s
AMBER parameter files.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="mmtk-forcefields-amber99forcefield">
<h4>MMTK.ForceFields.Amber99ForceField<a class="headerlink" href="#mmtk-forcefields-amber99forcefield" title="Permalink to this headline">¶</a></h4>
<dl class="class">
<dt id="MMTK.ForceFields.Amber.Amber99ForceField">
<em class="property">class </em><tt class="descclassname">MMTK.ForceFields.Amber.</tt><tt class="descname">Amber99ForceField</tt><big>(</big><em>lj_options=None</em>, <em>es_options=None</em>, <em>bonded_scale_factor=1.0</em>, <em>**kwargs</em><big>)</big><a class="headerlink" href="#MMTK.ForceFields.Amber.Amber99ForceField" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.ForceFields.MMForceField.MMForceField</span></tt></p>
<p>Amber 99 force field</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>lj_options</strong> &#8211; <p>parameters for Lennard-Jones
interactions; one of:</p>
<ul>
<li>a number, specifying the cutoff</li>
<li>None, meaning the default method
(no cutoff; inclusion of all
pairs, using the minimum-image
conventions for periodic universes)</li>
<li>a dictionary with an entry &#8220;method&#8221;
which specifies the calculation
method as either &#8220;direct&#8221; (all pair
terms) or &#8220;cutoff&#8221;, with the cutoff
specified by the dictionary
entry &#8220;cutoff&#8221;.</li>
</ul>
</li>
<li><strong>es_options</strong> &#8211; <p>parameters for electrostatic
interactions; one of:</p>
<ul>
<li>a number, specifying the cutoff</li>
<li>None, meaning the default method
(all pairs without cutoff for
non-periodic system, Ewald summation
for periodic systems)</li>
<li>a dictionary with an entry &#8220;method&#8221;
which specifies the calculation
method as either &#8220;direct&#8221; (all pair
terms), &#8220;cutoff&#8221; (with the cutoff
specified by the dictionary
entry &#8220;cutoff&#8221;), &#8220;ewald&#8221; (Ewald
summation, only for periodic
universes), or &#8220;screened&#8221;.</li>
</ul>
</li>
<li><strong>mod_files</strong> &#8211; a list of parameter modification files. The file
format is the one defined by AMBER. Each item
in the list can be either a file object
or a filename, filenames are looked up
first relative to the current directory and then
relative to the directory containing MMTK&#8217;s
AMBER parameter files.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="mmtk-forcefields-oplsforcefield">
<h4>MMTK.ForceFields.OPLSForceField<a class="headerlink" href="#mmtk-forcefields-oplsforcefield" title="Permalink to this headline">¶</a></h4>
<dl class="class">
<dt id="MMTK.ForceFields.Amber.OPLSForceField">
<em class="property">class </em><tt class="descclassname">MMTK.ForceFields.Amber.</tt><tt class="descname">OPLSForceField</tt><big>(</big><em>lj_options=None</em>, <em>es_options=None</em>, <em>bonded_scale_factor=1.0</em>, <em>**kwargs</em><big>)</big><a class="headerlink" href="#MMTK.ForceFields.Amber.OPLSForceField" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.ForceFields.MMForceField.MMForceField</span></tt></p>
</dd></dl>

</div>
</div>
<div class="section" id="mmtk-forcefields-anisotropicnetworkforcefield">
<h3>MMTK.ForceFields.AnisotropicNetworkForceField<a class="headerlink" href="#mmtk-forcefields-anisotropicnetworkforcefield" title="Permalink to this headline">¶</a></h3>
<dl class="class">
<dt id="MMTK.ForceFields.ANMFF.AnisotropicNetworkForceField">
<em class="property">class </em><tt class="descclassname">MMTK.ForceFields.ANMFF.</tt><tt class="descname">AnisotropicNetworkForceField</tt><big>(</big><em>cutoff=None</em>, <em>scale_factor=1.0</em><big>)</big><a class="reference internal" href="_modules/MMTK/ForceFields/ANMFF.html#AnisotropicNetworkForceField"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ForceFields.ANMFF.AnisotropicNetworkForceField" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.ForceFields.ForceField.ForceField</span></tt></p>
<p>Effective harmonic force field for an ANM protein model</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>cutoff</strong> (<em>float</em>) &#8211; the cutoff for pair interactions. Pair interactions
in periodic systems are calculated using
the minimum-image convention; the cutoff should
therefore never be larger than half the smallest
edge length of the elementary cell.</li>
<li><strong>scale_factor</strong> (<em>float</em>) &#8211; a global scaling factor</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="mmtk-forcefields-calphaforcefield">
<h3>MMTK.ForceFields.CalphaForceField<a class="headerlink" href="#mmtk-forcefields-calphaforcefield" title="Permalink to this headline">¶</a></h3>
<dl class="class">
<dt id="MMTK.ForceFields.CalphaFF.CalphaForceField">
<em class="property">class </em><tt class="descclassname">MMTK.ForceFields.CalphaFF.</tt><tt class="descname">CalphaForceField</tt><big>(</big><em>cutoff=None</em>, <em>scale_factor=1.0</em>, <em>version=1</em><big>)</big><a class="reference internal" href="_modules/MMTK/ForceFields/CalphaFF.html#CalphaForceField"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ForceFields.CalphaFF.CalphaForceField" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.ForceFields.ForceField.ForceField</span></tt></p>
<p>Effective harmonic force field for a C-alpha protein model</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>cutoff</strong> (<em>float</em>) &#8211; the cutoff for pair interactions, should be
at least 2.5 nm. Pair interactions in periodic
systems are calculated using the minimum-image
convention; the cutoff should therefore never be
larger than half the smallest edge length of the
elementary cell.</li>
<li><strong>scale_factor</strong> (<em>float</em>) &#8211; a global scaling factor</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="mmtk-forcefields-deformationforcefield">
<h3>MMTK.ForceFields.DeformationForceField<a class="headerlink" href="#mmtk-forcefields-deformationforcefield" title="Permalink to this headline">¶</a></h3>
<dl class="class">
<dt id="MMTK.ForceFields.DeformationFF.DeformationForceField">
<em class="property">class </em><tt class="descclassname">MMTK.ForceFields.DeformationFF.</tt><tt class="descname">DeformationForceField</tt><big>(</big><em>fc_length=0.7</em>, <em>cutoff=1.2</em>, <em>factor=46402.0</em><big>)</big><a class="reference internal" href="_modules/MMTK/ForceFields/DeformationFF.html#DeformationForceField"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ForceFields.DeformationFF.DeformationForceField" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.ForceFields.ForceField.ForceField</span></tt></p>
<p>Deformation force field for protein normal mode calculations</p>
<p>The pair interaction force constant has the form
k(r)=factor*exp(-(r**2-0.01)/range**2). The default value
for range is appropriate for a C-alpha model of a protein.
The offset of 0.01 is a convenience for defining factor;
with this definition, factor is the force constant for a
distance of 0.1nm.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>fc_length</strong> (<em>float</em>) &#8211; a range parameter</li>
<li><strong>cutoff</strong> (<em>float</em>) &#8211; the cutoff for pair interactions, should be
at least 2.5 nm. Pair interactions in periodic
systems are calculated using the minimum-image
convention; the cutoff should therefore never be
larger than half the smallest edge length of the
elementary cell.</li>
<li><strong>factor</strong> (<em>float</em>) &#8211; a global scaling factor</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="mmtk-forcefields-harmonicforcefield">
<h3>MMTK.ForceFields.HarmonicForceField<a class="headerlink" href="#mmtk-forcefields-harmonicforcefield" title="Permalink to this headline">¶</a></h3>
<dl class="class">
<dt id="MMTK.ForceFields.BondFF.HarmonicForceField">
<em class="property">class </em><tt class="descclassname">MMTK.ForceFields.BondFF.</tt><tt class="descname">HarmonicForceField</tt><big>(</big><em>fc_length=0.45</em>, <em>cutoff=0.6</em>, <em>factor=400.0</em><big>)</big><a class="reference internal" href="_modules/MMTK/ForceFields/BondFF.html#HarmonicForceField"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ForceFields.BondFF.HarmonicForceField" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.ForceFields.MMForceField.MMAtomParameters</span></tt>, <tt class="xref py py-class docutils literal"><span class="pre">MMTK.ForceFields.ForceField.CompoundForceField</span></tt></p>
<p>Simplified harmonic force field for normal mode calculations</p>
<p>This force field is made up of the bonded terms from the Amber 94
force field with the equilibrium positions of all terms changed
to the corresponding values in the input configuration, such that
the input configuration becomes an energy minimum by construction.
The nonbonded terms are replaced by a generic short-ranged
deformation term.</p>
<dl class="docutils">
<dt>For a description of this force field, see:</dt>
<dd><div class="first last line-block">
<div class="line">Hinsen &amp; Kneller, J. Chem. Phys. 24, 10766 (1999)</div>
</div>
</dd>
<dt>For an application to DNA, see:</dt>
<dd><div class="first last line-block">
<div class="line">Viduna, Hinsen &amp; Kneller, Phys. Rev. E 3, 3986 (2000)</div>
</div>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="mmtk-forcefields-lennardjonesforcefield">
<h3>MMTK.ForceFields.LennardJonesForceField<a class="headerlink" href="#mmtk-forcefields-lennardjonesforcefield" title="Permalink to this headline">¶</a></h3>
<dl class="class">
<dt id="MMTK.ForceFields.LennardJonesFF.LennardJonesForceField">
<em class="property">class </em><tt class="descclassname">MMTK.ForceFields.LennardJonesFF.</tt><tt class="descname">LennardJonesForceField</tt><big>(</big><em>cutoff=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/ForceFields/LennardJonesFF.html#LennardJonesForceField"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ForceFields.LennardJonesFF.LennardJonesForceField" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.ForceFields.NonBondedInteractions.LJForceField</span></tt></p>
<p>Lennard-Jones force field</p>
<p>The Lennard-Jones parameters are taken from the atom attributes
LJ_radius and LJ_energy. The pair interaction energy has the form
U(r)=4*LJ_energy*((LJ_radius/r)**12-(LJ_radius/r)**6).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>cutoff</strong> (<em>float</em>) &#8211; a cutoff value or None, meaning no cutoff.
Pair interactions in periodic systems are calculated
using the minimum-image convention; the cutoff should
therefore never be larger than half the smallest edge
length of the elementary cell.</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="mmtk-forcefields-spceforcefield">
<h3>MMTK.ForceFields.SPCEForceField<a class="headerlink" href="#mmtk-forcefields-spceforcefield" title="Permalink to this headline">¶</a></h3>
<dl class="class">
<dt id="MMTK.ForceFields.SPCEFF.SPCEForceField">
<em class="property">class </em><tt class="descclassname">MMTK.ForceFields.SPCEFF.</tt><tt class="descname">SPCEForceField</tt><big>(</big><em>lj_options=None</em>, <em>es_options=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/ForceFields/SPCEFF.html#SPCEForceField"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ForceFields.SPCEFF.SPCEForceField" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.ForceFields.MMForceField.MMForceField</span></tt></p>
<p>Force field for water simulations with the SPC/E model</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>lj_options</strong> &#8211; <p>parameters for Lennard-Jones
interactions; one of:</p>
<blockquote>
<div><ul>
<li>a number, specifying the cutoff</li>
<li>None, meaning the default method
(no cutoff; inclusion of all
pairs, using the minimum-image
conventions for periodic universes)</li>
<li>a dictionary with an entry &#8220;method&#8221;
which specifies the calculation
method as either &#8220;direct&#8221; (all pair
terms) or &#8220;cutoff&#8221;, with the cutoff
specified by the dictionary
entry &#8220;cutoff&#8221;.</li>
</ul>
</div></blockquote>
</li>
<li><strong>es_options</strong> &#8211; <p>parameters for electrostatic
interactions; one of:</p>
<blockquote>
<div><ul>
<li>a number, specifying the cutoff</li>
<li>None, meaning the default method
(all pairs without cutoff for
non-periodic system, Ewald summation
for periodic systems)</li>
<li>a dictionary with an entry &#8220;method&#8221;
which specifies the calculation
method as either &#8220;direct&#8221; (all pair
terms), &#8220;cutoff&#8221; (with the cutoff
specified by the dictionary
entry &#8220;cutoff&#8221;), &#8220;ewald&#8221; (Ewald
summation, only for periodic
universes), &#8220;screened&#8221; or
&#8220;multipole&#8221; (fast-multipole method).</li>
</ul>
</div></blockquote>
</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.ForceFields.ForceFieldTest">
<span id="mmtk-forcefields-forcefieldtest"></span><h3>MMTK.ForceFields.ForceFieldTest<a class="headerlink" href="#module-MMTK.ForceFields.ForceFieldTest" title="Permalink to this headline">¶</a></h3>
<p>Force field consistency tests</p>
<dl class="function">
<dt id="MMTK.ForceFields.ForceFieldTest.forceConstantTest">
<tt class="descclassname">MMTK.ForceFields.ForceFieldTest.</tt><tt class="descname">forceConstantTest</tt><big>(</big><em>universe</em>, <em>atoms=None</em>, <em>delta=0.0001</em><big>)</big><a class="reference internal" href="_modules/MMTK/ForceFields/ForceFieldTest.html#forceConstantTest"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ForceFields.ForceFieldTest.forceConstantTest" title="Permalink to this definition">¶</a></dt>
<dd><p>Test force constants by comparing to the numerical derivatives
of the gradients.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe on which the test is performed</li>
<li><strong>atoms</strong> (<em>list</em>) &#8211; the atoms of the universe for which the gradient
is tested (default: all atoms)</li>
<li><strong>delta</strong> (<em>float</em>) &#8211; the step size used in calculating the numerical derivatives</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.ForceFields.ForceFieldTest.gradientTest">
<tt class="descclassname">MMTK.ForceFields.ForceFieldTest.</tt><tt class="descname">gradientTest</tt><big>(</big><em>universe</em>, <em>atoms=None</em>, <em>delta=0.0001</em><big>)</big><a class="reference internal" href="_modules/MMTK/ForceFields/ForceFieldTest.html#gradientTest"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ForceFields.ForceFieldTest.gradientTest" title="Permalink to this definition">¶</a></dt>
<dd><p>Test gradients by comparing to numerical derivatives of the energy.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe on which the test is performed</li>
<li><strong>atoms</strong> (<em>list</em>) &#8211; the atoms of the universe for which the gradient
is tested (default: all atoms)</li>
<li><strong>delta</strong> (<em>float</em>) &#8211; the step size used in calculating the numerical derivatives</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.ForceFields.ForceFieldTest.virialTest">
<tt class="descclassname">MMTK.ForceFields.ForceFieldTest.</tt><tt class="descname">virialTest</tt><big>(</big><em>universe</em><big>)</big><a class="reference internal" href="_modules/MMTK/ForceFields/ForceFieldTest.html#virialTest"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ForceFields.ForceFieldTest.virialTest" title="Permalink to this definition">¶</a></dt>
<dd><p>Test the virial by comparing to an explicit computation from
positions and gradients.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe on which the test is performed</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.ForceFields.Restraints">
<span id="mmtk-forcefields-restraints"></span><h3>MMTK.ForceFields.Restraints<a class="headerlink" href="#module-MMTK.ForceFields.Restraints" title="Permalink to this headline">¶</a></h3>
<p>Harmonic restraint terms that can be added to any force field</p>
<p>Example:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="kn">from</span> <span class="nn">MMTK</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">MMTK.ForceFields</span> <span class="kn">import</span> <span class="n">Amber99ForceField</span>
<span class="kn">from</span> <span class="nn">MMTK.ForceFields.Restraints</span> <span class="kn">import</span> <span class="n">HarmonicDistanceRestraint</span>

<span class="n">universe</span> <span class="o">=</span> <span class="n">InfiniteUniverse</span><span class="p">()</span>
<span class="n">universe</span><span class="o">.</span><span class="n">protein</span> <span class="o">=</span> <span class="n">Protein</span><span class="p">(</span><span class="s">&#39;bala1&#39;</span><span class="p">)</span>
<span class="n">force_field</span> <span class="o">=</span> <span class="n">Amber99ForceField</span><span class="p">()</span> <span class="o">+</span>                <span class="n">HarmonicDistanceRestraint</span><span class="p">(</span><span class="n">universe</span><span class="o">.</span><span class="n">protein</span><span class="p">[</span><span class="mi">0</span><span class="p">][</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">peptide</span><span class="o">.</span><span class="n">N</span><span class="p">,</span>
                                        <span class="n">universe</span><span class="o">.</span><span class="n">protein</span><span class="p">[</span><span class="mi">0</span><span class="p">][</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">peptide</span><span class="o">.</span><span class="n">O</span><span class="p">,</span>
                                        <span class="mf">0.5</span><span class="p">,</span> <span class="mf">10.</span><span class="p">)</span>
<span class="n">universe</span><span class="o">.</span><span class="n">setForceField</span><span class="p">(</span><span class="n">force_field</span><span class="p">)</span>
</pre></div>
</div>
<dl class="class">
<dt id="MMTK.ForceFields.Restraints.HarmonicAngleRestraint">
<em class="property">class </em><tt class="descclassname">MMTK.ForceFields.Restraints.</tt><tt class="descname">HarmonicAngleRestraint</tt><big>(</big><em>atom1</em>, <em>atom2</em>, <em>atom3</em>, <em>angle</em>, <em>force_constant</em><big>)</big><a class="reference internal" href="_modules/MMTK/ForceFields/Restraints.html#HarmonicAngleRestraint"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ForceFields.Restraints.HarmonicAngleRestraint" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.ForceFields.ForceField.ForceField</span></tt></p>
<p>Harmonic angle restraint between three atoms</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>atom1</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; first atom</li>
<li><strong>atom2</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; second (central) atom</li>
<li><strong>atom3</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; third atom</li>
<li><strong>angle</strong> (<em>float</em>) &#8211; the angle at which the restraint is zero</li>
<li><strong>force_constant</strong> &#8211; the force constant of the restraint term.
The functional form of the restraint is
force_constant*(phi-angle)**2, where
phi is the angle atom1-atom2-atom3.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.ForceFields.Restraints.HarmonicDihedralRestraint">
<em class="property">class </em><tt class="descclassname">MMTK.ForceFields.Restraints.</tt><tt class="descname">HarmonicDihedralRestraint</tt><big>(</big><em>atom1</em>, <em>atom2</em>, <em>atom3</em>, <em>atom4</em>, <em>dihedral</em>, <em>force_constant</em><big>)</big><a class="reference internal" href="_modules/MMTK/ForceFields/Restraints.html#HarmonicDihedralRestraint"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ForceFields.Restraints.HarmonicDihedralRestraint" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.ForceFields.ForceField.ForceField</span></tt></p>
<p>Harmonic dihedral angle restraint between four atoms</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>atom1</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; first atom</li>
<li><strong>atom2</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; second (axis) atom</li>
<li><strong>atom3</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; third (axis)atom</li>
<li><strong>atom4</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; fourth atom</li>
<li><strong>dihedral</strong> (<em>float</em>) &#8211; the dihedral angle at which the restraint is zero</li>
<li><strong>force_constant</strong> &#8211; the force constant of the restraint term.
The functional form of the restraint is
force_constant*(phi-abs(dihedral))**2, where
phi is the dihedral angle
atom1-atom2-atom3-atom4.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.ForceFields.Restraints.HarmonicDistanceRestraint">
<em class="property">class </em><tt class="descclassname">MMTK.ForceFields.Restraints.</tt><tt class="descname">HarmonicDistanceRestraint</tt><big>(</big><em>obj1</em>, <em>obj2</em>, <em>distance</em>, <em>force_constant</em>, <em>nb_exclusion=False</em><big>)</big><a class="reference internal" href="_modules/MMTK/ForceFields/Restraints.html#HarmonicDistanceRestraint"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ForceFields.Restraints.HarmonicDistanceRestraint" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.ForceFields.ForceField.ForceField</span></tt></p>
<p>Harmonic distance restraint between two atoms</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>obj1</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; the object defining center-of-mass 1</li>
<li><strong>obj2</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; the object defining center-of-mass 2</li>
<li><strong>distance</strong> (<em>float</em>) &#8211; the distance between cm 1 and cm2 at which
the restraint is zero</li>
<li><strong>force_constant</strong> (<em>float</em>) &#8211; the force constant of the restraint term.
The functional form of the restraint is
force_constant*((cm1-cm2).length()-distance)**2,
where cm1 and cm2 are the centrer-of-mass
positions of the two objects.</li>
<li><strong>nb_exclussion</strong> (<em>bool</em>) &#8211; if True, non-bonded interactions between
the restrained atoms are suppressed, as
for a chemical bond</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.ForceFields.Restraints.HarmonicTrapForceField">
<em class="property">class </em><tt class="descclassname">MMTK.ForceFields.Restraints.</tt><tt class="descname">HarmonicTrapForceField</tt><big>(</big><em>obj</em>, <em>center</em>, <em>force_constant</em><big>)</big><a class="reference internal" href="_modules/MMTK/ForceFields/Restraints.html#HarmonicTrapForceField"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ForceFields.Restraints.HarmonicTrapForceField" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.ForceFields.ForceField.ForceField</span></tt></p>
<p>Harmonic potential with respect to a fixed point in space</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>obj</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; the object on whose center of mass the force field acts</li>
<li><strong>center</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the point to which the atom is attached by
the harmonic potential</li>
<li><strong>force_constant</strong> (<em>float</em>) &#8211; the force constant of the harmonic potential</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
</div>
<div class="section" id="module-MMTK.FourierBasis">
<span id="mmtk-fourierbasis"></span><h2>MMTK.FourierBasis<a class="headerlink" href="#module-MMTK.FourierBasis" title="Permalink to this headline">¶</a></h2>
<p>Fourier basis for low-frequency normal mode calculations</p>
<p>This module provides a basis that is suitable for the
calculation of low-frequency normal modes. The basis is
derived from vector fields whose components are stationary
waves in a box surrounding the system. For a description,
see</p>
<blockquote>
<div><div class="line-block">
<div class="line">K. Hinsen</div>
<div class="line">Analysis of domain motions by approximate normal mode calculations</div>
<div class="line">Proteins 33 (1998): 417-429</div>
</div>
</div></blockquote>
<dl class="class">
<dt id="MMTK.FourierBasis.FourierBasis">
<em class="property">class </em><tt class="descclassname">MMTK.FourierBasis.</tt><tt class="descname">FourierBasis</tt><big>(</big><em>universe</em>, <em>cutoff</em><big>)</big><a class="reference internal" href="_modules/MMTK/FourierBasis.html#FourierBasis"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.FourierBasis.FourierBasis" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Collective-motion basis for low-frequency normal mode calculations</p>
<p>A FourierBasis behaves like a sequence of
<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a> objects. The vectors are
B{not} orthonormal, because orthonormalization is handled
automatically by the normal mode class.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe for which the basis will be used</li>
<li><strong>cutoff</strong> (<em>float</em>) &#8211; the wavelength cutoff. A smaller value yields
a larger basis.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.FourierBasis.countBasisVectors">
<tt class="descclassname">MMTK.FourierBasis.</tt><tt class="descname">countBasisVectors</tt><big>(</big><em>universe</em>, <em>cutoff</em><big>)</big><a class="reference internal" href="_modules/MMTK/FourierBasis.html#countBasisVectors"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.FourierBasis.countBasisVectors" title="Permalink to this definition">¶</a></dt>
<dd><p>Estimate the number of basis vectors for a given universe and cutoff</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe</li>
<li><strong>cutoff</strong> (<em>float</em>) &#8211; the wavelength cutoff. A smaller value yields a larger basis.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the number of basis vectors in a FourierBasis</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">int</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.FourierBasis.estimateCutoff">
<tt class="descclassname">MMTK.FourierBasis.</tt><tt class="descname">estimateCutoff</tt><big>(</big><em>universe</em>, <em>nmodes</em><big>)</big><a class="reference internal" href="_modules/MMTK/FourierBasis.html#estimateCutoff"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.FourierBasis.estimateCutoff" title="Permalink to this definition">¶</a></dt>
<dd><p>Estimate the cutoff that yields a given number of basis vectors
for a given universe.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe</li>
<li><strong>nmodes</strong> (<em>int</em>) &#8211; the number of basis vectors in a FourierBasis</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the wavelength cutoff and the precise number of basis vectors</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">tuple (float, int)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.Geometry">
<span id="mmtk-geometry"></span><h2>MMTK.Geometry<a class="headerlink" href="#module-MMTK.Geometry" title="Permalink to this headline">¶</a></h2>
<p>Elementary geometrical objects and operations</p>
<p>There are essentially two kinds of geometrical objects: shape objects
(spheres, planes, etc.), from which intersections can be calculated,
and lattice objects, which define a regular arrangements of points.</p>
<dl class="class">
<dt id="MMTK.Geometry.BCCLattice">
<em class="property">class </em><tt class="descclassname">MMTK.Geometry.</tt><tt class="descname">BCCLattice</tt><big>(</big><em>cellsize</em>, <em>cells</em>, <em>function=None</em>, <em>base=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#BCCLattice"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.BCCLattice" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Geometry.RhombicLattice" title="MMTK.Geometry.RhombicLattice"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Geometry.RhombicLattice</span></tt></a></p>
<p>Body-Centered Cubic lattice</p>
<p>A Body-Centered Cubic lattice has two points per elementary cell.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>cellsize</strong> (<em>float</em>) &#8211; the edge length of the elementary cell</li>
<li><strong>cells</strong> &#8211; a list of integers, whose length must equal the number
of dimensions. Each entry specifies how often a cell is
repeated along this dimension.</li>
<li><strong>function</strong> &#8211; a function that is called for every lattice point with
the vector describing the point as argument. The return
value of this function is stored in the lattice object.
If the function is &#8216;None&#8217;, the vector is directly
stored in the lattice object.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Geometry.Box">
<em class="property">class </em><tt class="descclassname">MMTK.Geometry.</tt><tt class="descname">Box</tt><big>(</big><em>corner1</em>, <em>corner2</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#Box"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.Box" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Geometry.GeometricalObject3D" title="MMTK.Geometry.GeometricalObject3D"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Geometry.GeometricalObject3D</span></tt></a></p>
<p>Rectangular box aligned with the coordinate axes</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>corner1</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; one corner of the box</li>
<li><strong>corner2</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the diagonally opposite corner</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Geometry.BravaisLattice">
<em class="property">class </em><tt class="descclassname">MMTK.Geometry.</tt><tt class="descname">BravaisLattice</tt><big>(</big><em>lattice_vectors</em>, <em>cells</em>, <em>function=None</em>, <em>base=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#BravaisLattice"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.BravaisLattice" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Geometry.RhombicLattice" title="MMTK.Geometry.RhombicLattice"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Geometry.RhombicLattice</span></tt></a></p>
<p>Bravais lattice</p>
<p>A Bravais lattice is a special case of a general rhombic lattice
in which the elementary cell contains only one point.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>lattice_vectors</strong> &#8211; a list of lattice vectors. Each lattice
vector defines a lattice dimension (only
values from one to three make sense) and
indicates the displacement along this
dimension from one cell to the next.</li>
<li><strong>cells</strong> &#8211; a list of integers, whose length must equal the number
of dimensions. Each entry specifies how often a cell is
repeated along this dimension.</li>
<li><strong>function</strong> &#8211; a function that is called for every lattice point with
the vector describing the point as argument. The return
value of this function is stored in the lattice object.
If the function is &#8216;None&#8217;, the vector is directly
stored in the lattice object.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Geometry.Circle">
<em class="property">class </em><tt class="descclassname">MMTK.Geometry.</tt><tt class="descname">Circle</tt><big>(</big><em>center</em>, <em>normal</em>, <em>radius</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#Circle"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.Circle" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Geometry.GeometricalObject3D" title="MMTK.Geometry.GeometricalObject3D"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Geometry.GeometricalObject3D</span></tt></a></p>
<p>2D circle in 3D space</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>center</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the center of the circle</li>
<li><strong>normal</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the normal vector of the circle&#8217;s plane</li>
<li><strong>radius</strong> (<em>float</em>) &#8211; the radius of the circle</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Geometry.Cone">
<em class="property">class </em><tt class="descclassname">MMTK.Geometry.</tt><tt class="descname">Cone</tt><big>(</big><em>center</em>, <em>axis</em>, <em>angle</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#Cone"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.Cone" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Geometry.GeometricalObject3D" title="MMTK.Geometry.GeometricalObject3D"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Geometry.GeometricalObject3D</span></tt></a></p>
<p>Cone</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>center</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the center (tip) of the cone</li>
<li><strong>axis</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the direction of the axis of rotational symmetry</li>
<li><strong>angle</strong> (<em>float</em>) &#8211; the angle between any straight line on the cone
surface and the axis of symmetry</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Geometry.Cylinder">
<em class="property">class </em><tt class="descclassname">MMTK.Geometry.</tt><tt class="descname">Cylinder</tt><big>(</big><em>center1</em>, <em>center2</em>, <em>radius</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#Cylinder"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.Cylinder" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Geometry.GeometricalObject3D" title="MMTK.Geometry.GeometricalObject3D"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Geometry.GeometricalObject3D</span></tt></a></p>
<p>Cylinder</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>center1</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the center of the bottom circle</li>
<li><strong>center2</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the center of the top circle</li>
<li><strong>radius</strong> (<em>float</em>) &#8211; the radius of the cylinder</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Geometry.FCCLattice">
<em class="property">class </em><tt class="descclassname">MMTK.Geometry.</tt><tt class="descname">FCCLattice</tt><big>(</big><em>cellsize</em>, <em>cells</em>, <em>function=None</em>, <em>base=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#FCCLattice"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.FCCLattice" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Geometry.RhombicLattice" title="MMTK.Geometry.RhombicLattice"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Geometry.RhombicLattice</span></tt></a></p>
<p>Face-Centered Cubic lattice</p>
<p>A Face-Centered Cubic lattice has four points per elementary cell.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>cellsize</strong> (<em>float</em>) &#8211; the edge length of the elementary cell</li>
<li><strong>cells</strong> &#8211; a list of integers, whose length must equal the number
of dimensions. Each entry specifies how often a cell is
repeated along this dimension.</li>
<li><strong>function</strong> &#8211; a function that is called for every lattice point with
the vector describing the point as argument. The return
value of this function is stored in the lattice object.
If the function is &#8216;None&#8217;, the vector is directly
stored in the lattice object.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Geometry.GeometricalObject3D">
<em class="property">class </em><tt class="descclassname">MMTK.Geometry.</tt><tt class="descname">GeometricalObject3D</tt><a class="reference internal" href="_modules/MMTK/Geometry.html#GeometricalObject3D"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.GeometricalObject3D" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>3D shape object</p>
<p>This is an abstract base class. To create 3D objects,
use one of its subclasses.</p>
<dl class="method">
<dt id="MMTK.Geometry.GeometricalObject3D.enclosesPoint">
<tt class="descname">enclosesPoint</tt><big>(</big><em>point</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#GeometricalObject3D.enclosesPoint"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.GeometricalObject3D.enclosesPoint" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>point</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; a point in 3D space</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">True of the point is inside the volume of the object</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">bool</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Geometry.GeometricalObject3D.hasPoint">
<tt class="descname">hasPoint</tt><big>(</big><em>point</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#GeometricalObject3D.hasPoint"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.GeometricalObject3D.hasPoint" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>point</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; a point in 3D space</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">True of the point lies on the surface of the object</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">bool</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Geometry.GeometricalObject3D.intersectWith">
<tt class="descname">intersectWith</tt><big>(</big><em>other</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#GeometricalObject3D.intersectWith"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.GeometricalObject3D.intersectWith" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>other</strong> &#8211; another 3D object</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a 3D object that represents the intersection with other</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Geometry.GeometricalObject3D.volume">
<tt class="descname">volume</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#GeometricalObject3D.volume"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.GeometricalObject3D.volume" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the volume of the object</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Geometry.Lattice">
<em class="property">class </em><tt class="descclassname">MMTK.Geometry.</tt><tt class="descname">Lattice</tt><big>(</big><em>function</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#Lattice"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.Lattice" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>General lattice</p>
<p>Lattices are special sequence objects that contain vectors
(points on the lattice) or objects that are constructed as
functions of these vectors. Lattice objects behave like
lists, i.e. they permit indexing, length inquiry, and iteration
by &#8216;for&#8217;-loops. Note that the lattices represented by these
objects are finite, they have a finite (and fixed) number
of repetitions along each lattice vector.</p>
<p>This is an abstract base class. To create lattice objects,
use one of its subclasses.</p>
</dd></dl>

<dl class="class">
<dt id="MMTK.Geometry.Line">
<em class="property">class </em><tt class="descclassname">MMTK.Geometry.</tt><tt class="descname">Line</tt><big>(</big><em>point</em>, <em>direction</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#Line"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.Line" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Geometry.GeometricalObject3D" title="MMTK.Geometry.GeometricalObject3D"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Geometry.GeometricalObject3D</span></tt></a></p>
<p>Line</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>point</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; any point on the line</li>
<li><strong>direction</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the direction of the line</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.Geometry.Line.distanceFrom">
<tt class="descname">distanceFrom</tt><big>(</big><em>point</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#Line.distanceFrom"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.Line.distanceFrom" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>point</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; a point in space</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the smallest distance of the point from the line</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Geometry.Line.perpendicularVector">
<tt class="descname">perpendicularVector</tt><big>(</big><em>plane</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#Line.perpendicularVector"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.Line.perpendicularVector" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>plane</strong> (<em>Plane</em>) &#8211; a plane</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a vector in the plane perpendicular to the line</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Vector</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Geometry.Line.projectionOf">
<tt class="descname">projectionOf</tt><big>(</big><em>point</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#Line.projectionOf"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.Line.projectionOf" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>point</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; a point in space</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the orthogonal projection of the point onto the line</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Vector</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Geometry.Plane">
<em class="property">class </em><tt class="descclassname">MMTK.Geometry.</tt><tt class="descname">Plane</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#Plane"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.Plane" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Geometry.GeometricalObject3D" title="MMTK.Geometry.GeometricalObject3D"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Geometry.GeometricalObject3D</span></tt></a></p>
<p>2D plane in 3D space</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>args</strong> &#8211; three points (of type Scientific.Geometry.Vector)
that are not collinear, or a point in the plane and
the normal vector of the plane</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Geometry.RhombicLattice">
<em class="property">class </em><tt class="descclassname">MMTK.Geometry.</tt><tt class="descname">RhombicLattice</tt><big>(</big><em>elementary_cell</em>, <em>lattice_vectors</em>, <em>cells</em>, <em>function=None</em>, <em>base=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#RhombicLattice"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.RhombicLattice" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Geometry.Lattice" title="MMTK.Geometry.Lattice"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Geometry.Lattice</span></tt></a></p>
<p>Rhombic lattice</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>elementary_cell</strong> &#8211; a list of points in the elementary cell</li>
<li><strong>lattice_vectors</strong> &#8211; a list of lattice vectors. Each lattice
vector defines a lattice dimension (only
values from one to three make sense) and
indicates the displacement along this
dimension from one cell to the next.</li>
<li><strong>cells</strong> &#8211; a list of integers, whose length must equal the number
of dimensions. Each entry specifies how often a cell is
repeated along this dimension.</li>
<li><strong>function</strong> &#8211; a function that is called for every lattice point with
the vector describing the point as argument. The return
value of this function is stored in the lattice object.
If the function is &#8216;None&#8217;, the vector is directly
stored in the lattice object.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Geometry.SCLattice">
<em class="property">class </em><tt class="descclassname">MMTK.Geometry.</tt><tt class="descname">SCLattice</tt><big>(</big><em>cellsize</em>, <em>cells</em>, <em>function=None</em>, <em>base=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#SCLattice"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.SCLattice" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Geometry.BravaisLattice" title="MMTK.Geometry.BravaisLattice"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Geometry.BravaisLattice</span></tt></a></p>
<p>Simple Cubic lattice</p>
<p>A Simple Cubic lattice is a special case of a Bravais lattice
in which the elementary cell is a cube.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>cellsize</strong> (<em>float</em>) &#8211; the edge length of the elementary cell</li>
<li><strong>cells</strong> &#8211; a list of integers, whose length must equal the number
of dimensions. Each entry specifies how often a cell is
repeated along this dimension.</li>
<li><strong>function</strong> &#8211; a function that is called for every lattice point with
the vector describing the point as argument. The return
value of this function is stored in the lattice object.
If the function is &#8216;None&#8217;, the vector is directly
stored in the lattice object.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Geometry.Sphere">
<em class="property">class </em><tt class="descclassname">MMTK.Geometry.</tt><tt class="descname">Sphere</tt><big>(</big><em>center</em>, <em>radius</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#Sphere"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.Sphere" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Geometry.GeometricalObject3D" title="MMTK.Geometry.GeometricalObject3D"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Geometry.GeometricalObject3D</span></tt></a></p>
<p>Sphere</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>center</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the center of the sphere</li>
<li><strong>radius</strong> (<em>float</em>) &#8211; the radius of the sphere</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.Geometry.superpositionFit">
<tt class="descclassname">MMTK.Geometry.</tt><tt class="descname">superpositionFit</tt><big>(</big><em>confs</em><big>)</big><a class="reference internal" href="_modules/MMTK/Geometry.html#superpositionFit"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Geometry.superpositionFit" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>confs</strong> (<em>sequence of (float, Vector, Vector)</em>) &#8211; the weight, reference position, and alternate
position for each atom</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the quaternion representing the rotation,
the center of mass in the alternate configuraton,
the center of mass in the reference configuration,
and the RMS distance after the optimal superposition</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.InternalCoordinates">
<span id="mmtk-internalcoordinates"></span><h2>MMTK.InternalCoordinates<a class="headerlink" href="#module-MMTK.InternalCoordinates" title="Permalink to this headline">¶</a></h2>
<p>Manipulation of molecular configurations in terms of internal coordinates</p>
<dl class="class">
<dt id="MMTK.InternalCoordinates.BondAngle">
<em class="property">class </em><tt class="descclassname">MMTK.InternalCoordinates.</tt><tt class="descname">BondAngle</tt><big>(</big><em>atom1</em>, <em>atom2</em>, <em>atom3</em><big>)</big><a class="reference internal" href="_modules/MMTK/InternalCoordinates.html#BondAngle"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.InternalCoordinates.BondAngle" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.InternalCoordinates.InternalCoordinate</span></tt></p>
<p>Bond angle</p>
<p>A BondAngle object permits calculating and modifying the
angle between two bonds in a molecule, under the condition that
the bonds are not part of a circular bond structure (but it
is not a problem to have a circular bond structure elsewhere
in the molecule). Modifying the bond angle rotates the parts
of the molecule on both sides of the central atom around
an axis passing through the central atom and perpendicular
to the plane defined by the two bonds in such a way that
there is no overall rotation of the molecule. The central
atom and any other atoms bonded to it do not move.</p>
<p>The initial construction of the BondAngle object can be
expensive (the bond structure of the molecule must be
analyzed). It is therefore advisable to keep the object
rather than recreate it frequently. Note that if you only
want to calculate bond angles (no modification), the
method Universe.angle is simpler and faster.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>atom1</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; the first atom that defines the angle</li>
<li><strong>atom2</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; the second and central atom that defines the bond</li>
<li><strong>atom3</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; the third atom that defines the bond</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.InternalCoordinates.BondAngle.getValue">
<tt class="descname">getValue</tt><big>(</big><em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/InternalCoordinates.html#BondAngle.getValue"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.InternalCoordinates.BondAngle.getValue" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>conf</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a>) &#8211; a configuration (defaults to the current configuration)</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the size of the angle in the configuration conf</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.InternalCoordinates.BondAngle.setValue">
<tt class="descname">setValue</tt><big>(</big><em>value</em><big>)</big><a class="reference internal" href="_modules/MMTK/InternalCoordinates.html#BondAngle.setValue"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.InternalCoordinates.BondAngle.setValue" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets the size of the angle
:param value: the desired angle
:type value: float</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.InternalCoordinates.BondLength">
<em class="property">class </em><tt class="descclassname">MMTK.InternalCoordinates.</tt><tt class="descname">BondLength</tt><big>(</big><em>atom1</em>, <em>atom2</em><big>)</big><a class="reference internal" href="_modules/MMTK/InternalCoordinates.html#BondLength"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.InternalCoordinates.BondLength" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.InternalCoordinates.InternalCoordinate</span></tt></p>
<p>Bond length coordinate</p>
<p>A BondLength object permits calculating and modifying the
length of a bond in a molecule, under the condition that
the bond is not part of a circular bond structure (but it
is not a problem to have a circular bond structure elsewhere
in the molecule). Modifying the bond length moves the parts
of the molecule on both sides of the bond along the bond direction
in such a way that the center of mass of the molecule does not
change.</p>
<p>The initial construction of the BondLength object can be
expensive (the bond structure of the molecule must be
analyzed). It is therefore advisable to keep the object
rather than recreate it frequently. Note that if you only
want to calculate bond lengths (no modification), the
method Universe.distance is simpler and faster.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>atom1</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; the first atom that defines the bond</li>
<li><strong>atom2</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; the second atom that defines the bond</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.InternalCoordinates.BondLength.getValue">
<tt class="descname">getValue</tt><big>(</big><em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/InternalCoordinates.html#BondLength.getValue"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.InternalCoordinates.BondLength.getValue" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>conf</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a>) &#8211; a configuration (defaults to the current configuration)</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the length of the bond in the configuration conf</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.InternalCoordinates.BondLength.setValue">
<tt class="descname">setValue</tt><big>(</big><em>value</em><big>)</big><a class="reference internal" href="_modules/MMTK/InternalCoordinates.html#BondLength.setValue"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.InternalCoordinates.BondLength.setValue" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets the length of the bond
:param value: the desired length of the bond
:type value: float</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.InternalCoordinates.DihedralAngle">
<em class="property">class </em><tt class="descclassname">MMTK.InternalCoordinates.</tt><tt class="descname">DihedralAngle</tt><big>(</big><em>atom1</em>, <em>atom2</em>, <em>atom3</em>, <em>atom4</em><big>)</big><a class="reference internal" href="_modules/MMTK/InternalCoordinates.html#DihedralAngle"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.InternalCoordinates.DihedralAngle" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.InternalCoordinates.InternalCoordinate</span></tt></p>
<p>Dihedral angle</p>
<p>A DihedralAngle object permits calculating and modifying the
dihedral angle defined by three consecutive bonds in a molecule,
under the condition that the central bond is not part of a
circular bond structure (but it is not a problem to have a
circular bond structure elsewhere in the molecule). Modifying the
dihedral angle rotates the parts of the molecule on both sides of
the central bond around this central bond in such a way that there
is no overall rotation of the molecule.</p>
<p>The initial construction of the DihedralAngle object can be
expensive (the bond structure of the molecule must be
analyzed). It is therefore advisable to keep the object
rather than recreate it frequently. Note that if you only
want to calculate bond angles (no modification), the
method Universe.dihedral is simpler and faster.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>atom1</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; the first atom that defines the dihedral</li>
<li><strong>atom2</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; the second atom that defines the dihedral,
must be on the central bond</li>
<li><strong>atom3</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; the third atom that defines the dihedral,
must be on the central bond</li>
<li><strong>atom4</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; the fourth atom that defines the dihedral</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.InternalCoordinates.DihedralAngle.getValue">
<tt class="descname">getValue</tt><big>(</big><em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/InternalCoordinates.html#DihedralAngle.getValue"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.InternalCoordinates.DihedralAngle.getValue" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>conf</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a>) &#8211; a configuration (defaults to the current configuration)</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the size of the dihedral angle in the configuration conf</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.InternalCoordinates.DihedralAngle.setValue">
<tt class="descname">setValue</tt><big>(</big><em>value</em><big>)</big><a class="reference internal" href="_modules/MMTK/InternalCoordinates.html#DihedralAngle.setValue"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.InternalCoordinates.DihedralAngle.setValue" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets the size of the dihedral
:param value: the desired dihedral angle
:type value: float</p>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-MMTK.Minimization">
<span id="mmtk-minimization"></span><h2>MMTK.Minimization<a class="headerlink" href="#module-MMTK.Minimization" title="Permalink to this headline">¶</a></h2>
<p>Energy minimizers</p>
<dl class="class">
<dt id="MMTK.Minimization.ConjugateGradientMinimizer">
<em class="property">class </em><tt class="descclassname">MMTK.Minimization.</tt><tt class="descname">ConjugateGradientMinimizer</tt><big>(</big><em>universe</em>, <em>**options</em><big>)</big><a class="reference internal" href="_modules/MMTK/Minimization.html#ConjugateGradientMinimizer"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Minimization.ConjugateGradientMinimizer" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.Minimization.Minimizer</span></tt></p>
<p>Conjugate gradient minimizer</p>
<p>The minimizer can handle fixed atoms, but no distance constraints.
It is fully thread-safe.</p>
<p>The minimization is started by calling the minimizer object.
All the keyword options can be specified either when
creating the minimizer or when calling it.</p>
<p>The following data categories and variables are available for
output:</p>
<blockquote>
<div><ul>
<li><p class="first">category &#8220;configuration&#8221;: configuration and box size (for
periodic universes)</p>
</li>
<li><p class="first">category &#8220;gradients&#8221;: energy gradients for each atom</p>
</li>
<li><dl class="first docutils">
<dt>category &#8220;energy&#8221;: potential energy and</dt>
<dd><p class="first last">norm of the potential energy gradient</p>
</dd>
</dl>
</li>
</ul>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe on which the integrator acts</li>
<li><strong>steps</strong> (<em>int</em>) &#8211; the number of minimization steps (default is 100)</li>
<li><strong>step_size</strong> (<em>float</em>) &#8211; the initial size of a minimization step
(default is 0.002 nm)</li>
<li><strong>convergence</strong> (<em>float</em>) &#8211; the root-mean-square gradient length at which
minimization stops (default is 0.01 kJ/mol/nm)</li>
<li><strong>actions</strong> (<em>list</em>) &#8211; a list of actions to be executed periodically
(default is none)</li>
<li><strong>threads</strong> (<em>int</em>) &#8211; the number of threads to use in energy evaluation
(default set by MMTK_ENERGY_THREADS)</li>
<li><strong>background</strong> (<em>bool</em>) &#8211; if True, the integration is executed as a
separate thread (default: False)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Minimization.SteepestDescentMinimizer">
<em class="property">class </em><tt class="descclassname">MMTK.Minimization.</tt><tt class="descname">SteepestDescentMinimizer</tt><big>(</big><em>universe</em>, <em>**options</em><big>)</big><a class="reference internal" href="_modules/MMTK/Minimization.html#SteepestDescentMinimizer"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Minimization.SteepestDescentMinimizer" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.Minimization.Minimizer</span></tt></p>
<p>Steepest-descent minimizer</p>
<p>The minimizer can handle fixed atoms, but no distance constraints.
It is fully thread-safe.</p>
<p>The minimization is started by calling the minimizer object.
All the keyword options (see documnentation of __init__) can be
specified either when creating the minimizer or when calling it.</p>
<p>The following data categories and variables are available for
output:</p>
<blockquote>
<div><ul>
<li><p class="first">category &#8220;configuration&#8221;: configuration and box size (for
periodic universes)</p>
</li>
<li><p class="first">category &#8220;gradients&#8221;: energy gradients for each atom</p>
</li>
<li><dl class="first docutils">
<dt>category &#8220;energy&#8221;: potential energy and</dt>
<dd><p class="first last">norm of the potential energy gradient</p>
</dd>
</dl>
</li>
</ul>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe on which the integrator acts</li>
<li><strong>steps</strong> (<em>int</em>) &#8211; the number of minimization steps (default is 100)</li>
<li><strong>step_size</strong> (<em>float</em>) &#8211; the initial size of a minimization step
(default is 0.002 nm)</li>
<li><strong>convergence</strong> (<em>float</em>) &#8211; the root-mean-square gradient length at which
minimization stops (default is 0.01 kJ/mol/nm)</li>
<li><strong>actions</strong> (<em>list</em>) &#8211; a list of actions to be executed periodically
(default is none)</li>
<li><strong>threads</strong> (<em>int</em>) &#8211; the number of threads to use in energy evaluation
(default set by MMTK_ENERGY_THREADS)</li>
<li><strong>background</strong> (<em>bool</em>) &#8211; if True, the integration is executed as a
separate thread (default: False)</li>
<li><strong>mpi_communicator</strong> (<em>Scientific.MPI.MPICommunicator</em>) &#8211; an MPI communicator object, or None,
meaning no parallelization (default: None)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.MolecularSurface">
<span id="mmtk-molecularsurface"></span><h2>MMTK.MolecularSurface<a class="headerlink" href="#module-MMTK.MolecularSurface" title="Permalink to this headline">¶</a></h2>
<p>Molecular surfaces and volumes.</p>
<dl class="function">
<dt id="MMTK.MolecularSurface.findContacts">
<tt class="descclassname">MMTK.MolecularSurface.</tt><tt class="descname">findContacts</tt><big>(</big><em>object1</em>, <em>object2</em>, <em>contact_factor=1.0</em>, <em>cutoff=0.0</em><big>)</big><a class="reference internal" href="_modules/MMTK/MolecularSurface.html#findContacts"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.MolecularSurface.findContacts" title="Permalink to this definition">¶</a></dt>
<dd><p>Identifies contacts between two molecules. A contact is defined as a pair
of atoms whose distance is less than contact_factor*(r1+r2+cutoff),
where r1 and r2 are the atomic van-der-Waals radii.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>object1</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; a chemical object</li>
<li><strong>object2</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; a chemical object</li>
<li><strong>contact_factor</strong> (<em>float</em>) &#8211; a scale factor in the contact distance criterion</li>
<li><strong>cutoff</strong> (<em>float</em>) &#8211; a constant in the contact distance criterion</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">a list of Contact objects that describe atomic contacts
between object1 and object2.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">list</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.MolecularSurface.surfaceAndVolume">
<tt class="descclassname">MMTK.MolecularSurface.</tt><tt class="descname">surfaceAndVolume</tt><big>(</big><em>object</em>, <em>probe_radius=0.0</em><big>)</big><a class="reference internal" href="_modules/MMTK/MolecularSurface.html#surfaceAndVolume"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.MolecularSurface.surfaceAndVolume" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>object</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; a chemical object</li>
<li><strong>probe_radius</strong> (<em>float</em>) &#8211; the distance from the vdW-radii of the atoms
at which the surface is computed</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the molecular surface and volume of object</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">tuple</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.MolecularSurface.surfaceAtoms">
<tt class="descclassname">MMTK.MolecularSurface.</tt><tt class="descname">surfaceAtoms</tt><big>(</big><em>object</em>, <em>probe_radius=0.0</em><big>)</big><a class="reference internal" href="_modules/MMTK/MolecularSurface.html#surfaceAtoms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.MolecularSurface.surfaceAtoms" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>object</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; a chemical object</li>
<li><strong>probe_radius</strong> (<em>float</em>) &#8211; the distance from the vdW-radii of the atoms
at which the surface is computed</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">a dictionary that maps the surface atoms to their
exposed surface areas</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">dict</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.MolecularSurface.surfacePointsAndGradients">
<tt class="descclassname">MMTK.MolecularSurface.</tt><tt class="descname">surfacePointsAndGradients</tt><big>(</big><em>object</em>, <em>probe_radius=0.0</em>, <em>point_density=258</em><big>)</big><a class="reference internal" href="_modules/MMTK/MolecularSurface.html#surfacePointsAndGradients"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.MolecularSurface.surfacePointsAndGradients" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>object</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; a chemical object</li>
<li><strong>probe_radius</strong> (<em>float</em>) &#8211; the distance from the vdW-radii of the atoms
at which the surface is computed</li>
<li><strong>point_density</strong> (<em>int</em>) &#8211; the density of points that describe the surface</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">a dictionary that maps the surface atoms to a tuple
containing three surface-related quantities: the exposed surface
area, a list of points in the exposed surface, and a gradient
vector pointing outward from the surface.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">dict</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.MoleculeFactory">
<span id="mmtk-moleculefactory"></span><h2>MMTK.MoleculeFactory<a class="headerlink" href="#module-MMTK.MoleculeFactory" title="Permalink to this headline">¶</a></h2>
<p>Molecule factory for creating chemical objects</p>
<dl class="class">
<dt id="MMTK.MoleculeFactory.MoleculeFactory">
<em class="property">class </em><tt class="descclassname">MMTK.MoleculeFactory.</tt><tt class="descname">MoleculeFactory</tt><a class="reference internal" href="_modules/MMTK/MoleculeFactory.html#MoleculeFactory"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.MoleculeFactory.MoleculeFactory" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>MoleculeFactory</p>
<p>A MoleculeFactory serves to create molecules without reference to database
definitions. Molecules and groups are defined in terms of atoms, groups, and
bonds. Each MoleculeFactory constitutes an independent set of definitions.
Definitions within a MoleculeFactory can refer to each other.</p>
<p>Each MoleculeFactory stores a set of <a class="reference internal" href="#MMTK.ChemicalObjects.Group" title="MMTK.ChemicalObjects.Group"><tt class="xref py py-class docutils literal"><span class="pre">Group</span></tt></a>
objects which are referred to by names. The typical operation sequence is to
create a new group and then add atoms, bonds, and subgroups. It is also
possible to define coordinates and arbitrary attributes (in particular for
force fields). In the end, a finished object can be retrieved as a 
<a class="reference internal" href="#MMTK.ChemicalObjects.Molecule" title="MMTK.ChemicalObjects.Molecule"><tt class="xref py py-class docutils literal"><span class="pre">Molecule</span></tt></a> object.</p>
<dl class="method">
<dt id="MMTK.MoleculeFactory.MoleculeFactory.addAtom">
<tt class="descname">addAtom</tt><big>(</big><em>group</em>, <em>atom_name</em>, <em>element</em><big>)</big><a class="reference internal" href="_modules/MMTK/MoleculeFactory.html#MoleculeFactory.addAtom"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.MoleculeFactory.MoleculeFactory.addAtom" title="Permalink to this definition">¶</a></dt>
<dd><p>Add an atom to a group</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>group</strong> (<em>str</em>) &#8211; the name of the group</li>
<li><strong>atom_name</strong> (<em>str</em>) &#8211; the name of the atom</li>
<li><strong>element</strong> (<em>str</em>) &#8211; the chemical element symbol</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.MoleculeFactory.MoleculeFactory.addBond">
<tt class="descname">addBond</tt><big>(</big><em>group</em>, <em>atom1</em>, <em>atom2</em><big>)</big><a class="reference internal" href="_modules/MMTK/MoleculeFactory.html#MoleculeFactory.addBond"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.MoleculeFactory.MoleculeFactory.addBond" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a bond to a group</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>group</strong> (<em>str</em>) &#8211; the name of the group</li>
<li><strong>atom1</strong> (<em>str</em>) &#8211; the name of the first atom</li>
<li><strong>atom2</strong> (<em>str</em>) &#8211; the name of the second atom</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.MoleculeFactory.MoleculeFactory.addSubgroup">
<tt class="descname">addSubgroup</tt><big>(</big><em>group</em>, <em>subgroup_name</em>, <em>subgroup</em><big>)</big><a class="reference internal" href="_modules/MMTK/MoleculeFactory.html#MoleculeFactory.addSubgroup"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.MoleculeFactory.MoleculeFactory.addSubgroup" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a subgroup to a group</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>group</strong> (<em>str</em>) &#8211; the name of the group</li>
<li><strong>subgroup_name</strong> (<em>str</em>) &#8211; the name of the subgroup within the group</li>
<li><strong>subgroup</strong> (<em>str</em>) &#8211; the subgroup type</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.MoleculeFactory.MoleculeFactory.createGroup">
<tt class="descname">createGroup</tt><big>(</big><em>name</em><big>)</big><a class="reference internal" href="_modules/MMTK/MoleculeFactory.html#MoleculeFactory.createGroup"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.MoleculeFactory.MoleculeFactory.createGroup" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a new (initially empty) group object.</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.MoleculeFactory.MoleculeFactory.retrieveMolecule">
<tt class="descname">retrieveMolecule</tt><big>(</big><em>group</em><big>)</big><a class="reference internal" href="_modules/MMTK/MoleculeFactory.html#MoleculeFactory.retrieveMolecule"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.MoleculeFactory.MoleculeFactory.retrieveMolecule" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>group</strong> (<em>str</em>) &#8211; the name of the group to be used as a template</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a molecule defined by the contents of the group</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ChemicalObjects.Molecule" title="MMTK.ChemicalObjects.Molecule"><tt class="xref py py-class docutils literal"><span class="pre">Molecule</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-MMTK.NormalModes">
<span id="mmtk-normalmodes"></span><h2>MMTK.NormalModes<a class="headerlink" href="#module-MMTK.NormalModes" title="Permalink to this headline">¶</a></h2>
<p>Normal modes</p>
<div class="section" id="module-MMTK.NormalModes.BrownianModes">
<span id="mmtk-normalmodes-brownianmodes"></span><h3>MMTK.NormalModes.BrownianModes<a class="headerlink" href="#module-MMTK.NormalModes.BrownianModes" title="Permalink to this headline">¶</a></h3>
<p>Brownian normal modes</p>
<dl class="class">
<dt id="MMTK.NormalModes.BrownianModes.BrownianMode">
<em class="property">class </em><tt class="descclassname">MMTK.NormalModes.BrownianModes.</tt><tt class="descname">BrownianMode</tt><big>(</big><em>universe</em>, <em>n</em>, <em>inv_relaxation_time</em>, <em>mode</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/BrownianModes.html#BrownianMode"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.BrownianModes.BrownianMode" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.NormalModes.Core.Mode</span></tt></p>
<p>Single Brownian normal mode</p>
<p>Mode objects are created by indexing a <a class="reference internal" href="#MMTK.NormalModes.BrownianModes.BrownianModes" title="MMTK.NormalModes.BrownianModes.BrownianModes"><tt class="xref py py-class docutils literal"><span class="pre">BrownianModes</span></tt></a> object.
They contain the atomic displacements corresponding to a single
mode. In addition, the inverse of the relaxation time corresponding
to the mode is stored in the attribute &#8220;inv_relaxation_time&#8221;.</p>
<p>Note: the Brownian mode vectors are not friction weighted and therefore
not orthogonal to each other.</p>
</dd></dl>

<dl class="class">
<dt id="MMTK.NormalModes.BrownianModes.BrownianModes">
<em class="property">class </em><tt class="descclassname">MMTK.NormalModes.BrownianModes.</tt><tt class="descname">BrownianModes</tt><big>(</big><em>universe=None</em>, <em>friction=None</em>, <em>temperature=300.0</em>, <em>subspace=None</em>, <em>delta=None</em>, <em>sparse=False</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/BrownianModes.html#BrownianModes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.BrownianModes.BrownianModes" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.NormalModes.Core.NormalModes</span></tt></p>
<p>Brownian modes describe the independent relaxation motions of a
harmonic system with friction. They are obtained by diagonalizing the
friction-weighted force constant matrix.</p>
<p>In order to obtain physically reasonable normal modes, the configuration
of the universe must correspond to a local minimum of the potential
energy.</p>
<p>Individual modes (see <a class="reference internal" href="#MMTK.NormalModes.BrownianModes.BrownianMode" title="MMTK.NormalModes.BrownianModes.BrownianMode"><tt class="xref py py-class docutils literal"><span class="pre">BrownianMode</span></tt></a>)
can be extracted by indexing with an integer. Looping over the modes
is possible as well.</p>
<p>Brownian modes describe the independent relaxation motions of a
harmonic system with friction. They are obtained by diagonalizing the
friction-weighted force constant matrix.</p>
<p>In order to obtain physically reasonable normal modes, the configuration
of the universe must correspond to a local minimum of the potential
energy.</p>
<p>A BrownianModes object behaves like a sequence of modes.
Individual modes (see <tt class="xref py py-class docutils literal"><span class="pre">BrownianMode</span></tt>)
can be extracted by indexing with an integer. Looping over the modes
is possible as well.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the system for which the normal modes are calculated;
it must have a force field which provides the second
derivatives of the potential energy</li>
<li><strong>friction</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a>) &#8211; the friction coefficient for each particle.
Note: The friction coefficients are not mass-weighted,
i.e. they have the dimension of an inverse time.</li>
<li><strong>temperature</strong> (<em>float</em>) &#8211; the temperature for which the amplitudes of the
atomic displacement vectors are calculated. A
value of None can be specified to have no scaling
at all. In that case the mass-weighted norm
of each normal mode is one.</li>
<li><strong>subspace</strong> &#8211; the basis for the subspace in which the normal modes
are calculated (or, more precisely, a set of vectors
spanning the subspace; it does not have to be
orthogonal). This can either be a sequence of
<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a> objects
or a tuple of two such sequences. In the second case,
the subspace is defined by the space spanned by the
second set of vectors projected on the complement of
the space spanned by the first set of vectors.
The first set thus defines directions that are
excluded from the subspace.
The default value of None indicates a standard
normal mode calculation in the 3N-dimensional
configuration space.</li>
<li><strong>delta</strong> (<em>float</em>) &#8211; the rms step length for numerical differentiation.
The default value of None indicates analytical
differentiation.
Numerical differentiation is available only when a
subspace basis is used as well. Instead of calculating
the full force constant matrix and then multiplying
with the subspace basis, the subspace force constant
matrix is obtained by numerical differentiation of the
energy gradients along the basis vectors of the subspace.
If the basis is much smaller than the full configuration
space, this approach needs much less memory.</li>
<li><strong>sparse</strong> (<em>bool</em>) &#8211; a flag that indicates if a sparse representation of
the force constant matrix is to be used. This is of
interest when there are no long-range interactions and
a subspace of smaller size then 3N is specified. In that
case, the calculation will use much less memory with a
sparse representation.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.NormalModes.BrownianModes.BrownianModes.EISF">
<tt class="descname">EISF</tt><big>(</big><em>q_range=(0.0</em>, <em>15.0)</em>, <em>subset=None</em>, <em>weights=None</em>, <em>random_vectors=15</em>, <em>first_mode=6</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/BrownianModes.html#BrownianModes.EISF"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.BrownianModes.BrownianModes.EISF" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>q_range</strong> (<em>tuple</em>) &#8211; the range of angular wavenumber values</li>
<li><strong>subset</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; the subset of the universe used in the calculation
(default: the whole universe)</li>
<li><strong>weights</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a>) &#8211; the weight to be given to each atom in the average
(default: incoherent scattering lengths)</li>
<li><strong>random_vectors</strong> (<em>int</em>) &#8211; the number of random direction vectors
used in the orientational average</li>
<li><strong>first_mode</strong> (<em>int</em>) &#8211; the first mode to be taken into account for
the fluctuation calculation. The default value
of 6 is right for molecules in vacuum.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the Elastic Incoherent Structure Factor (EISF) as a
function of angular wavenumber</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">Scientific.Functions.Interpolation.InterpolatingFunction</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.NormalModes.BrownianModes.BrownianModes.coherentScatteringFunction">
<tt class="descname">coherentScatteringFunction</tt><big>(</big><em>q</em>, <em>time_range=(0.0</em>, <em>None</em>, <em>None)</em>, <em>subset=None</em>, <em>weights=None</em>, <em>random_vectors=15</em>, <em>first_mode=6</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/BrownianModes.html#BrownianModes.coherentScatteringFunction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.BrownianModes.BrownianModes.coherentScatteringFunction" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>q</strong> (<em>float</em>) &#8211; the angular wavenumber</li>
<li><strong>time_range</strong> (<em>tuple</em>) &#8211; the time values at which the mean-square
displacement is evaluated, specified as a
range tuple (first, last, step).
The defaults are first=0, last=
20 times the longest vibration perdiod,
and step defined such that 300 points are
used in total.</li>
<li><strong>subset</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; the subset of the universe used in the calculation
(default: the whole universe)</li>
<li><strong>weights</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a>) &#8211; the weight to be given to each atom in the average
(default: coherent scattering lengths)</li>
<li><strong>random_vectors</strong> (<em>int</em>) &#8211; the number of random direction vectors
used in the orientational average</li>
<li><strong>first_mode</strong> (<em>int</em>) &#8211; the first mode to be taken into account for
the fluctuation calculation. The default value
of 6 is right for molecules in vacuum.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the Coherent Scattering Function as a function of time</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">Scientific.Functions.Interpolation.InterpolatingFunction</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.NormalModes.BrownianModes.BrownianModes.incoherentScatteringFunction">
<tt class="descname">incoherentScatteringFunction</tt><big>(</big><em>q</em>, <em>time_range=(0.0</em>, <em>None</em>, <em>None)</em>, <em>subset=None</em>, <em>random_vectors=15</em>, <em>first_mode=6</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/BrownianModes.html#BrownianModes.incoherentScatteringFunction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.BrownianModes.BrownianModes.incoherentScatteringFunction" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>q</strong> (<em>float</em>) &#8211; the angular wavenumber</li>
<li><strong>time_range</strong> (<em>tuple</em>) &#8211; the time values at which the mean-square
displacement is evaluated, specified as a
range tuple (first, last, step).
The defaults are first=0, last=
20 times the longest vibration perdiod,
and step defined such that 300 points are
used in total.</li>
<li><strong>subset</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; the subset of the universe used in the calculation
(default: the whole universe)</li>
<li><strong>random_vectors</strong> (<em>int</em>) &#8211; the number of random direction vectors
used in the orientational average</li>
<li><strong>first_mode</strong> (<em>int</em>) &#8211; the first mode to be taken into account for
the fluctuation calculation. The default value
of 6 is right for molecules in vacuum.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the Incoherent Scattering Function as a function of time</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">Scientific.Functions.Interpolation.InterpolatingFunction</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.NormalModes.BrownianModes.BrownianModes.meanSquareDisplacement">
<tt class="descname">meanSquareDisplacement</tt><big>(</big><em>subset=None</em>, <em>weights=None</em>, <em>time_range=(0.0</em>, <em>None</em>, <em>None)</em>, <em>first_mode=6</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/BrownianModes.html#BrownianModes.meanSquareDisplacement"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.BrownianModes.BrownianModes.meanSquareDisplacement" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>subset</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; the subset of the universe used in the calculation
(default: the whole universe)</li>
<li><strong>weights</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a>) &#8211; the weight to be given to each atom in the average
(default: atomic masses)</li>
<li><strong>time_range</strong> (<em>tuple</em>) &#8211; the time values at which the mean-square
displacement is evaluated, specified as a
range tuple (first, last, step).
The defaults are first=0, last=
20 times the longest vibration perdiod,
and step defined such that 300 points are
used in total.</li>
<li><strong>first_mode</strong> (<em>int</em>) &#8211; the first mode to be taken into account for
the fluctuation calculation. The default value
of 6 is right for molecules in vacuum.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the averaged mean-square displacement of the
atoms in subset as a function of time</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">Scientific.Functions.Interpolation.InterpolatingFunction</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.NormalModes.BrownianModes.BrownianModes.rawMode">
<tt class="descname">rawMode</tt><big>(</big><em>item</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/BrownianModes.html#BrownianModes.rawMode"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.BrownianModes.BrownianModes.rawMode" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>item</strong> (<em>int</em>) &#8211; the index of a normal mode</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the unscaled mode vector</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.NormalModes.BrownianModes.BrownianMode" title="MMTK.NormalModes.BrownianModes.BrownianMode"><tt class="xref py py-class docutils literal"><span class="pre">BrownianMode</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.NormalModes.BrownianModes.BrownianModes.staticStructureFactor">
<tt class="descname">staticStructureFactor</tt><big>(</big><em>q_range=(1.0</em>, <em>15.0)</em>, <em>subset=None</em>, <em>weights=None</em>, <em>random_vectors=15</em>, <em>first_mode=6</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/BrownianModes.html#BrownianModes.staticStructureFactor"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.BrownianModes.BrownianModes.staticStructureFactor" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>q_range</strong> (<em>tuple</em>) &#8211; the range of angular wavenumber values</li>
<li><strong>subset</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; the subset of the universe used in the calculation
(default: the whole universe)</li>
<li><strong>weights</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a>) &#8211; the weight to be given to each atom in the average
(default: coherent scattering lengths)</li>
<li><strong>random_vectors</strong> (<em>int</em>) &#8211; the number of random direction vectors
used in the orientational average</li>
<li><strong>first_mode</strong> (<em>int</em>) &#8211; the first mode to be taken into account for
the fluctuation calculation. The default value
of 6 is right for molecules in vacuum.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the Static Structure Factor as a
function of angular wavenumber</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">Scientific.Functions.Interpolation.InterpolatingFunction</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-MMTK.NormalModes.EnergeticModes">
<span id="mmtk-normalmodes-energeticmodes"></span><h3>MMTK.NormalModes.EnergeticModes<a class="headerlink" href="#module-MMTK.NormalModes.EnergeticModes" title="Permalink to this headline">¶</a></h3>
<p>Energetic normal modes</p>
<dl class="class">
<dt id="MMTK.NormalModes.EnergeticModes.EnergeticMode">
<em class="property">class </em><tt class="descclassname">MMTK.NormalModes.EnergeticModes.</tt><tt class="descname">EnergeticMode</tt><big>(</big><em>universe</em>, <em>n</em>, <em>force_constant</em>, <em>mode</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/EnergeticModes.html#EnergeticMode"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.EnergeticModes.EnergeticMode" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.NormalModes.Core.Mode</span></tt></p>
<p>Single energetic normal mode</p>
<p>Mode objects are created by indexing a <a class="reference internal" href="#MMTK.NormalModes.EnergeticModes.EnergeticModes" title="MMTK.NormalModes.EnergeticModes.EnergeticModes"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.NormalModes.EnergeticModes.EnergeticModes</span></tt></a> object.
They contain the atomic displacements corresponding to a
single mode. In addition, the force constant corresponding to the mode
is stored in the attribute &#8220;force_constant&#8221;.</p>
</dd></dl>

<dl class="class">
<dt id="MMTK.NormalModes.EnergeticModes.EnergeticModes">
<em class="property">class </em><tt class="descclassname">MMTK.NormalModes.EnergeticModes.</tt><tt class="descname">EnergeticModes</tt><big>(</big><em>universe=None</em>, <em>temperature=300</em>, <em>subspace=None</em>, <em>delta=None</em>, <em>sparse=False</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/EnergeticModes.html#EnergeticModes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.EnergeticModes.EnergeticModes" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.NormalModes.Core.NormalModes</span></tt></p>
<p>Energetic modes describe the principal axes of an harmonic approximation
to the potential energy surface of a system. They are obtained by
diagonalizing the force constant matrix without prior mass-weighting.</p>
<p>In order to obtain physically reasonable normal modes, the configuration
of the universe must correspond to a local minimum of the potential
energy.</p>
<p>Individual modes (see class <a class="reference internal" href="#MMTK.NormalModes.EnergeticModes.EnergeticMode" title="MMTK.NormalModes.EnergeticModes.EnergeticMode"><tt class="xref py py-class docutils literal"><span class="pre">EnergeticMode</span></tt></a>)
can be extracted by indexing with an integer. Looping over the modes
is possible as well.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the system for which the normal modes are calculated;
it must have a force field which provides the second
derivatives of the potential energy</li>
<li><strong>temperature</strong> (<em>float</em>) &#8211; the temperature for which the amplitudes of the
atomic displacement vectors are calculated. A
value of None can be specified to have no scaling
at all. In that case the mass-weighted norm
of each normal mode is one.</li>
<li><strong>subspace</strong> &#8211; the basis for the subspace in which the normal modes
are calculated (or, more precisely, a set of vectors
spanning the subspace; it does not have to be
orthogonal). This can either be a sequence of
<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a> objects
or a tuple of two such sequences. In the second case,
the subspace is defined by the space spanned by the
second set of vectors projected on the complement of
the space spanned by the first set of vectors.
The first set thus defines directions that are
excluded from the subspace.
The default value of None indicates a standard
normal mode calculation in the 3N-dimensional
configuration space.</li>
<li><strong>delta</strong> (<em>float</em>) &#8211; the rms step length for numerical differentiation.
The default value of None indicates analytical
differentiation.
Numerical differentiation is available only when a
subspace basis is used as well. Instead of calculating
the full force constant matrix and then multiplying
with the subspace basis, the subspace force constant
matrix is obtained by numerical differentiation of the
energy gradients along the basis vectors of the subspace.
If the basis is much smaller than the full configuration
space, this approach needs much less memory.</li>
<li><strong>sparse</strong> (<em>bool</em>) &#8211; a flag that indicates if a sparse representation of
the force constant matrix is to be used. This is of
interest when there are no long-range interactions and
a subspace of smaller size then 3N is specified. In that
case, the calculation will use much less memory with a
sparse representation.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.NormalModes.EnergeticModes.EnergeticModes.rawMode">
<tt class="descname">rawMode</tt><big>(</big><em>item</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/EnergeticModes.html#EnergeticModes.rawMode"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.EnergeticModes.EnergeticModes.rawMode" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>item</strong> (<em>int</em>) &#8211; the index of a normal mode</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the unscaled mode vector</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.NormalModes.EnergeticModes.EnergeticMode" title="MMTK.NormalModes.EnergeticModes.EnergeticMode"><tt class="xref py py-class docutils literal"><span class="pre">EnergeticMode</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-MMTK.NormalModes.VibrationalModes">
<span id="mmtk-normalmodes-vibrationalmodes"></span><h3>MMTK.NormalModes.VibrationalModes<a class="headerlink" href="#module-MMTK.NormalModes.VibrationalModes" title="Permalink to this headline">¶</a></h3>
<p>Vibrational normal modes</p>
<dl class="class">
<dt id="MMTK.NormalModes.VibrationalModes.VibrationalMode">
<em class="property">class </em><tt class="descclassname">MMTK.NormalModes.VibrationalModes.</tt><tt class="descname">VibrationalMode</tt><big>(</big><em>universe</em>, <em>n</em>, <em>frequency</em>, <em>mode</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/VibrationalModes.html#VibrationalMode"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.VibrationalModes.VibrationalMode" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.NormalModes.Core.Mode</span></tt></p>
<p>Single vibrational normal mode</p>
<p>Mode objects are created by indexing a 
<a class="reference internal" href="#MMTK.NormalModes.VibrationalModes.VibrationalModes" title="MMTK.NormalModes.VibrationalModes.VibrationalModes"><tt class="xref py py-class docutils literal"><span class="pre">VibrationalModes</span></tt></a> object.
They contain the atomic displacements corresponding to a
single mode. In addition, the frequency corresponding to the mode
is stored in the attribute &#8220;frequency&#8221;.</p>
<p>Note: the normal mode vectors are <em>not</em> mass weighted, and therefore
not orthogonal to each other.</p>
</dd></dl>

<dl class="class">
<dt id="MMTK.NormalModes.VibrationalModes.VibrationalModes">
<em class="property">class </em><tt class="descclassname">MMTK.NormalModes.VibrationalModes.</tt><tt class="descname">VibrationalModes</tt><big>(</big><em>universe=None</em>, <em>temperature=300.0</em>, <em>subspace=None</em>, <em>delta=None</em>, <em>sparse=False</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/VibrationalModes.html#VibrationalModes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.VibrationalModes.VibrationalModes" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.NormalModes.Core.NormalModes</span></tt></p>
<p>Vibrational modes describe the independent vibrational motions
of a harmonic system. They are obtained by diagonalizing the mass-weighted
force constant matrix.</p>
<p>In order to obtain physically reasonable normal modes, the configuration
of the universe must correspond to a local minimum of the potential
energy.</p>
<p>Individual modes (see <a class="reference internal" href="#MMTK.NormalModes.VibrationalModes.VibrationalMode" title="MMTK.NormalModes.VibrationalModes.VibrationalMode"><tt class="xref py py-class docutils literal"><span class="pre">VibrationalMode</span></tt></a>)
can be extracted by indexing with an integer. Looping over the modes
is possible as well.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the system for which the normal modes are calculated;
it must have a force field which provides the second
derivatives of the potential energy</li>
<li><strong>temperature</strong> (<em>float</em>) &#8211; the temperature for which the amplitudes of the
atomic displacement vectors are calculated. A
value of None can be specified to have no scaling
at all. In that case the mass-weighted norm
of each normal mode is one.</li>
<li><strong>subspace</strong> &#8211; the basis for the subspace in which the normal modes
are calculated (or, more precisely, a set of vectors
spanning the subspace; it does not have to be
orthogonal). This can either be a sequence of
<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a> objects
or a tuple of two such sequences. In the second case,
the subspace is defined by the space spanned by the
second set of vectors projected on the complement of
the space spanned by the first set of vectors.
The first set thus defines directions that are
excluded from the subspace.
The default value of None indicates a standard
normal mode calculation in the 3N-dimensional
configuration space.</li>
<li><strong>delta</strong> (<em>float</em>) &#8211; the rms step length for numerical differentiation.
The default value of None indicates analytical
differentiation.
Numerical differentiation is available only when a
subspace basis is used as well. Instead of calculating
the full force constant matrix and then multiplying
with the subspace basis, the subspace force constant
matrix is obtained by numerical differentiation of the
energy gradients along the basis vectors of the subspace.
If the basis is much smaller than the full configuration
space, this approach needs much less memory.</li>
<li><strong>sparse</strong> (<em>bool</em>) &#8211; a flag that indicates if a sparse representation of
the force constant matrix is to be used. This is of
interest when there are no long-range interactions and
a subspace of smaller size then 3N is specified. In that
case, the calculation will use much less memory with a
sparse representation.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.NormalModes.VibrationalModes.VibrationalModes.EISF">
<tt class="descname">EISF</tt><big>(</big><em>q_range=(0.0</em>, <em>15.0)</em>, <em>subset=None</em>, <em>weights=None</em>, <em>random_vectors=15</em>, <em>first_mode=6</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/VibrationalModes.html#VibrationalModes.EISF"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.VibrationalModes.VibrationalModes.EISF" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>q_range</strong> (<em>tuple</em>) &#8211; the range of angular wavenumber values</li>
<li><strong>subset</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; the subset of the universe used in the calculation
(default: the whole universe)</li>
<li><strong>weights</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a>) &#8211; the weight to be given to each atom in the average
(default: incoherent scattering lengths)</li>
<li><strong>random_vectors</strong> (<em>int</em>) &#8211; the number of random direction vectors
used in the orientational average</li>
<li><strong>first_mode</strong> (<em>int</em>) &#8211; the first mode to be taken into account for
the fluctuation calculation. The default value
of 6 is right for molecules in vacuum.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the Elastic Incoherent Structure Factor (EISF) as a
function of angular wavenumber</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">Scientific.Functions.Interpolation.InterpolatingFunction</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.NormalModes.VibrationalModes.VibrationalModes.incoherentScatteringFunction">
<tt class="descname">incoherentScatteringFunction</tt><big>(</big><em>q</em>, <em>time_range=(0.0</em>, <em>None</em>, <em>None)</em>, <em>subset=None</em>, <em>weights=None</em>, <em>tau=None</em>, <em>random_vectors=15</em>, <em>first_mode=6</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/VibrationalModes.html#VibrationalModes.incoherentScatteringFunction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.VibrationalModes.VibrationalModes.incoherentScatteringFunction" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>q</strong> (<em>float</em>) &#8211; the angular wavenumber</li>
<li><strong>time_range</strong> (<em>tuple</em>) &#8211; the time values at which the mean-square
displacement is evaluated, specified as a
range tuple (first, last, step).
The defaults are first=0, last=
20 times the longest vibration perdiod,
and step defined such that 300 points are
used in total.</li>
<li><strong>subset</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; the subset of the universe used in the calculation
(default: the whole universe)</li>
<li><strong>weights</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a>) &#8211; the weight to be given to each atom in the average
(default: incoherent scattering lengths)</li>
<li><strong>tau</strong> (<em>float</em>) &#8211; the relaxation time of an exponential damping factor
(default: no damping)</li>
<li><strong>random_vectors</strong> (<em>int</em>) &#8211; the number of random direction vectors
used in the orientational average</li>
<li><strong>first_mode</strong> (<em>int</em>) &#8211; the first mode to be taken into account for
the fluctuation calculation. The default value
of 6 is right for molecules in vacuum.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the Incoherent Scattering Function as a function of time</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">Scientific.Functions.Interpolation.InterpolatingFunction</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.NormalModes.VibrationalModes.VibrationalModes.meanSquareDisplacement">
<tt class="descname">meanSquareDisplacement</tt><big>(</big><em>subset=None</em>, <em>weights=None</em>, <em>time_range=(0.0</em>, <em>None</em>, <em>None)</em>, <em>tau=None</em>, <em>first_mode=6</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/VibrationalModes.html#VibrationalModes.meanSquareDisplacement"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.VibrationalModes.VibrationalModes.meanSquareDisplacement" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>subset</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; the subset of the universe used in the calculation
(default: the whole universe)</li>
<li><strong>weights</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a>) &#8211; the weight to be given to each atom in the average
(default: atomic masses)</li>
<li><strong>time_range</strong> (<em>tuple</em>) &#8211; the time values at which the mean-square
displacement is evaluated, specified as a
range tuple (first, last, step).
The defaults are first=0, last=
20 times the longest vibration perdiod,
and step defined such that 300 points are
used in total.</li>
<li><strong>tau</strong> (<em>float</em>) &#8211; the relaxation time of an exponential damping factor
(default: no damping)</li>
<li><strong>first_mode</strong> (<em>int</em>) &#8211; the first mode to be taken into account for
the fluctuation calculation. The default value
of 6 is right for molecules in vacuum.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the averaged mean-square displacement of the
atoms in subset as a function of time</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">Scientific.Functions.Interpolation.InterpolatingFunction</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.NormalModes.VibrationalModes.VibrationalModes.rawMode">
<tt class="descname">rawMode</tt><big>(</big><em>item</em><big>)</big><a class="reference internal" href="_modules/MMTK/NormalModes/VibrationalModes.html#VibrationalModes.rawMode"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NormalModes.VibrationalModes.VibrationalModes.rawMode" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>item</strong> (<em>int</em>) &#8211; the index of a normal mode</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the unscaled mode vector</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.NormalModes.VibrationalModes.VibrationalMode" title="MMTK.NormalModes.VibrationalModes.VibrationalMode"><tt class="xref py py-class docutils literal"><span class="pre">VibrationalMode</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
</div>
<div class="section" id="module-MMTK.NucleicAcids">
<span id="mmtk-nucleicacids"></span><h2>MMTK.NucleicAcids<a class="headerlink" href="#module-MMTK.NucleicAcids" title="Permalink to this headline">¶</a></h2>
<p>Nucleic acid chains</p>
<dl class="class">
<dt id="MMTK.NucleicAcids.Nucleotide">
<em class="property">class </em><tt class="descclassname">MMTK.NucleicAcids.</tt><tt class="descname">Nucleotide</tt><big>(</big><em>name=None</em>, <em>model='all'</em><big>)</big><a class="reference internal" href="_modules/MMTK/NucleicAcids.html#Nucleotide"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NucleicAcids.Nucleotide" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Biopolymers.Residue" title="MMTK.Biopolymers.Residue"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Biopolymers.Residue</span></tt></a></p>
<p>Nucleic acid residue</p>
<p>Nucleotides are a special kind of group. Like any other
group, they are defined in the chemical database. Each residue
has two or three subgroups (&#8216;sugar&#8217; and &#8216;base&#8217;, plus &#8216;phosphate&#8217;
except for 5&#8217;-terminal residues) and is usually
connected to other residues to form a nucleotide chain. The database
contains three variants of each residue (5&#8217;-terminal, 3&#8217;-terminal,
non-terminal).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>name</strong> (<em>str</em>) &#8211; the name of the nucleotide in the chemical database. This
is the full name of the residue plus the suffix
&#8220;_5ter&#8221; or &#8220;_3ter&#8221; for the terminal variants.</li>
<li><strong>model</strong> (<em>str</em>) &#8211; one of &#8220;all&#8221; (all-atom), &#8220;none&#8221; (no hydrogens),
&#8220;polar&#8221; (united-atom with only polar hydrogens),
&#8220;polar_charmm&#8221; (like &#8220;polar&#8221;, but defining
polar hydrogens like in the CHARMM force field).
Currently the database has definitions only for &#8220;all&#8221;.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.NucleicAcids.Nucleotide.backbone">
<tt class="descname">backbone</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/NucleicAcids.html#Nucleotide.backbone"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NucleicAcids.Nucleotide.backbone" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the sugar and phosphate groups</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ChemicalObjects.Group" title="MMTK.ChemicalObjects.Group"><tt class="xref py py-class docutils literal"><span class="pre">Group</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.NucleicAcids.Nucleotide.bases">
<tt class="descname">bases</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/NucleicAcids.html#Nucleotide.bases"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NucleicAcids.Nucleotide.bases" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the base group</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ChemicalObjects.Group" title="MMTK.ChemicalObjects.Group"><tt class="xref py py-class docutils literal"><span class="pre">Group</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.NucleicAcids.NucleotideChain">
<em class="property">class </em><tt class="descclassname">MMTK.NucleicAcids.</tt><tt class="descname">NucleotideChain</tt><big>(</big><em>sequence</em>, <em>**properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/NucleicAcids.html#NucleotideChain"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NucleicAcids.NucleotideChain" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Biopolymers.ResidueChain" title="MMTK.Biopolymers.ResidueChain"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Biopolymers.ResidueChain</span></tt></a></p>
<p>Nucleotide chain</p>
<p>Nucleotide chains consist of nucleotides that are linked together.
They are a special kind of molecule, i.e. all molecule operations
are available.</p>
<p>Nucleotide chains act as sequences of residues. If n is a NucleotideChain
object, then</p>
<blockquote>
<div><ul>
<li><p class="first">len(n) yields the number of nucleotides</p>
</li>
<li><p class="first">n[i] yields nucleotide number i</p>
</li>
<li><dl class="first docutils">
<dt>n[i:j] yields the subchain from nucleotide number i up to but</dt>
<dd><p class="first last">excluding nucleotide number j</p>
</dd>
</dl>
</li>
</ul>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>sequence</strong> &#8211; the nucleotide sequence. This can be a string
containing the one-letter codes, or a list
of two-letter codes (a &#8220;d&#8221; or &#8220;r&#8221; for the type of
sugar, and the one-letter base code), or a
<a class="reference internal" href="#MMTK.PDB.PDBNucleotideChain" title="MMTK.PDB.PDBNucleotideChain"><tt class="xref py py-class docutils literal"><span class="pre">PDBNucleotideChain</span></tt></a> object.
If a PDBNucleotideChain object is supplied, the
atomic positions it contains are assigned to the
atoms of the newly generated nucleotide chain,
otherwise the positions of all atoms are undefined.</li>
<li><strong>model</strong> (<em>str</em>) &#8211; one of &#8220;all&#8221; (all-atom), &#8220;no_hydrogens&#8221; or &#8220;none&#8221;
(no hydrogens), &#8220;polar_hydrogens&#8221; or &#8220;polar&#8221;
(united-atom with only polar hydrogens),
&#8220;polar_charmm&#8221; (like &#8220;polar&#8221;, but defining
polar hydrogens like in the CHARMM force field),
&#8220;polar_opls&#8221; (like &#8220;polar&#8221;, but defining
polar hydrogens like in the latest OPLS force field).
Default is &#8220;all&#8221;. Currently the database contains
definitions only for &#8220;all&#8221;.</li>
<li><strong>terminus_5</strong> (<em>bool</em>) &#8211; if True, the first residue is constructed
using the 5&#8217;-terminal variant, if False the
non-terminal version is used. Default is True.</li>
<li><strong>terminus_3</strong> (<em>bool</em>) &#8211; if True, the last residue is constructed
using the 3&#8217;-terminal variant, if False the
non-terminal version is used. Default is True.</li>
<li><strong>circular</strong> (<em>bool</em>) &#8211; if True, a bond is constructed
between the first and the last residue.
Default is False.</li>
<li><strong>name</strong> (<em>str</em>) &#8211; a name for the chain (a string)</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.NucleicAcids.NucleotideChain.backbone">
<tt class="descname">backbone</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/NucleicAcids.html#NucleotideChain.backbone"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NucleicAcids.NucleotideChain.backbone" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the sugar and phosphate groups of all nucleotides</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Collections.Collection" title="MMTK.Collections.Collection"><tt class="xref py py-class docutils literal"><span class="pre">Collection</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.NucleicAcids.NucleotideChain.bases">
<tt class="descname">bases</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/NucleicAcids.html#NucleotideChain.bases"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NucleicAcids.NucleotideChain.bases" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the base groups of all nucleotides</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Collections.Collection" title="MMTK.Collections.Collection"><tt class="xref py py-class docutils literal"><span class="pre">Collection</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.NucleicAcids.NucleotideSubChain">
<em class="property">class </em><tt class="descclassname">MMTK.NucleicAcids.</tt><tt class="descname">NucleotideSubChain</tt><big>(</big><em>chain</em>, <em>groups</em>, <em>name=''</em><big>)</big><a class="reference internal" href="_modules/MMTK/NucleicAcids.html#NucleotideSubChain"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NucleicAcids.NucleotideSubChain" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.NucleicAcids.NucleotideChain" title="MMTK.NucleicAcids.NucleotideChain"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.NucleicAcids.NucleotideChain</span></tt></a></p>
<p>A contiguous part of a nucleotide chain</p>
<p>NucleotideSubChain objects are the result of slicing operations on
NucleotideChain objects. They cannot be created directly.
NucleotideSubChain objects permit all operations of NucleotideChain
objects, but cannot be added to a universe.</p>
</dd></dl>

<dl class="function">
<dt id="MMTK.NucleicAcids.isNucleotideChain">
<tt class="descclassname">MMTK.NucleicAcids.</tt><tt class="descname">isNucleotideChain</tt><big>(</big><em>x</em><big>)</big><a class="reference internal" href="_modules/MMTK/NucleicAcids.html#isNucleotideChain"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.NucleicAcids.isNucleotideChain" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns True if x is a NucleotideChain.</p>
</dd></dl>

</div>
<div class="section" id="module-MMTK.PDB">
<span id="mmtk-pdb"></span><h2>MMTK.PDB<a class="headerlink" href="#module-MMTK.PDB" title="Permalink to this headline">¶</a></h2>
<p>PDB files</p>
<p>This module provides classes that represent molecules in a PDB file.
They permit various manipulations and the creation of MMTK objects.
Note that the classes defined in this module are specializations
of classed defined in Scientific.IO.PDB; the methods defined in
that module are also available.</p>
<dl class="class">
<dt id="MMTK.PDB.PDBConfiguration">
<em class="property">class </em><tt class="descclassname">MMTK.PDB.</tt><tt class="descname">PDBConfiguration</tt><big>(</big><em>file_or_filename</em>, <em>model=0</em>, <em>alternate_code='A'</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBConfiguration"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBConfiguration" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">Scientific.IO.PDB.Structure</span></tt></p>
<p>Everything in a PDB file</p>
<p>A PDBConfiguration object represents the full contents of a PDB
file. It can be used to create MMTK objects for all or part
of the molecules, or to change the configuration of an existing
system.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>file_or_filename</strong> &#8211; the name of a PDB file, or a file object</li>
<li><strong>model</strong> (<em>int</em>) &#8211; the number of the model to be used from a
multiple model file</li>
<li><strong>alternate_code</strong> (<em>str</em>) &#8211; the alternate code to be used for atoms that
have multiple positions</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.PDB.PDBConfiguration.applyTo">
<tt class="descname">applyTo</tt><big>(</big><em>object</em>, <em>atom_map=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBConfiguration.applyTo"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBConfiguration.applyTo" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets the configuration of object from the coordinates in the
PDB file. The object must be compatible with the PDB file, i.e.
contain the same subobjects and in the same order. This is usually
only guaranteed if the object was created by the method</p>
<p><a class="reference internal" href="#MMTK.PDB.PDBConfiguration.createAll" title="MMTK.PDB.PDBConfiguration.createAll"><tt class="xref py py-func docutils literal"><span class="pre">createAll()</span></tt></a> from a PDB file with the same layout.
:param object: a chemical object or collection of chemical objects</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.PDB.PDBConfiguration.asuToUnitCell">
<tt class="descname">asuToUnitCell</tt><big>(</big><em>asu_contents</em>, <em>compact=True</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBConfiguration.asuToUnitCell"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBConfiguration.asuToUnitCell" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>asu_contents</strong> &#8211; the molecules in the asymmetric unit, usually
obtained from <a class="reference internal" href="#MMTK.PDB.PDBConfiguration.createAll" title="MMTK.PDB.PDBConfiguration.createAll"><tt class="xref py py-func docutils literal"><span class="pre">createAll()</span></tt></a>.</li>
<li><strong>compact</strong> (<em>bool</em>) &#8211; if True, all molecules images are shifted such that
their centers of mass lie inside the unit cell.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">a collection containing all molecules in the unit cell,
obtained by copying and moving the molecules from the
asymmetric unit according to the crystallographic
symmetry operations.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last"><a class="reference internal" href="#MMTK.Collections.Collection" title="MMTK.Collections.Collection"><tt class="xref py py-class docutils literal"><span class="pre">Collection</span></tt></a></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.PDB.PDBConfiguration.createAll">
<tt class="descname">createAll</tt><big>(</big><em>molecule_names=None</em>, <em>permit_undefined=True</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBConfiguration.createAll"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBConfiguration.createAll" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a collection containing all objects contained in the
PDB file, i.e. the combination of the objects
returned by <a class="reference internal" href="#MMTK.PDB.PDBConfiguration.createPeptideChains" title="MMTK.PDB.PDBConfiguration.createPeptideChains"><tt class="xref py py-func docutils literal"><span class="pre">createPeptideChains()</span></tt></a>,
<a class="reference internal" href="#MMTK.PDB.PDBConfiguration.createNucleotideChains" title="MMTK.PDB.PDBConfiguration.createNucleotideChains"><tt class="xref py py-func docutils literal"><span class="pre">createNucleotideChains()</span></tt></a>,
and <a class="reference internal" href="#MMTK.PDB.PDBConfiguration.createMolecules" title="MMTK.PDB.PDBConfiguration.createMolecules"><tt class="xref py py-func docutils literal"><span class="pre">createMolecules()</span></tt></a>.
The parameters have the same meaning as for
<a class="reference internal" href="#MMTK.PDB.PDBConfiguration.createMolecules" title="MMTK.PDB.PDBConfiguration.createMolecules"><tt class="xref py py-func docutils literal"><span class="pre">createMolecules()</span></tt></a>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><tt class="xref py py-class docutils literal"><span class="pre">Collection</span></tt></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.PDB.PDBConfiguration.createMolecules">
<tt class="descname">createMolecules</tt><big>(</big><em>names=None</em>, <em>permit_undefined=True</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBConfiguration.createMolecules"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBConfiguration.createMolecules" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>names</strong> (<em>list</em>) &#8211; If a list of molecule names (as defined in the
chemical database) and/or PDB residue names,
only molecules mentioned in this list will be
constructed. If a dictionary, it is used to map
PDB residue names to molecule names. With the
default (None), only water molecules are
built.</li>
<li><strong>permit_undefined</strong> &#8211; If False, an exception is raised
when a PDB residue is encountered for
which no molecule name is supplied
in names. If True, an AtomCluster
object is constructed for each unknown
molecule.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">a collection of <a class="reference internal" href="#MMTK.ChemicalObjects.Molecule" title="MMTK.ChemicalObjects.Molecule"><tt class="xref py py-class docutils literal"><span class="pre">Molecule</span></tt></a> objects,
one for each molecule in the PDB file. Each PDB residue not 
describing an amino acid or nucleotide residue is considered a
molecule.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last"><a class="reference internal" href="#MMTK.Collections.Collection" title="MMTK.Collections.Collection"><tt class="xref py py-class docutils literal"><span class="pre">Collection</span></tt></a></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.PDB.PDBConfiguration.createNucleotideChains">
<tt class="descname">createNucleotideChains</tt><big>(</big><em>model='all'</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBConfiguration.createNucleotideChains"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBConfiguration.createNucleotideChains" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a list of <a class="reference internal" href="#MMTK.NucleicAcids.NucleotideChain" title="MMTK.NucleicAcids.NucleotideChain"><tt class="xref py py-class docutils literal"><span class="pre">NucleotideChain</span></tt></a> objects, one for each
nucleotide chain in the PDB file. The parameter model
has the same meaning as for the NucleotideChain constructor.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.PDB.PDBConfiguration.createPeptideChains">
<tt class="descname">createPeptideChains</tt><big>(</big><em>model='all'</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBConfiguration.createPeptideChains"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBConfiguration.createPeptideChains" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a list of <a class="reference internal" href="#MMTK.Proteins.PeptideChain" title="MMTK.Proteins.PeptideChain"><tt class="xref py py-class docutils literal"><span class="pre">PeptideChain</span></tt></a> objects, one for each
peptide chain in the PDB file. The parameter model
has the same meaning as for the PeptideChain constructor.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.PDB.PDBConfiguration.createUnitCellUniverse">
<tt class="descname">createUnitCellUniverse</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBConfiguration.createUnitCellUniverse"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBConfiguration.createUnitCellUniverse" title="Permalink to this definition">¶</a></dt>
<dd><p>Constructs an empty universe (OrthrhombicPeriodicUniverse or
ParallelepipedicPeriodicUniverse) representing the
unit cell of the crystal. If the PDB file does not define
a unit cell at all, an InfiniteUniverse is returned.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a universe</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="MMTK.PDB.PDBConfiguration.molecule_constructor">
<tt class="descname">molecule_constructor</tt><a class="headerlink" href="#MMTK.PDB.PDBConfiguration.molecule_constructor" title="Permalink to this definition">¶</a></dt>
<dd><p>alias of <a class="reference internal" href="#MMTK.PDB.PDBMolecule" title="MMTK.PDB.PDBMolecule"><tt class="xref py py-class docutils literal"><span class="pre">PDBMolecule</span></tt></a></p>
</dd></dl>

<dl class="attribute">
<dt id="MMTK.PDB.PDBConfiguration.nucleotide_chain_constructor">
<tt class="descname">nucleotide_chain_constructor</tt><a class="headerlink" href="#MMTK.PDB.PDBConfiguration.nucleotide_chain_constructor" title="Permalink to this definition">¶</a></dt>
<dd><p>alias of <a class="reference internal" href="#MMTK.PDB.PDBNucleotideChain" title="MMTK.PDB.PDBNucleotideChain"><tt class="xref py py-class docutils literal"><span class="pre">PDBNucleotideChain</span></tt></a></p>
</dd></dl>

<dl class="attribute">
<dt id="MMTK.PDB.PDBConfiguration.peptide_chain_constructor">
<tt class="descname">peptide_chain_constructor</tt><a class="headerlink" href="#MMTK.PDB.PDBConfiguration.peptide_chain_constructor" title="Permalink to this definition">¶</a></dt>
<dd><p>alias of <a class="reference internal" href="#MMTK.PDB.PDBPeptideChain" title="MMTK.PDB.PDBPeptideChain"><tt class="xref py py-class docutils literal"><span class="pre">PDBPeptideChain</span></tt></a></p>
</dd></dl>

</dd></dl>

<dl class="attribute">
<dt id="MMTK.PDB.PDBFile">
<tt class="descclassname">MMTK.PDB.</tt><tt class="descname">PDBFile</tt><a class="headerlink" href="#MMTK.PDB.PDBFile" title="Permalink to this definition">¶</a></dt>
<dd><p>alias of <a class="reference internal" href="#MMTK.PDB.PDBConfiguration" title="MMTK.PDB.PDBConfiguration"><tt class="xref py py-class docutils literal"><span class="pre">PDBConfiguration</span></tt></a></p>
</dd></dl>

<dl class="class">
<dt id="MMTK.PDB.PDBMolecule">
<em class="property">class </em><tt class="descclassname">MMTK.PDB.</tt><tt class="descname">PDBMolecule</tt><big>(</big><em>name</em>, <em>atoms=None</em>, <em>number=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBMolecule"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBMolecule" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">Scientific.IO.PDB.Molecule</span></tt></p>
<p>Molecule in a PDB file</p>
<p>See the description of Scientific.IO.PDB.Molecule for the
constructor and additional methods. In MMTK, PDBMolecule objects
are usually obtained from a PDBConfiguration object via its
attribute molecules (see the documentation of
Scientific.IO.PDB.Structure). A molecule is by definition any
residue in a PDB file that is not an amino acid or nucleotide
residue.</p>
<p>&#64;param name: the name of the group
&#64;type name: C{str}
&#64;param atoms: a list of atoms (or C{None} for no atoms)
&#64;type atoms: C{list} or C{NoneType}
&#64;param number: the PDB residue number (or C{None})
&#64;type number: C{int} or C{NoneType}</p>
<dl class="method">
<dt id="MMTK.PDB.PDBMolecule.createMolecule">
<tt class="descname">createMolecule</tt><big>(</big><em>name=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBMolecule.createMolecule"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBMolecule.createMolecule" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a <a class="reference internal" href="#MMTK.ChemicalObjects.Molecule" title="MMTK.ChemicalObjects.Molecule"><tt class="xref py py-class docutils literal"><span class="pre">Molecule</span></tt></a> object corresponding 
to the molecule in the PDB file. The parameter name 
specifies the molecule name as defined in the chemical database.
It can be left out for known molecules (currently
only water).</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ChemicalObjects.Molecule" title="MMTK.ChemicalObjects.Molecule"><tt class="xref py py-class docutils literal"><span class="pre">Molecule</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.PDB.PDBNucleotideChain">
<em class="property">class </em><tt class="descclassname">MMTK.PDB.</tt><tt class="descname">PDBNucleotideChain</tt><big>(</big><em>residues=None</em>, <em>chain_id=None</em>, <em>segment_id=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBNucleotideChain"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBNucleotideChain" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">Scientific.IO.PDB.NucleotideChain</span></tt>, <tt class="xref py py-class docutils literal"><span class="pre">MMTK.PDB.PDBChain</span></tt></p>
<p>Nucleotide chain in a PDB file</p>
<p>See the description of Scientific.IO.PDB.NucleotideChain for the
constructor and additional methods. In MMTK, PDBNucleotideChain
objects are usually obtained from a PDBConfiguration object via
its attribute nucleotide_chains (see the documentation of
Scientific.IO.PDB.Structure).</p>
<dl class="docutils">
<dt>&#64;param residues: a list of residue objects, or C{None} meaning</dt>
<dd>that the chain is initially empty</dd>
</dl>
<p>&#64;type residues: C{list} or C{NoneType}
&#64;param chain_id: a one-letter chain identifier or C{None}
&#64;type chain_id: C{str} or C{NoneType}
&#64;param segment_id: a multi-character segment identifier or C{None}
&#64;type segment_id: C{str} or C{NoneType}</p>
<dl class="method">
<dt id="MMTK.PDB.PDBNucleotideChain.createNucleotideChain">
<tt class="descname">createNucleotideChain</tt><big>(</big><em>model='all'</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBNucleotideChain.createNucleotideChain"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBNucleotideChain.createNucleotideChain" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a <a class="reference internal" href="#MMTK.NucleicAcids.NucleotideChain" title="MMTK.NucleicAcids.NucleotideChain"><tt class="xref py py-class docutils literal"><span class="pre">NucleotideChain</span></tt></a> object corresponding 
to the nucleotide chain in the PDB file. The parameter model
has the same meaning as for the NucleotideChain constructor.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.NucleicAcids.NucleotideChain" title="MMTK.NucleicAcids.NucleotideChain"><tt class="xref py py-class docutils literal"><span class="pre">NucleotideChain</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.PDB.PDBOutputFile">
<em class="property">class </em><tt class="descclassname">MMTK.PDB.</tt><tt class="descname">PDBOutputFile</tt><big>(</big><em>filename</em>, <em>subformat=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBOutputFile"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBOutputFile" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>PDB file for output</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>filename</strong> (<em>str</em>) &#8211; the name of the PDB file that is created</li>
<li><strong>subformat</strong> (<em>str</em>) &#8211; a variant of the PDB format; these subformats
are defined in module Scientific.IO.PDB. The
default is the standard PDB format.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.PDB.PDBOutputFile.close">
<tt class="descname">close</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBOutputFile.close"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBOutputFile.close" title="Permalink to this definition">¶</a></dt>
<dd><p>Closes the file. Must be called in order to prevent data loss.</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.PDB.PDBOutputFile.nextModel">
<tt class="descname">nextModel</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBOutputFile.nextModel"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBOutputFile.nextModel" title="Permalink to this definition">¶</a></dt>
<dd><p>Start a new model</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.PDB.PDBOutputFile.write">
<tt class="descname">write</tt><big>(</big><em>object</em>, <em>configuration=None</em>, <em>tag=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBOutputFile.write"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBOutputFile.write" title="Permalink to this definition">¶</a></dt>
<dd><p>Write  an object to the file</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>object</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; the object to be written</li>
<li><strong>configuration</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a>) &#8211; the configuration from which the coordinates 
are taken (default: current configuration)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.PDB.PDBPeptideChain">
<em class="property">class </em><tt class="descclassname">MMTK.PDB.</tt><tt class="descname">PDBPeptideChain</tt><big>(</big><em>residues=None</em>, <em>chain_id=None</em>, <em>segment_id=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBPeptideChain"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBPeptideChain" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">Scientific.IO.PDB.PeptideChain</span></tt>, <tt class="xref py py-class docutils literal"><span class="pre">MMTK.PDB.PDBChain</span></tt></p>
<p>Peptide chain in a PDB file</p>
<p>See the description of Scientific.IO.PDB.PeptideChain for the
constructor and additional methods. In MMTK, PDBPeptideChain
objects are usually obtained from a PDBConfiguration object via
its attribute peptide_chains (see the documentation of
Scientific.IO.PDB.Structure).</p>
<dl class="docutils">
<dt>&#64;param residues: a list of residue objects, or C{None} meaning</dt>
<dd>that the chain is initially empty</dd>
</dl>
<p>&#64;type residues: C{list} or C{NoneType}
&#64;param chain_id: a one-letter chain identifier or C{None}
&#64;type chain_id: C{str} or C{NoneType}
&#64;param segment_id: a multi-character segment identifier or C{None}
&#64;type segment_id: C{str} or C{NoneType}</p>
<dl class="method">
<dt id="MMTK.PDB.PDBPeptideChain.createPeptideChain">
<tt class="descname">createPeptideChain</tt><big>(</big><em>model='all'</em>, <em>n_terminus=None</em>, <em>c_terminus=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDB.html#PDBPeptideChain.createPeptideChain"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDB.PDBPeptideChain.createPeptideChain" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a <a class="reference internal" href="#MMTK.Proteins.PeptideChain" title="MMTK.Proteins.PeptideChain"><tt class="xref py py-class docutils literal"><span class="pre">PeptideChain</span></tt></a> object corresponding to the
peptide chain in the PDB file. The parameter model
has the same meaning as for the PeptideChain constructor.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Proteins.PeptideChain" title="MMTK.Proteins.PeptideChain"><tt class="xref py py-class docutils literal"><span class="pre">PeptideChain</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<div class="section" id="module-MMTK.PDBMoleculeFactory">
<span id="mmtk-pdbmoleculefactory"></span><h3>MMTK.PDBMoleculeFactory<a class="headerlink" href="#module-MMTK.PDBMoleculeFactory" title="Permalink to this headline">¶</a></h3>
<p>Molecule factory defined by a PDB file</p>
<p>This module permits the construction of molecular objects that correspond
exactly to the contents of a PDB file. It is used for working with
experimental data. Note that most force fields cannot be applied to the
systems generated in this way because the molecule factory does not know
any force field parameters.</p>
<dl class="class">
<dt id="MMTK.PDBMoleculeFactory.PDBMoleculeFactory">
<em class="property">class </em><tt class="descclassname">MMTK.PDBMoleculeFactory.</tt><tt class="descname">PDBMoleculeFactory</tt><big>(</big><em>pdb_conf</em>, <em>residue_filter=None</em>, <em>atom_filter=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDBMoleculeFactory.html#PDBMoleculeFactory"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDBMoleculeFactory.PDBMoleculeFactory" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.MoleculeFactory.MoleculeFactory" title="MMTK.MoleculeFactory.MoleculeFactory"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.MoleculeFactory.MoleculeFactory</span></tt></a></p>
<p>A PDBMoleculeFactory generates molecules and universes from the
contents of a PDBConfiguration. Nothing is added or left out
(except as defined by the optional residue and atom filters), and
the atom and residue names are exactly those of the original PDB
file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>pdb_conf</strong> (<a class="reference internal" href="#MMTK.PDB.PDBConfiguration" title="MMTK.PDB.PDBConfiguration"><tt class="xref py py-class docutils literal"><span class="pre">PDBConfiguration</span></tt></a>) &#8211; a PDBConfiguration</li>
<li><strong>residue_filter</strong> (<em>callable</em>) &#8211; a function taking a residue object
(as defined in Scientific.IO.PDB)
and returning True if that residue is
to be kept in the molecule factory</li>
<li><strong>atom_filter</strong> (<em>callable</em>) &#8211; a function taking a residue object and an
atom object (as defined in Scientific.IO.PDB)     
and returning True if that atom is 
to be kept in the molecule factory</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.PDBMoleculeFactory.PDBMoleculeFactory.retrieveAsymmetricUnit">
<tt class="descname">retrieveAsymmetricUnit</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/PDBMoleculeFactory.html#PDBMoleculeFactory.retrieveAsymmetricUnit"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDBMoleculeFactory.PDBMoleculeFactory.retrieveAsymmetricUnit" title="Permalink to this definition">¶</a></dt>
<dd><p>Constructs a universe (OrthrhombicPeriodicUniverse or
ParallelepipedicPeriodicUniverse) representing the
unit cell of the crystal and adds the molecules representing
the asymmetric unit.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a universe</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.PDBMoleculeFactory.PDBMoleculeFactory.retrieveMolecules">
<tt class="descname">retrieveMolecules</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/PDBMoleculeFactory.html#PDBMoleculeFactory.retrieveMolecules"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDBMoleculeFactory.PDBMoleculeFactory.retrieveMolecules" title="Permalink to this definition">¶</a></dt>
<dd><p>Constructs Molecule objects corresponding to the contents of the
PDBConfiguration. Each peptide or nucleotide chain becomes
one molecule. For residues that are neither amino acids nor
nucleic acids, each residue becomes one molecule.</p>
<p>Chain molecules (peptide and nucleotide chains) can be iterated
over to retrieve the individual residues. The residues can also
be accessed as attributes whose names are the three-letter
residue name (upper case) followed by an underscore and the
residue number from the PDB file (e.g. &#8216;GLY_34&#8217;).</p>
<p>Each object that corresponds to a PDB residue (i.e. each
residue in a chain and each non-chain molecule) has the
attributes &#8216;residue_name&#8217; and &#8216;residue_number&#8217;. Each atom has
the attributes &#8216;serial_number&#8217;, &#8216;occupancy&#8217; and
&#8216;temperature_factor&#8217;. Atoms for which a ANISOU record exists
also have an attribute &#8216;u&#8217; whose value is a tensor object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a list of Molecule objects</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.PDBMoleculeFactory.PDBMoleculeFactory.retrieveUnitCell">
<tt class="descname">retrieveUnitCell</tt><big>(</big><em>compact=True</em><big>)</big><a class="reference internal" href="_modules/MMTK/PDBMoleculeFactory.html#PDBMoleculeFactory.retrieveUnitCell"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDBMoleculeFactory.PDBMoleculeFactory.retrieveUnitCell" title="Permalink to this definition">¶</a></dt>
<dd><p>Constructs a universe (OrthrhombicPeriodicUniverse or
ParallelepipedicPeriodicUniverse) representing the
unit cell of the crystal and adds all the molecules it
contains, i.e. the molecules of the asymmetric unit and
its images obtained by applying the crystallographic
symmetry operations.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>compact</strong> (<em>bool</em>) &#8211; if True, the images are shifted such that
their centers of mass lie inside the unit cell.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a universe</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.PDBMoleculeFactory.PDBMoleculeFactory.retrieveUniverse">
<tt class="descname">retrieveUniverse</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/PDBMoleculeFactory.html#PDBMoleculeFactory.retrieveUniverse"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.PDBMoleculeFactory.PDBMoleculeFactory.retrieveUniverse" title="Permalink to this definition">¶</a></dt>
<dd><p>Constructs an empty universe (OrthrhombicPeriodicUniverse or
ParallelepipedicPeriodicUniverse) representing the
unit cell of the crystal. If the PDB file does not define
a unit cell at all, an InfiniteUniverse is returned.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a universe</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
</div>
<div class="section" id="module-MMTK.ParticleProperties">
<span id="mmtk-particleproperties"></span><h2>MMTK.ParticleProperties<a class="headerlink" href="#module-MMTK.ParticleProperties" title="Permalink to this headline">¶</a></h2>
<p>Quantities defined for each particle in a universe</p>
<dl class="class">
<dt id="MMTK.ParticleProperties.Configuration">
<em class="property">class </em><tt class="descclassname">MMTK.ParticleProperties.</tt><tt class="descname">Configuration</tt><big>(</big><em>universe</em>, <em>data_array=None</em>, <em>cell=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#Configuration"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.Configuration" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ParticleProperties.ParticleVector</span></tt></a></p>
<p>Configuration of a universe</p>
<p>Configuration instances represent a configuration of a universe,
consisting of positions for all atoms (like in a ParticleVector) plus
the geometry of the universe itself, e.g. the cell shape for
periodic universes.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe for which the values are defined</li>
<li><strong>data_array</strong> (<em>Scientific.N.array_type</em>) &#8211; the data array containing the values for each
particle. If None, a new array containing
zeros is created and used. Otherwise, the
array myst be of shape (N,3), where N is the
number of particles in the universe.</li>
<li><strong>cell</strong> &#8211; the cell parameters of the universe,
i.e. the return value of universe.cellParameters()</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.ParticleProperties.ParticleProperty">
<em class="property">class </em><tt class="descclassname">MMTK.ParticleProperties.</tt><tt class="descname">ParticleProperty</tt><big>(</big><em>universe</em>, <em>data_rank</em>, <em>value_rank</em><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleProperty"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleProperty" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Property defined for each particle</p>
<p>This is an abstract base class; for creating instances, use one of
its subclasses.</p>
<p>ParticleProperty objects store properties that are defined per
particle, such as mass, position, velocity, etc. The value
corresponding to a particular atom can be retrieved or changed by
indexing with the atom object.</p>
<dl class="method">
<dt id="MMTK.ParticleProperties.ParticleProperty.assign">
<tt class="descname">assign</tt><big>(</big><em>other</em><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleProperty.assign"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleProperty.assign" title="Permalink to this definition">¶</a></dt>
<dd><p>Copy all values from another compatible ParticleProperty object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>other</strong> &#8211; the data source</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="MMTK.ParticleProperties.ParticleProperty.return_class">
<tt class="descname">return_class</tt><a class="headerlink" href="#MMTK.ParticleProperties.ParticleProperty.return_class" title="Permalink to this definition">¶</a></dt>
<dd><p>alias of <a class="reference internal" href="#MMTK.ParticleProperties.ParticleProperty" title="MMTK.ParticleProperties.ParticleProperty"><tt class="xref py py-class docutils literal"><span class="pre">ParticleProperty</span></tt></a></p>
</dd></dl>

<dl class="method">
<dt id="MMTK.ParticleProperties.ParticleProperty.scaleBy">
<tt class="descname">scaleBy</tt><big>(</big><em>factor</em><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleProperty.scaleBy"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleProperty.scaleBy" title="Permalink to this definition">¶</a></dt>
<dd><p>Multiply all values by a factor</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>factor</strong> (<em>float</em>) &#8211; the scale factor</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ParticleProperties.ParticleProperty.selectAtoms">
<tt class="descname">selectAtoms</tt><big>(</big><em>condition</em><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleProperty.selectAtoms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleProperty.selectAtoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a collection containing all atoms a for which 
condition(property[a]) is True.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>condition</strong> (<em>callable</em>) &#8211; a test function</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ParticleProperties.ParticleProperty.sumOverParticles">
<tt class="descname">sumOverParticles</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleProperty.sumOverParticles"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleProperty.sumOverParticles" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the sum of the values for all particles.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">element type</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ParticleProperties.ParticleProperty.zero">
<tt class="descname">zero</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleProperty.zero"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleProperty.zero" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">an object of the element type (scalar, vector, etc.)
with the value 0.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">element type</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.ParticleProperties.ParticleScalar">
<em class="property">class </em><tt class="descclassname">MMTK.ParticleProperties.</tt><tt class="descname">ParticleScalar</tt><big>(</big><em>universe</em>, <em>data_array=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleScalar"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleScalar" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.ParticleProperties.ParticleProperty" title="MMTK.ParticleProperties.ParticleProperty"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ParticleProperties.ParticleProperty</span></tt></a></p>
<p>Scalar property defined for each particle</p>
<p>ParticleScalar objects can be added to each other and
multiplied with scalars.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe for which the values are defined</li>
<li><strong>data_array</strong> (<em>Scientific.N.array_type</em>) &#8211; the data array containing the values for each
particle. If None, a new array containing
zeros is created and used. Otherwise, the
array myst be of shape (N,), where N is the
number of particles in the universe.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.ParticleProperties.ParticleScalar.applyFunction">
<tt class="descname">applyFunction</tt><big>(</big><em>function</em><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleScalar.applyFunction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleScalar.applyFunction" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>function</strong> &#8211; a function that is applied to each data value</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a new ParticleScalar object containing the function results</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ParticleProperties.ParticleScalar.maximum">
<tt class="descname">maximum</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleScalar.maximum"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleScalar.maximum" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the highest value in the data array particle</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ParticleProperties.ParticleScalar.minimum">
<tt class="descname">minimum</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleScalar.minimum"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleScalar.minimum" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the smallest value in the data array particle</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="MMTK.ParticleProperties.ParticleScalar.return_class">
<tt class="descname">return_class</tt><a class="headerlink" href="#MMTK.ParticleProperties.ParticleScalar.return_class" title="Permalink to this definition">¶</a></dt>
<dd><p>alias of <a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a></p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.ParticleProperties.ParticleTensor">
<em class="property">class </em><tt class="descclassname">MMTK.ParticleProperties.</tt><tt class="descname">ParticleTensor</tt><big>(</big><em>universe</em>, <em>data_array=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleTensor"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleTensor" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.ParticleProperties.ParticleProperty" title="MMTK.ParticleProperties.ParticleProperty"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ParticleProperties.ParticleProperty</span></tt></a></p>
<p>Rank-2 tensor property defined for each particle</p>
<p>ParticleTensor objects can be added to each other and
multiplied with scalars or <a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a> objects; all
of these operations result in another ParticleTensor object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe for which the values are defined</li>
<li><strong>data_array</strong> (<em>Scientific.N.array_type</em>) &#8211; the data array containing the values for each
particle. If None, a new array containing
zeros is created and used. Otherwise, the
array myst be of shape (N,3,3), where N is the
number of particles in the universe.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="MMTK.ParticleProperties.ParticleTensor.return_class">
<tt class="descname">return_class</tt><a class="headerlink" href="#MMTK.ParticleProperties.ParticleTensor.return_class" title="Permalink to this definition">¶</a></dt>
<dd><p>alias of <a class="reference internal" href="#MMTK.ParticleProperties.ParticleTensor" title="MMTK.ParticleProperties.ParticleTensor"><tt class="xref py py-class docutils literal"><span class="pre">ParticleTensor</span></tt></a></p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.ParticleProperties.ParticleVector">
<em class="property">class </em><tt class="descclassname">MMTK.ParticleProperties.</tt><tt class="descname">ParticleVector</tt><big>(</big><em>universe</em>, <em>data_array=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleVector"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleVector" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.ParticleProperties.ParticleProperty" title="MMTK.ParticleProperties.ParticleProperty"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ParticleProperties.ParticleProperty</span></tt></a></p>
<p>Vector property defined for each particle</p>
<p>ParticleVector objects can be added to each other and
multiplied with scalars or <a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a> objects; all
of these operations result in another ParticleVector
object. Multiplication with a vector or another ParticleVector object
yields a <a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a> object containing the dot products
for each particle. Multiplications that treat ParticleVectors
as vectors in a 3N-dimensional space are implemented as methods.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe for which the values are defined</li>
<li><strong>data_array</strong> (<em>Scientific.N.array_type</em>) &#8211; the data array containing the values for each
particle. If None, a new array containing
zeros is created and used. Otherwise, the
array myst be of shape (N,3), where N is the
number of particles in the universe.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.ParticleProperties.ParticleVector.dotProduct">
<tt class="descname">dotProduct</tt><big>(</big><em>other</em><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleVector.dotProduct"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleVector.dotProduct" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>other</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>) &#8211; another ParticleVector</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the dot product with other, treating both operands
as 3N-dimensional vectors.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ParticleProperties.ParticleVector.dyadicProduct">
<tt class="descname">dyadicProduct</tt><big>(</big><em>other</em><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleVector.dyadicProduct"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleVector.dyadicProduct" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>other</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>) &#8211; another ParticleVector</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the dyadic product with other</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ParticleProperties.ParticleTensor" title="MMTK.ParticleProperties.ParticleTensor"><tt class="xref py py-class docutils literal"><span class="pre">ParticleTensor</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ParticleProperties.ParticleVector.length">
<tt class="descname">length</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleVector.length"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleVector.length" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the length (norm) of the vector for each particle</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ParticleProperties.ParticleVector.massWeightedDotProduct">
<tt class="descname">massWeightedDotProduct</tt><big>(</big><em>other</em><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleVector.massWeightedDotProduct"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleVector.massWeightedDotProduct" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>other</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>) &#8211; another ParticleVector</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the mass-weighted dot product with other treating both
operands as 3N-dimensional vectors</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ParticleProperties.ParticleVector.massWeightedNorm">
<tt class="descname">massWeightedNorm</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleVector.massWeightedNorm"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleVector.massWeightedNorm" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the mass-weighted norm of the ParticleVector seen as a
3N-dimensional vector</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.ParticleProperties.ParticleVector.norm">
<tt class="descname">norm</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#ParticleVector.norm"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.ParticleVector.norm" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the norm of the ParticleVector seen as a 3N-dimensional
vector</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="MMTK.ParticleProperties.ParticleVector.return_class">
<tt class="descname">return_class</tt><a class="headerlink" href="#MMTK.ParticleProperties.ParticleVector.return_class" title="Permalink to this definition">¶</a></dt>
<dd><p>alias of <a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a></p>
</dd></dl>

<dl class="method">
<dt id="MMTK.ParticleProperties.ParticleVector.totalNorm">
<tt class="descname">totalNorm</tt><big>(</big><big>)</big><a class="headerlink" href="#MMTK.ParticleProperties.ParticleVector.totalNorm" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the norm of the ParticleVector seen as a 3N-dimensional
vector</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="MMTK.ParticleProperties.isConfiguration">
<tt class="descclassname">MMTK.ParticleProperties.</tt><tt class="descname">isConfiguration</tt><big>(</big><em>object</em><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#isConfiguration"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.isConfiguration" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>object</strong> &#8211; any object</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">True if object is a <a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a></td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">bool</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.ParticleProperties.isParticleProperty">
<tt class="descclassname">MMTK.ParticleProperties.</tt><tt class="descname">isParticleProperty</tt><big>(</big><em>object</em><big>)</big><a class="reference internal" href="_modules/MMTK/ParticleProperties.html#isParticleProperty"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ParticleProperties.isParticleProperty" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>object</strong> &#8211; any object</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">True if object is a <a class="reference internal" href="#MMTK.ParticleProperties.ParticleProperty" title="MMTK.ParticleProperties.ParticleProperty"><tt class="xref py py-class docutils literal"><span class="pre">ParticleProperty</span></tt></a></td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">bool</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.ProteinFriction">
<span id="mmtk-proteinfriction"></span><h2>MMTK.ProteinFriction<a class="headerlink" href="#module-MMTK.ProteinFriction" title="Permalink to this headline">¶</a></h2>
<p>A friction constant model for C<sub>α</sub> models of proteins</p>
<dl class="function">
<dt id="MMTK.ProteinFriction.calphaFrictionConstants">
<tt class="descclassname">MMTK.ProteinFriction.</tt><tt class="descname">calphaFrictionConstants</tt><big>(</big><em>protein</em>, <em>set=2</em><big>)</big><a class="reference internal" href="_modules/MMTK/ProteinFriction.html#calphaFrictionConstants"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.ProteinFriction.calphaFrictionConstants" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>protein</strong> (<a class="reference internal" href="#MMTK.Proteins.Protein" title="MMTK.Proteins.Protein"><tt class="xref py py-class docutils literal"><span class="pre">Protein</span></tt></a>) &#8211; a C<sub>α</sub> model protein</li>
<li><strong>set</strong> &#8211; the number of a friction constant set (1, 2, 3, or 4)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the estimated friction constants for the atoms in the protein</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last"><a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.Proteins">
<span id="mmtk-proteins"></span><h2>MMTK.Proteins<a class="headerlink" href="#module-MMTK.Proteins" title="Permalink to this headline">¶</a></h2>
<p>Peptide chains and proteins</p>
<dl class="class">
<dt id="MMTK.Proteins.ConnectedChains">
<em class="property">class </em><tt class="descclassname">MMTK.Proteins.</tt><tt class="descname">ConnectedChains</tt><big>(</big><em>chains=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#ConnectedChains"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.ConnectedChains" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Proteins.PeptideChain" title="MMTK.Proteins.PeptideChain"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Proteins.PeptideChain</span></tt></a></p>
<p>Peptide chains connected by disulfide bridges</p>
<p>A group of peptide chains connected by disulfide bridges must be considered
a single molecule due to the presence of chemical bonds. Such a molecule
is represented by a ConnectedChains object. These objects are created
automatically when a Protein object is assembled. They are normally
not used directly by application programs. When a chain with disulfide
bridges to other chains is extracted from a Protein object, the
return value is a SubChain object that indirectly refers to a
ConnectedChains object.</p>
</dd></dl>

<dl class="class">
<dt id="MMTK.Proteins.PeptideChain">
<em class="property">class </em><tt class="descclassname">MMTK.Proteins.</tt><tt class="descname">PeptideChain</tt><big>(</big><em>sequence</em>, <em>**properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#PeptideChain"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.PeptideChain" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Biopolymers.ResidueChain" title="MMTK.Biopolymers.ResidueChain"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Biopolymers.ResidueChain</span></tt></a></p>
<p>Peptide chain</p>
<p>Peptide chains consist of amino acid residues that are linked
by peptide bonds. They are a special kind of molecule, i.e.
all molecule operations are available.</p>
<p>Peptide chains act as sequences of residues. If p is a PeptideChain
object, then</p>
<blockquote>
<div><ul>
<li><p class="first">len(p) yields the number of residues</p>
</li>
<li><p class="first">p[i] yields residue number i</p>
</li>
<li><dl class="first docutils">
<dt>p[i:j] yields the subchain from residue number i up to</dt>
<dd><p class="first last">but excluding residue number j</p>
</dd>
</dl>
</li>
</ul>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>sequence</strong> &#8211; the amino acid sequence. This can be a string
containing the one-letter codes, or a list
of three-letter codes, or a
<a class="reference internal" href="#MMTK.PDB.PDBPeptideChain" title="MMTK.PDB.PDBPeptideChain"><tt class="xref py py-class docutils literal"><span class="pre">PDBPeptideChain</span></tt></a> object.
If a PDBPeptideChain object is supplied, the atomic
positions it contains are assigned to the atoms
of the newly generated peptide chain, otherwise the
positions of all atoms are undefined.</li>
<li><strong>model</strong> (<em>str</em>) &#8211; one of &#8220;all&#8221; (all-atom), &#8220;no_hydrogens&#8221; or &#8220;none&#8221;
(no hydrogens), &#8220;polar_hydrogens&#8221; or &#8220;polar&#8221;
(united-atom with only polar hydrogens),
&#8220;polar_charmm&#8221; (like &#8220;polar&#8221;, but defining
polar hydrogens like in the CHARMM force field),
&#8220;polar_opls&#8221; (like &#8220;polar&#8221;, but defining
polar hydrogens like in the latest OPLS force field),
&#8220;calpha&#8221; (only the C<sub>α</sub> atom of each residue).
Default is &#8220;all&#8221;.</li>
<li><strong>n_terminus</strong> (<em>bool</em>) &#8211; if True, the first residue is constructed
using the N-terminal variant, if False the
non-terminal version is used. Default is True.</li>
<li><strong>c_terminus</strong> (<em>bool</em>) &#8211; if True, the last residue is constructed
using the C-terminal variant, if False the
non-terminal version is used. Default is True.</li>
<li><strong>circular</strong> (<em>bool</em>) &#8211; if True, a peptide bond is constructed
between the first and the last residues.
Default is False.</li>
<li><strong>name</strong> (<em>str</em>) &#8211; a name for the chain (a string)</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.Proteins.PeptideChain.backbone">
<tt class="descname">backbone</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#PeptideChain.backbone"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.PeptideChain.backbone" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the peptide groups of all residues</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Collections.Collection" title="MMTK.Collections.Collection"><tt class="xref py py-class docutils literal"><span class="pre">Collection</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Proteins.PeptideChain.phiPsi">
<tt class="descname">phiPsi</tt><big>(</big><em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#PeptideChain.phiPsi"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.PeptideChain.phiPsi" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a list of the (phi, psi) backbone angles for each residue</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list of tuple of float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Proteins.PeptideChain.replaceResidue">
<tt class="descname">replaceResidue</tt><big>(</big><em>r_old</em>, <em>r_new</em><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#PeptideChain.replaceResidue"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.PeptideChain.replaceResidue" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>r_old</strong> (<em>Residue</em>) &#8211; the residue to be replaced (must be part of the chain)</li>
<li><strong>r_new</strong> (<em>Residue</em>) &#8211; the residue that replaces r_old</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Proteins.PeptideChain.sequence">
<tt class="descname">sequence</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#PeptideChain.sequence"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.PeptideChain.sequence" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the primary sequence as a list of three-letter
residue codes.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Proteins.PeptideChain.sidechains">
<tt class="descname">sidechains</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#PeptideChain.sidechains"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.PeptideChain.sidechains" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the sidechain groups of all residues</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Collections.Collection" title="MMTK.Collections.Collection"><tt class="xref py py-class docutils literal"><span class="pre">Collection</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Proteins.Protein">
<em class="property">class </em><tt class="descclassname">MMTK.Proteins.</tt><tt class="descname">Protein</tt><big>(</big><em>*items</em>, <em>**properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#Protein"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.Protein" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.ChemicalObjects.Complex" title="MMTK.ChemicalObjects.Complex"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.ChemicalObjects.Complex</span></tt></a></p>
<p>Protein</p>
<p>A Protein object is a special kind of <a class="reference internal" href="#MMTK.ChemicalObjects.Complex" title="MMTK.ChemicalObjects.Complex"><tt class="xref py py-class docutils literal"><span class="pre">Complex</span></tt></a>
object which is made up of peptide chains and possibly ligands.</p>
<p>If the atoms in the peptide chains that make up a protein have
defined positions, sulfur bridges within chains and between
chains will be constructed automatically during protein generation
based on a distance criterion between cystein sidechains.</p>
<p>Proteins act as sequences of chains. If p is a Protein object, then</p>
<ul class="simple">
<li>len(p) yields the number of chains</li>
<li>p[i] yields chain number i</li>
</ul>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>items</strong> &#8211; either a sequence of peptide chain objects, or
a string, which is interpreted as the name of a
database definition for a protein.
If that definition does not exist, the string
is taken to be the name of a PDB file, from which
all peptide chains are constructed and
assembled into a protein.</li>
<li><strong>model</strong> (<em>str</em>) &#8211; one of &#8220;all&#8221; (all-atom), &#8220;no_hydrogens&#8221; or &#8220;none&#8221;
(no hydrogens),&#8221;polar_hydrogens&#8221; or &#8220;polar&#8221;
(united-atom with only polar hydrogens),
&#8220;polar_charmm&#8221; (like &#8220;polar&#8221;, but defining
polar hydrogens like in the CHARMM force field),
&#8220;polar_opls&#8221; (like &#8220;polar&#8221;, but defining
polar hydrogens like in the latest OPLS force field),
&#8220;calpha&#8221; (only the C<sub>α</sub> atom of each residue).
Default is &#8220;all&#8221;.</li>
<li><strong>position</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the center-of-mass position of the protein</li>
<li><strong>name</strong> (<em>str</em>) &#8211; a name for the protein</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.Proteins.Protein.backbone">
<tt class="descname">backbone</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#Protein.backbone"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.Protein.backbone" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the peptide groups of all residues in all chains</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Collections.Collection" title="MMTK.Collections.Collection"><tt class="xref py py-class docutils literal"><span class="pre">Collection</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Proteins.Protein.phiPsi">
<tt class="descname">phiPsi</tt><big>(</big><em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#Protein.phiPsi"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.Protein.phiPsi" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a list of the (phi, psi) backbone angles for all residue
in all chains</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list of list of tuple of float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Proteins.Protein.residues">
<tt class="descname">residues</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#Protein.residues"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.Protein.residues" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">all residues in all chains</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Collections.Collection" title="MMTK.Collections.Collection"><tt class="xref py py-class docutils literal"><span class="pre">Collection</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Proteins.Protein.residuesOfType">
<tt class="descname">residuesOfType</tt><big>(</big><em>*types</em><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#Protein.residuesOfType"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.Protein.residuesOfType" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>types</strong> (<em>sequence of str</em>) &#8211; a sequence of residue codes (one- or three-letter)</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">all residues whose type (one- or three-letter code)
is contained in types</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Collections.Collection" title="MMTK.Collections.Collection"><tt class="xref py py-class docutils literal"><span class="pre">Collection</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Proteins.Protein.sidechains">
<tt class="descname">sidechains</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#Protein.sidechains"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.Protein.sidechains" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the sidechain groups of all residues in all chains</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Collections.Collection" title="MMTK.Collections.Collection"><tt class="xref py py-class docutils literal"><span class="pre">Collection</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Proteins.Residue">
<em class="property">class </em><tt class="descclassname">MMTK.Proteins.</tt><tt class="descname">Residue</tt><big>(</big><em>name=None</em>, <em>model='all'</em><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#Residue"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.Residue" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Biopolymers.Residue" title="MMTK.Biopolymers.Residue"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Biopolymers.Residue</span></tt></a></p>
<p>Amino acid residue</p>
<p>Amino acid residues are a special kind of group. They are defined
in the chemical database. Each residue has two subgroups
(&#8216;peptide&#8217; and &#8216;sidechain&#8217;) and is usually connected to other
residues to form a peptide chain. The database contains three
variants of each residue (N-terminal, C-terminal,
non-terminal) and various models (all-atom, united-atom,
C<sub>α</sub>).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>name</strong> (<em>str</em>) &#8211; the name of the residue in the chemical database. This
is the full name of the residue plus the suffix
&#8220;_nt&#8221; or &#8220;_ct&#8221; for the terminal variants.</li>
<li><strong>model</strong> (<em>str</em>) &#8211; one of &#8220;all&#8221; (all-atom), &#8220;none&#8221; (no hydrogens),
&#8220;polar&#8221; (united-atom with only polar hydrogens),
&#8220;polar_charmm&#8221; (like &#8220;polar&#8221;, but defining
polar hydrogens like in the CHARMM force field),
&#8220;polar_opls&#8221; (like &#8220;polar&#8221;, but defining
polar hydrogens like in the latest OPLS force field),
&#8220;calpha&#8221; (only the C<sub>α</sub> atom).</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.Proteins.Residue.backbone">
<tt class="descname">backbone</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#Residue.backbone"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.Residue.backbone" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the peptide group</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ChemicalObjects.Group" title="MMTK.ChemicalObjects.Group"><tt class="xref py py-class docutils literal"><span class="pre">Group</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Proteins.Residue.chiAngle">
<tt class="descname">chiAngle</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#Residue.chiAngle"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.Residue.chiAngle" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">an object representing the chi angle and allowing to modify it</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">MMTK.InternalCoordinates.DihedralAngle</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Proteins.Residue.phiAngle">
<tt class="descname">phiAngle</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#Residue.phiAngle"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.Residue.phiAngle" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">an object representing the phi angle and allowing to modify it</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">MMTK.InternalCoordinates.DihedralAngle</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Proteins.Residue.phiPsi">
<tt class="descname">phiPsi</tt><big>(</big><em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#Residue.phiPsi"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.Residue.phiPsi" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the values of the backbone dihedral angles phi and psi.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">tuple (float, float)</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Proteins.Residue.psiAngle">
<tt class="descname">psiAngle</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#Residue.psiAngle"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.Residue.psiAngle" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">an object representing the psi angle and allowing to modify it</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">MMTK.InternalCoordinates.DihedralAngle</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Proteins.Residue.sidechains">
<tt class="descname">sidechains</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#Residue.sidechains"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.Residue.sidechains" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the sidechain group</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ChemicalObjects.Group" title="MMTK.ChemicalObjects.Group"><tt class="xref py py-class docutils literal"><span class="pre">Group</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Proteins.SubChain">
<em class="property">class </em><tt class="descclassname">MMTK.Proteins.</tt><tt class="descname">SubChain</tt><big>(</big><em>chain=None</em>, <em>groups=None</em>, <em>name=''</em><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#SubChain"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.SubChain" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Proteins.PeptideChain" title="MMTK.Proteins.PeptideChain"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Proteins.PeptideChain</span></tt></a></p>
<p>A contiguous part of a peptide chain</p>
<p>SubChain objects are the result of slicing operations on
PeptideChain objects. They cannot be created directly.
SubChain objects permit all operations of PeptideChain
objects, but cannot be added to a universe.</p>
</dd></dl>

<dl class="function">
<dt id="MMTK.Proteins.isPeptideChain">
<tt class="descclassname">MMTK.Proteins.</tt><tt class="descname">isPeptideChain</tt><big>(</big><em>x</em><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#isPeptideChain"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.isPeptideChain" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>x</strong> &#8211; any object</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">True if x is a peptide chain</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">bool</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.Proteins.isProtein">
<tt class="descclassname">MMTK.Proteins.</tt><tt class="descname">isProtein</tt><big>(</big><em>x</em><big>)</big><a class="reference internal" href="_modules/MMTK/Proteins.html#isProtein"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Proteins.isProtein" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>x</strong> &#8211; any object</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">True if x is a protein</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">bool</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.Random">
<span id="mmtk-random"></span><h2>MMTK.Random<a class="headerlink" href="#module-MMTK.Random" title="Permalink to this headline">¶</a></h2>
<p>Random quantities for use in molecular simulations</p>
<dl class="function">
<dt id="MMTK.Random.randomDirection">
<tt class="descclassname">MMTK.Random.</tt><tt class="descname">randomDirection</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Random.html#randomDirection"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Random.randomDirection" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a vector drawn from a uniform distribution on the surface
of a unit sphere.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Vector</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.Random.randomDirections">
<tt class="descclassname">MMTK.Random.</tt><tt class="descname">randomDirections</tt><big>(</big><em>n</em><big>)</big><a class="reference internal" href="_modules/MMTK/Random.html#randomDirections"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Random.randomDirections" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>n</strong> &#8211; the number of direction vectors</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a list of n vectors drawn from a uniform distribution on
the surface of a unit sphere. If n is negative, returns
a deterministic list of not more than -n vectors of unit
length (useful for testing purposes).</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.Random.randomParticleVector">
<tt class="descclassname">MMTK.Random.</tt><tt class="descname">randomParticleVector</tt><big>(</big><em>universe</em>, <em>width</em><big>)</big><a class="reference internal" href="_modules/MMTK/Random.html#randomParticleVector"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Random.randomParticleVector" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; a universe</li>
<li><strong>width</strong> (<em>float</em>) &#8211; the width (standard deviation) of a Gaussian distribution</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">a set of vectors drawn from a Gaussian distribution
with a given width centered  around zero.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last"><a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.Random.randomPointInBox">
<tt class="descclassname">MMTK.Random.</tt><tt class="descname">randomPointInBox</tt><big>(</big><em>a</em>, <em>b=None</em>, <em>c=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Random.html#randomPointInBox"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Random.randomPointInBox" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>a</strong> (<em>float</em>) &#8211; the edge length of a box along the x axis</li>
<li><strong>b</strong> (<em>float</em>) &#8211; the edge length of a box along the y axis (default: a)</li>
<li><strong>c</strong> (<em>float</em>) &#8211; the edge length of a box along the z axis (default: a)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">a vector drawn from a uniform distribution within a
rectangular box with edge lengths a, b, c.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">Scientific.Geometry.Vector</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.Random.randomPointInSphere">
<tt class="descclassname">MMTK.Random.</tt><tt class="descname">randomPointInSphere</tt><big>(</big><em>r</em><big>)</big><a class="reference internal" href="_modules/MMTK/Random.html#randomPointInSphere"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Random.randomPointInSphere" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>r</strong> (<em>float</em>) &#8211; the radius of a sphere</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a vector drawn from a uniform distribution within
a sphere of radius r.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Vector</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.Random.randomRotation">
<tt class="descclassname">MMTK.Random.</tt><tt class="descname">randomRotation</tt><big>(</big><em>max_angle=3.141592653589793</em><big>)</big><a class="reference internal" href="_modules/MMTK/Random.html#randomRotation"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Random.randomRotation" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>max_angle</strong> (<em>float</em>) &#8211; the upper limit for the rotation angle</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a random rotation with a uniform axis distribution
and angles drawn from a uniform distribution between
-max_angle and max_angle.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Transformations.Rotation</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.Random.randomVelocity">
<tt class="descclassname">MMTK.Random.</tt><tt class="descname">randomVelocity</tt><big>(</big><em>temperature</em>, <em>mass</em><big>)</big><a class="reference internal" href="_modules/MMTK/Random.html#randomVelocity"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Random.randomVelocity" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>temperature</strong> (<em>float</em>) &#8211; the temperature defining a Maxwell distribution</li>
<li><strong>mass</strong> (<em>float</em>) &#8211; the mass of a particle</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">a random velocity vector for a particle of a given mass
at a given temperature</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">Scientific.Geometry.Vector</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.Solvation">
<span id="mmtk-solvation"></span><h2>MMTK.Solvation<a class="headerlink" href="#module-MMTK.Solvation" title="Permalink to this headline">¶</a></h2>
<p>Solvation of solute molecules</p>
<dl class="function">
<dt id="MMTK.Solvation.addSolvent">
<tt class="descclassname">MMTK.Solvation.</tt><tt class="descname">addSolvent</tt><big>(</big><em>universe</em>, <em>solvent</em>, <em>density</em>, <em>scale_factor=4.0</em><big>)</big><a class="reference internal" href="_modules/MMTK/Solvation.html#addSolvent"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Solvation.addSolvent" title="Permalink to this definition">¶</a></dt>
<dd><p>Scales up the universe and adds as many solvent molecules
as are necessary to obtain the specified solvent density,
taking account of the solute molecules that are already present
in the universe. The molecules are placed at random positions
in the scaled-up universe, but without overlaps between
any two molecules.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; a finite universe</li>
<li><strong>solvent</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Molecule" title="MMTK.ChemicalObjects.Molecule"><tt class="xref py py-class docutils literal"><span class="pre">Molecule</span></tt></a> or str) &#8211; a molecule, or the name of a molecule in the database</li>
<li><strong>density</strong> (<em>float</em>) &#8211; the density of the solvent (amu/nm**3)</li>
<li><strong>scale_factor</strong> (<em>float</em>) &#8211; the factor by which the initial universe is
expanded before adding the solvent molecules</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.Solvation.numberOfSolventMolecules">
<tt class="descclassname">MMTK.Solvation.</tt><tt class="descname">numberOfSolventMolecules</tt><big>(</big><em>universe</em>, <em>solvent</em>, <em>density</em><big>)</big><a class="reference internal" href="_modules/MMTK/Solvation.html#numberOfSolventMolecules"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Solvation.numberOfSolventMolecules" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; a finite universe</li>
<li><strong>solvent</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Molecule" title="MMTK.ChemicalObjects.Molecule"><tt class="xref py py-class docutils literal"><span class="pre">Molecule</span></tt></a> or str) &#8211; a molecule, or the name of a molecule in the database</li>
<li><strong>density</strong> (<em>float</em>) &#8211; the density of the solvent (amu/nm**3)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the number of solvent molecules that must be added to the
universe, in addition to whatever it already contains,
to obtain the given solvent density.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">int</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.Solvation.shrinkUniverse">
<tt class="descclassname">MMTK.Solvation.</tt><tt class="descname">shrinkUniverse</tt><big>(</big><em>universe</em>, <em>temperature=300.0</em>, <em>trajectory=None</em>, <em>scale_factor=0.95</em><big>)</big><a class="reference internal" href="_modules/MMTK/Solvation.html#shrinkUniverse"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Solvation.shrinkUniverse" title="Permalink to this definition">¶</a></dt>
<dd><p>Shrinks the universe, which must have been scaled up by
<a class="reference internal" href="#MMTK.Solvation.addSolvent" title="MMTK.Solvation.addSolvent"><tt class="xref py py-class docutils literal"><span class="pre">addSolvent</span></tt></a>, back to its original size.
The compression is performed in small steps, in between which
some energy minimization and molecular dynamics steps are executed.
The molecular dynamics is run at the given temperature, and
an optional trajectory can be specified in which intermediate
configurations are stored.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; a finite universe</li>
<li><strong>temperature</strong> (<em>float</em>) &#8211; the temperature at which the Molecular Dynamics
steps are run</li>
<li><strong>trajectory</strong> (<a class="reference internal" href="#MMTK.Trajectory.Trajectory" title="MMTK.Trajectory.Trajectory"><tt class="xref py py-class docutils literal"><span class="pre">Trajectory</span></tt></a> or str) &#8211; the trajectory in which the progress of the
shrinking procedure is stored, or a filename</li>
<li><strong>scale_factor</strong> (<em>float</em>) &#8211; the factor by which the universe is scaled
at each reduction step</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.Subspace">
<span id="mmtk-subspace"></span><h2>MMTK.Subspace<a class="headerlink" href="#module-MMTK.Subspace" title="Permalink to this headline">¶</a></h2>
<p>Linear subspaces for infinitesimal motions</p>
<p>This module implements subspaces for infinitesimal (or finite
small-amplitude) atomic motions. They can be used in normal mode
calculations or for analyzing complex motions. For an explanation, see:</p>
<blockquote>
<div><div class="line-block">
<div class="line">K. Hinsen and G.R. Kneller</div>
<div class="line">Projection methods for the analysis of complex motions in macromolecules</div>
<div class="line">Mol. Sim. 23:275-292 (2000)</div>
</div>
</div></blockquote>
<dl class="class">
<dt id="MMTK.Subspace.LinkedRigidBodyMotionSubspace">
<em class="property">class </em><tt class="descclassname">MMTK.Subspace.</tt><tt class="descname">LinkedRigidBodyMotionSubspace</tt><big>(</big><em>universe</em>, <em>rigid_bodies</em><big>)</big><a class="reference internal" href="_modules/MMTK/Subspace.html#LinkedRigidBodyMotionSubspace"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Subspace.LinkedRigidBodyMotionSubspace" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Subspace.Subspace" title="MMTK.Subspace.Subspace"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Subspace.Subspace</span></tt></a></p>
<p>Subspace for the motion of linked rigid bodies</p>
<p>This class describes the same subspace as RigidBodyMotionSubspace,
and is used by the latter. For collections of several chains of
linked rigid bodies, RigidBodyMotionSubspace is more efficient.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe for which the subspace is created</li>
<li><strong>rigid_bodies</strong> &#8211; a list or set of rigid bodies
with some common atoms</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Subspace.PairDistanceSubspace">
<em class="property">class </em><tt class="descclassname">MMTK.Subspace.</tt><tt class="descname">PairDistanceSubspace</tt><big>(</big><em>universe</em>, <em>atom_pairs</em><big>)</big><a class="reference internal" href="_modules/MMTK/Subspace.html#PairDistanceSubspace"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Subspace.PairDistanceSubspace" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Subspace.Subspace" title="MMTK.Subspace.Subspace"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Subspace.Subspace</span></tt></a></p>
<p>Subspace of pair-distance motions</p>
<p>A pair-distance motion subspace is the subspace which contains
the relative motions of any number of atom pairs along
their distance vector, e.g. bond elongation between two
bonded atoms.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe for which the subspace is created</li>
<li><strong>atom_pairs</strong> &#8211; a sequence of atom pairs whose distance-vector
motion is included in the subspace</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Subspace.RigidMotionSubspace">
<em class="property">class </em><tt class="descclassname">MMTK.Subspace.</tt><tt class="descname">RigidMotionSubspace</tt><big>(</big><em>universe</em>, <em>objects</em><big>)</big><a class="reference internal" href="_modules/MMTK/Subspace.html#RigidMotionSubspace"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Subspace.RigidMotionSubspace" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Subspace.Subspace" title="MMTK.Subspace.Subspace"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Subspace.Subspace</span></tt></a></p>
<p>Subspace of rigid-body motions</p>
<p>A rigid-body motion subspace is the subspace which contains
the rigid-body motions of any number of chemical objects.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe for which the subspace is created</li>
<li><strong>objects</strong> &#8211; a sequence of objects whose rigid-body motion is
included in the subspace</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Subspace.Subspace">
<em class="property">class </em><tt class="descclassname">MMTK.Subspace.</tt><tt class="descname">Subspace</tt><big>(</big><em>universe</em>, <em>vectors</em><big>)</big><a class="reference internal" href="_modules/MMTK/Subspace.html#Subspace"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Subspace.Subspace" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Subspace of infinitesimal atomic motions</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> (<a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">Universe</span></tt></a>) &#8211; the universe for which the subspace is created</li>
<li><strong>vectors</strong> (<em>list</em>) &#8211; a list of <a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>
objects that define the subspace. They need not be
orthogonal or linearly independent.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.Subspace.Subspace.complement">
<tt class="descname">complement</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Subspace.html#Subspace.complement"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Subspace.Subspace.complement" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the orthogonal complement subspace</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.Subspace.Subspace" title="MMTK.Subspace.Subspace"><tt class="xref py py-class docutils literal"><span class="pre">Subspace</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Subspace.Subspace.getBasis">
<tt class="descname">getBasis</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Subspace.html#Subspace.getBasis"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Subspace.Subspace.getBasis" title="Permalink to this definition">¶</a></dt>
<dd><p>Construct a basis for the subspace by orthonormalization of
the input vectors using Singular Value Decomposition. The
basis consists of a sequence of
<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>
objects that are orthonormal in configuration space.
:returns: the basis</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.Subspace.Subspace.projectionComplementOf">
<tt class="descname">projectionComplementOf</tt><big>(</big><em>vector</em><big>)</big><a class="reference internal" href="_modules/MMTK/Subspace.html#Subspace.projectionComplementOf"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Subspace.Subspace.projectionComplementOf" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>vector</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>) &#8211; a particle vector</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the projection of the vector onto the orthogonal complement
of the subspace.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Subspace.Subspace.projectionOf">
<tt class="descname">projectionOf</tt><big>(</big><em>vector</em><big>)</big><a class="reference internal" href="_modules/MMTK/Subspace.html#Subspace.projectionOf"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Subspace.Subspace.projectionOf" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>vector</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>) &#8211; a particle vector</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the projection of the vector onto the subspace.</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-MMTK.Trajectory">
<span id="mmtk-trajectory"></span><h2>MMTK.Trajectory<a class="headerlink" href="#module-MMTK.Trajectory" title="Permalink to this headline">¶</a></h2>
<p>Trajectory files and their contents</p>
<dl class="class">
<dt id="MMTK.Trajectory.LogOutput">
<em class="property">class </em><tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">LogOutput</tt><big>(</big><em>file</em>, <em>data=None</em>, <em>first=0</em>, <em>last=None</em>, <em>skip=1</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#LogOutput"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.LogOutput" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Trajectory.TrajectoryOutput" title="MMTK.Trajectory.TrajectoryOutput"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Trajectory.TrajectoryOutput</span></tt></a></p>
<p>Protocol file output action</p>
<p>A LogOutput object can be used in the action list of any
trajectory-generating operation. It writes any of the available
data to a text file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>file</strong> &#8211; a file object or a string, which is interpreted as the
name of a file that is opened in write mode</li>
<li><strong>data</strong> &#8211; a list of data categories. All variables provided by the
trajectory generator that fall in any of the listed
categories are written to the trajectory file. See the
descriptions of the trajectory generators for a list
of variables and categories. By default (data = None)
the categories &#8220;configuration&#8221;, &#8220;energy&#8221;,
&#8220;thermodynamic&#8221;, and &#8220;time&#8221; are written.</li>
<li><strong>first</strong> (<em>int</em>) &#8211; the number of the first step at which the action is run</li>
<li><strong>last</strong> (<em>int</em>) &#8211; the number of the step at which the action is suspended.
A value of None indicates that the action should
be applied indefinitely.</li>
<li><strong>skip</strong> (<em>int</em>) &#8211; the number of steps to skip between two action runs</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Trajectory.ParticleTrajectory">
<em class="property">class </em><tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">ParticleTrajectory</tt><big>(</big><em>trajectory</em>, <em>atom</em>, <em>first=0</em>, <em>last=None</em>, <em>skip=1</em>, <em>variable='configuration'</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#ParticleTrajectory"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.ParticleTrajectory" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Trajectory data for a single particle</p>
<p>A ParticleTrajectory object is created by calling the method
<a class="reference internal" href="#MMTK.Trajectory.Trajectory.readParticleTrajectory" title="MMTK.Trajectory.Trajectory.readParticleTrajectory"><tt class="xref py py-func docutils literal"><span class="pre">readParticleTrajectory()</span></tt></a>
on a <a class="reference internal" href="#MMTK.Trajectory.Trajectory" title="MMTK.Trajectory.Trajectory"><tt class="xref py py-class docutils literal"><span class="pre">Trajectory</span></tt></a> object.</p>
<p>If pt is a ParticleTrajectory object, then</p>
<blockquote>
<div><ul class="simple">
<li>len(pt) is the number of steps stored in it</li>
<li>pt[i] is the value at step i (a vector)</li>
</ul>
</div></blockquote>
<dl class="method">
<dt id="MMTK.Trajectory.ParticleTrajectory.translateBy">
<tt class="descname">translateBy</tt><big>(</big><em>vector</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#ParticleTrajectory.translateBy"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.ParticleTrajectory.translateBy" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds a vector to the values at all steps. This does B{not}
change the data in the trajectory file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>vector</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the vector to be added</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Trajectory.RestartTrajectoryOutput">
<em class="property">class </em><tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">RestartTrajectoryOutput</tt><big>(</big><em>trajectory</em>, <em>skip=100</em>, <em>length=3</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#RestartTrajectoryOutput"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.RestartTrajectoryOutput" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Trajectory.TrajectoryOutput" title="MMTK.Trajectory.TrajectoryOutput"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Trajectory.TrajectoryOutput</span></tt></a></p>
<p>Restart trajectory output action</p>
<p>A RestartTrajectoryOutput object is used in the action list of any
trajectory-generating operation. It writes those variables to a
trajectory that the trajectory generator declares as necessary
for restarting.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>trajectory</strong> &#8211; a trajectory object or a string, which is interpreted
as the name of a file that is opened as a trajectory
in append mode with a cycle length of length and
double-precision variables</li>
<li><strong>skip</strong> (<em>int</em>) &#8211; the number of steps between two write operations to the
restart trajectory</li>
<li><strong>length</strong> &#8211; the number of steps stored in the restart trajectory;
used only if trajectory is a string</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Trajectory.RigidBodyTrajectory">
<em class="property">class </em><tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">RigidBodyTrajectory</tt><big>(</big><em>trajectory</em>, <em>object</em>, <em>first=0</em>, <em>last=None</em>, <em>skip=1</em>, <em>reference=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#RigidBodyTrajectory"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.RigidBodyTrajectory" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Rigid-body trajectory data</p>
<p>A RigidBodyTrajectory object is created by calling the method
<a class="reference internal" href="#MMTK.Trajectory.Trajectory.readRigidBodyTrajectory" title="MMTK.Trajectory.Trajectory.readRigidBodyTrajectory"><tt class="xref py py-func docutils literal"><span class="pre">readRigidBodyTrajectory()</span></tt></a>
on a <a class="reference internal" href="#MMTK.Trajectory.Trajectory" title="MMTK.Trajectory.Trajectory"><tt class="xref py py-class docutils literal"><span class="pre">Trajectory</span></tt></a> object.</p>
<p>If rbt is a RigidBodyTrajectory object, then</p>
<blockquote>
<div><ul class="simple">
<li>len(rbt) is the number of steps stored in it</li>
<li>rbt[i] is the value at step i (a vector for the center of mass
and a quaternion for the orientation)</li>
</ul>
</div></blockquote>
</dd></dl>

<dl class="class">
<dt id="MMTK.Trajectory.SnapshotGenerator">
<em class="property">class </em><tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">SnapshotGenerator</tt><big>(</big><em>universe</em>, <em>**options</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#SnapshotGenerator"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.SnapshotGenerator" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Trajectory.TrajectoryGenerator" title="MMTK.Trajectory.TrajectoryGenerator"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Trajectory.TrajectoryGenerator</span></tt></a></p>
<p>Trajectory generator for single steps</p>
<p>A SnapshotGenerator is used for manual assembly of trajectory
files. At each call it writes one step to the trajectory,
using the current state of the universe (configuration, velocities, etc.)
and data provided explicitly with the call.</p>
<p>Each call to the SnapshotGenerator object produces one step.
All the keyword options can be specified either when
creating the generator or when calling it.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>universe</strong> &#8211; the universe on which the generator acts</li>
<li><strong>data</strong> &#8211; a dictionary that supplies values for variables
that are not part of the universe state
(e.g. potential energy)</li>
<li><strong>actions</strong> &#8211; a list of actions to be executed periodically
(default is none)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Trajectory.StandardLogOutput">
<em class="property">class </em><tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">StandardLogOutput</tt><big>(</big><em>skip=50</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#StandardLogOutput"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.StandardLogOutput" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Trajectory.LogOutput" title="MMTK.Trajectory.LogOutput"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Trajectory.LogOutput</span></tt></a></p>
<p>Standard protocol output action</p>
<p>A StandardLogOutput object can be used in the action list of any
trajectory-generating operation. It is a specialization of
LogOutput to the most common case and writes data in the categories
&#8220;time&#8221; and &#8220;energy&#8221; to the standard output stream.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>skip</strong> (<em>int</em>) &#8211; the number of steps to skip between two action runs</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Trajectory.SubTrajectory">
<em class="property">class </em><tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">SubTrajectory</tt><big>(</big><em>trajectory</em>, <em>indices</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#SubTrajectory"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.SubTrajectory" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Reference to a subset of a trajectory</p>
<p>A SubTrajectory object is created by slicing a Trajectory object
or another SubTrajectory object. It provides all the operations
defined on Trajectory objects.</p>
</dd></dl>

<dl class="class">
<dt id="MMTK.Trajectory.SubVariable">
<em class="property">class </em><tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">SubVariable</tt><big>(</big><em>variable</em>, <em>indices</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#SubVariable"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.SubVariable" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Trajectory.TrajectoryVariable" title="MMTK.Trajectory.TrajectoryVariable"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Trajectory.TrajectoryVariable</span></tt></a></p>
<p>Reference to a subset of a <a class="reference internal" href="#MMTK.Trajectory.TrajectoryVariable" title="MMTK.Trajectory.TrajectoryVariable"><tt class="xref py py-class docutils literal"><span class="pre">TrajectoryVariable</span></tt></a></p>
<p>A SubVariable object is created by slicing a TrajectoryVariable
object or another SubVariable object. It provides all the operations
defined on TrajectoryVariable objects.</p>
</dd></dl>

<dl class="class">
<dt id="MMTK.Trajectory.Trajectory">
<em class="property">class </em><tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">Trajectory</tt><big>(</big><em>object</em>, <em>filename</em>, <em>mode='r'</em>, <em>comment=None</em>, <em>double_precision=False</em>, <em>cycle=0</em>, <em>block_size=1</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#Trajectory"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.Trajectory" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Trajectory file</p>
<p>The data in a trajectory file can be accessed by step or by
variable. If t is a Trajectory object, then:</p>
<blockquote>
<div><ul class="simple">
<li>len(t) is the number of steps</li>
<li>t[i] is the data for step i, in the form of a dictionary that
maps variable names to data</li>
<li>t[i:j] and t[i:j:n] return a <a class="reference internal" href="#MMTK.Trajectory.SubTrajectory" title="MMTK.Trajectory.SubTrajectory"><tt class="xref py py-class docutils literal"><span class="pre">SubTrajectory</span></tt></a> 
object that refers to a subset of the total number of steps 
(no data is copied)</li>
<li>t.variable returns the value of the named variable at all
time steps. If the variable is a simple scalar, it is read
completely and returned as an array. If the variable contains
data for each atom, a <a class="reference internal" href="#MMTK.Trajectory.TrajectoryVariable" title="MMTK.Trajectory.TrajectoryVariable"><tt class="xref py py-class docutils literal"><span class="pre">TrajectoryVariable</span></tt></a> 
object is returned from which data at specific steps can be obtained 
by further indexing operations.</li>
</ul>
</div></blockquote>
<p>The routines that generate trajectories decide what variables
are used and what they contain. The most frequently used variable
is &#8220;configuration&#8221;, which stores the positions of all atoms.
Other common variables are &#8220;time&#8221;, &#8220;velocities&#8221;, &#8220;temperature&#8221;,
&#8220;pressure&#8221;, and various energy terms whose name end with &#8220;_energy&#8221;.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>object</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.ChemicalObject" title="MMTK.ChemicalObjects.ChemicalObject"><tt class="xref py py-class docutils literal"><span class="pre">ChemicalObject</span></tt></a>) &#8211; the object whose data is stored in the trajectory file.
This can be &#8216;None&#8217; when opening a file for reading;
in that case, a universe object is constructed from the
description stored in the trajectory file. This universe
object can be accessed via the attribute &#8216;universe&#8217;
of the trajectory object.</li>
<li><strong>filename</strong> (<em>str</em>) &#8211; the name of the trajectory file</li>
<li><strong>mode</strong> (<em>str</em>) &#8211; one of &#8220;r&#8221; (read-only), &#8220;w&#8221; (create new file for writing),
or &#8220;a&#8221; (append to existing file or create if the file does
not exist)</li>
<li><strong>comment</strong> (<em>str</em>) &#8211; optional comment that is stored in the file;
allowed only with mode=&#8221;r&#8221;</li>
<li><strong>double_precision</strong> (<em>bool</em>) &#8211; if True, data in the file is stored using
double precision; default is single precision.
Note that all I/O via trajectory objects is
double precision; conversion from and to
single precision file variables is handled
automatically.</li>
<li><strong>cycle</strong> (<em>int</em>) &#8211; if non-zero, a trajectory is created for a fixed number
of steps equal to the value of cycle, and these steps
are used cyclically. This is meant for restart
trajectories.</li>
<li><strong>block_size</strong> (<em>int</em>) &#8211; an optimization parameter that influences the file
structure and the I/O performance for very large
files. A block size of 1 is optimal for sequential
access to configurations etc., whereas a block size
equal to the number of steps is optimal for reading
coordinates or scalar variables along the time axis.
The default value is 1. Note that older MMTK releases
always used a block size of 1 and cannot handle
trajectories with different block sizes.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.Trajectory.Trajectory.close">
<tt class="descname">close</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#Trajectory.close"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.Trajectory.close" title="Permalink to this definition">¶</a></dt>
<dd><p>Close the trajectory file. Must be called after writing to
ensure that all buffered data is written to the file. No data
access is possible after closing a file.</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.Trajectory.Trajectory.flush">
<tt class="descname">flush</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#Trajectory.flush"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.Trajectory.flush" title="Permalink to this definition">¶</a></dt>
<dd><p>Make sure that all data that has been written to the trajectory
is also written to the file.</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.Trajectory.Trajectory.readParticleTrajectory">
<tt class="descname">readParticleTrajectory</tt><big>(</big><em>atom</em>, <em>first=0</em>, <em>last=None</em>, <em>skip=1</em>, <em>variable='configuration'</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#Trajectory.readParticleTrajectory"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.Trajectory.readParticleTrajectory" title="Permalink to this definition">¶</a></dt>
<dd><p>Read trajectory information for a single atom but for multiple
time steps.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>atom</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.Atom" title="MMTK.ChemicalObjects.Atom"><tt class="xref py py-class docutils literal"><span class="pre">Atom</span></tt></a>) &#8211; the atom whose trajectory is requested</li>
<li><strong>first</strong> (<em>int</em>) &#8211; the number of the first step to be read</li>
<li><strong>last</strong> (<em>int</em>) &#8211; the number of the first step not to be read.
A value of None indicates that the
whole trajectory should be read.</li>
<li><strong>skip</strong> (<em>int</em>) &#8211; the number of steps to skip between two steps read</li>
<li><strong>variable</strong> (<em>str</em>) &#8211; the name of the trajectory variable to be read.
If the variable is &#8220;configuration&#8221;, the resulting
trajectory is made continuous by eliminating all
jumps caused by periodic boundary conditions.
The pseudo-variable &#8220;box_coordinates&#8221; can be read
to obtain the values of the variable &#8220;configuration&#8221;
scaled to box coordinates. For non-periodic universes
there is no difference between box coordinates
and real coordinates.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the trajectory for a single atom</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last"><a class="reference internal" href="#MMTK.Trajectory.ParticleTrajectory" title="MMTK.Trajectory.ParticleTrajectory"><tt class="xref py py-class docutils literal"><span class="pre">ParticleTrajectory</span></tt></a></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Trajectory.Trajectory.readRigidBodyTrajectory">
<tt class="descname">readRigidBodyTrajectory</tt><big>(</big><em>object</em>, <em>first=0</em>, <em>last=None</em>, <em>skip=1</em>, <em>reference=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#Trajectory.readRigidBodyTrajectory"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.Trajectory.readRigidBodyTrajectory" title="Permalink to this definition">¶</a></dt>
<dd><p>Read the positions for an object at multiple time steps
and extract the rigid-body motion (center-of-mass position plus
orientation as a quaternion) by an optimal-transformation fit.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>object</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; the object whose rigid-body trajectory is requested</li>
<li><strong>first</strong> (<em>int</em>) &#8211; the number of the first step to be read</li>
<li><strong>last</strong> (<em>int</em>) &#8211; the number of the first step not to be read.
A value of None indicates that the
whole trajectory should be read.</li>
<li><strong>skip</strong> (<em>int</em>) &#8211; the number of steps to skip between two steps read</li>
<li><strong>reference</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a>) &#8211; the reference configuration for the fit</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the trajectory for a single rigid body</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last"><a class="reference internal" href="#MMTK.Trajectory.RigidBodyTrajectory" title="MMTK.Trajectory.RigidBodyTrajectory"><tt class="xref py py-class docutils literal"><span class="pre">RigidBodyTrajectory</span></tt></a></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Trajectory.Trajectory.variables">
<tt class="descname">variables</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#Trajectory.variables"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.Trajectory.variables" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a list of the names of all variables that are stored
in the trajectory</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list of str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Trajectory.Trajectory.view">
<tt class="descname">view</tt><big>(</big><em>first=0</em>, <em>last=None</em>, <em>skip=1</em>, <em>object=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#Trajectory.view"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.Trajectory.view" title="Permalink to this definition">¶</a></dt>
<dd><p>Show an animation of the trajectory using an external visualization
program.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>first</strong> (<em>int</em>) &#8211; the number of the first step in the animation</li>
<li><strong>last</strong> (<em>int</em>) &#8211; the number of the first step not to include in the
animation. A value of None indicates that the
whole trajectory should be used.</li>
<li><strong>skip</strong> (<em>int</em>) &#8211; the number of steps to skip between two steps read</li>
<li><strong>object</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; the object to be animated, which must be in the
universe stored in the trajectory. None
stands for the whole universe.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Trajectory.TrajectoryAction">
<em class="property">class </em><tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">TrajectoryAction</tt><big>(</big><em>first</em>, <em>last</em>, <em>skip</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#TrajectoryAction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.TrajectoryAction" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Trajectory action base class</p>
<p>Subclasses of this base class implement the actions that can be
inserted into trajectory generation at regular intervals.</p>
</dd></dl>

<dl class="class">
<dt id="MMTK.Trajectory.TrajectoryGenerator">
<em class="property">class </em><tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">TrajectoryGenerator</tt><big>(</big><em>universe</em>, <em>options</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#TrajectoryGenerator"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.TrajectoryGenerator" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Trajectory generator base class</p>
<p>This base class implements the common aspects of everything that
generates trajectories: integrators, minimizers, etc.</p>
</dd></dl>

<dl class="class">
<dt id="MMTK.Trajectory.TrajectoryOutput">
<em class="property">class </em><tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">TrajectoryOutput</tt><big>(</big><em>trajectory</em>, <em>data=None</em>, <em>first=0</em>, <em>last=None</em>, <em>skip=1</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#TrajectoryOutput"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.TrajectoryOutput" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Trajectory.TrajectoryAction" title="MMTK.Trajectory.TrajectoryAction"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Trajectory.TrajectoryAction</span></tt></a></p>
<p>Trajectory output action</p>
<p>A TrajectoryOutput object can be used in the action list of any
trajectory-generating operation. It writes any of the available
data to a trajectory file. It is possible to use several
TrajectoryOutput objects at the same time in order to produce
multiple trajectories from a single run.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>trajectory</strong> &#8211; a trajectory object or a string, which is
interpreted as the name of a file that is opened
as a trajectory in append mode</li>
<li><strong>data</strong> &#8211; a list of data categories. All variables provided by the
trajectory generator that fall in any of the listed
categories are written to the trajectory file. See the
descriptions of the trajectory generators for a list
of variables and categories. By default (data = None)
the categories &#8220;configuration&#8221;, &#8220;energy&#8221;,
&#8220;thermodynamic&#8221;, and &#8220;time&#8221; are written.</li>
<li><strong>first</strong> (<em>int</em>) &#8211; the number of the first step at which the action is run</li>
<li><strong>last</strong> (<em>int</em>) &#8211; the number of the step at which the action is suspended.
A value of None indicates that the action should
be applied indefinitely.</li>
<li><strong>skip</strong> (<em>int</em>) &#8211; the number of steps to skip between two action runs</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Trajectory.TrajectorySet">
<em class="property">class </em><tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">TrajectorySet</tt><big>(</big><em>object</em>, <em>filenames</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#TrajectorySet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.TrajectorySet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Trajectory file set</p>
<p>A TrajectorySet permits to treat a sequence of trajectory files
like a single trajectory for reading data. It behaves exactly like a
<a class="reference internal" href="#MMTK.Trajectory.Trajectory" title="MMTK.Trajectory.Trajectory"><tt class="xref py py-class docutils literal"><span class="pre">Trajectory</span></tt></a> object. The trajectory files must all contain data
for the same system. The variables stored in the individual files
need not be the same, but only variables common to all files
can be accessed.</p>
<p>Note: depending on how the sequence of trajectories was constructed,
the first configuration of each trajectory might be the same as the
last one in the preceding trajectory. To avoid counting it twice,
specify (filename, 1, None, 1) for all but the first trajectory in
the set.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>object</strong> &#8211; the object whose data is stored in the trajectory files.
This can be (and usually is) None;
in that case, a universe object is constructed from the
description stored in the first trajectory file.
This universe object can be accessed via the attribute
universe of the trajectory set object.</li>
<li><strong>filenames</strong> &#8211; a list of trajectory file names or
(filename, first_step, last_step, increment)
tuples.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Trajectory.TrajectorySetVariable">
<em class="property">class </em><tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">TrajectorySetVariable</tt><big>(</big><em>universe</em>, <em>trajectory_set</em>, <em>name</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#TrajectorySetVariable"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.TrajectorySetVariable" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Trajectory.TrajectoryVariable" title="MMTK.Trajectory.TrajectoryVariable"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Trajectory.TrajectoryVariable</span></tt></a></p>
<p>Variable in a trajectory set</p>
<p>A TrajectorySetVariable object is created by extracting a variable from
a TrajectorySet object if that variable contains data for each atom and
is thus potentially large. It behaves exactly like a TrajectoryVariable
object.</p>
</dd></dl>

<dl class="class">
<dt id="MMTK.Trajectory.TrajectoryVariable">
<em class="property">class </em><tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">TrajectoryVariable</tt><big>(</big><em>universe</em>, <em>trajectory</em>, <em>name</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#TrajectoryVariable"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.TrajectoryVariable" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Variable in a trajectory</p>
<p>A TrajectoryVariable object is created by extracting a variable from
a Trajectory object if that variable contains data for each atom and
is thus potentially large. No data is read from the trajectory file
when a TrajectoryVariable object is created; the read operation
takes place when the TrajectoryVariable is indexed with a specific
step number.</p>
<p>If t is a TrajectoryVariable object, then:</p>
<blockquote>
<div><ul class="simple">
<li>len(t) is the number of steps</li>
<li>t[i] is the data for step i, in the form of a ParticleScalar,
a ParticleVector, or a Configuration object, depending on the
variable</li>
<li>t[i:j] and t[i:j:n] return a SubVariable object that refers
to a subset of the total number of steps</li>
</ul>
</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="MMTK.Trajectory.isTrajectory">
<tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">isTrajectory</tt><big>(</big><em>object</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#isTrajectory"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.isTrajectory" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>object</strong> &#8211; any Python object</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">True if object is a trajectory</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.Trajectory.trajectoryInfo">
<tt class="descclassname">MMTK.Trajectory.</tt><tt class="descname">trajectoryInfo</tt><big>(</big><em>filename</em><big>)</big><a class="reference internal" href="_modules/MMTK/Trajectory.html#trajectoryInfo"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Trajectory.trajectoryInfo" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>filename</strong> (<em>str</em>) &#8211; the name of a trajectory file</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a string with summarial information about the trajectory</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.Units">
<span id="mmtk-units"></span><h2>MMTK.Units<a class="headerlink" href="#module-MMTK.Units" title="Permalink to this headline">¶</a></h2>
<p>Units and physical constants</p>
<p>This module defines constants and prefactors that convert from various
units to MMTK&#8217;s internal unit system. There are also some common
physical constants.</p>
<p>MMTK&#8217;s unit system is defined by:</p>
<ul class="simple">
<li>nm for length</li>
<li>ps for time</li>
<li>amu (g/mol) for mass</li>
<li>the charge of a proton for charge</li>
<li>rad for angles</li>
</ul>
<p>All formulas that do not contain electrical quantities (charge,
electric/magnetic field, ...) have the same form in this unit system
as in the SI system. The energy unit that results from the choices
given above is kJ/mol.</p>
<p>The constants defined in this module are:</p>
<blockquote>
<div><ul>
<li><dl class="first docutils">
<dt>SI Prefixes: ato, femto, pico, nano, micro, milli, centi, deci,</dt>
<dd><p class="first last">deca, hecto, kilo, mega, giga, tera, peta</p>
</dd>
</dl>
</li>
<li><p class="first">Length units: m, cm, mm, nm, pm, fm, Ang, Bohr</p>
</li>
<li><p class="first">Angle units: rad, deg</p>
</li>
<li><p class="first">Volume units: l</p>
</li>
<li><p class="first">Time units: s, ns, ps, fs</p>
</li>
<li><p class="first">Frequency units: Hz, invcm (wavenumbers)</p>
</li>
<li><p class="first">Mass units: amu, g, kg</p>
</li>
<li><p class="first">Quantity-of-matter units: mol</p>
</li>
<li><p class="first">Energy units: J, kJ, cal (thermochemical), kcal, Hartree, Bohr</p>
</li>
<li><p class="first">Temperature units: K</p>
</li>
<li><p class="first">Force units: N, dyn</p>
</li>
<li><p class="first">Pressure units: Pa, MPa, GPa, bar, kbar, atm</p>
</li>
<li><p class="first">Electrostatic units: C, A, V, D, eV, e</p>
</li>
<li><p class="first">Physical constants:</p>
<ul class="simple">
<li>c (speed of light)</li>
<li>Nav (Avogadro number)</li>
<li>h (Planck constant)</li>
<li>hbar (Planck constant divided by 2*Pi)</li>
<li>k_B (Boltzmann constant)</li>
<li>eps0 (permittivity of vacuum)</li>
<li>me (electron mass)</li>
</ul>
</li>
<li><p class="first">Other:</p>
<ul class="simple">
<li>akma_time (the time unit in the DCD trajectory format)</li>
<li>electrostatic_energy (the prefactor in Coulomb&#8217;s law)</li>
</ul>
</li>
</ul>
</div></blockquote>
</div>
<div class="section" id="module-MMTK.Universe">
<span id="mmtk-universe"></span><h2>MMTK.Universe<a class="headerlink" href="#module-MMTK.Universe" title="Permalink to this headline">¶</a></h2>
<p>Universes</p>
<dl class="class">
<dt id="MMTK.Universe.CubicPeriodicUniverse">
<em class="property">class </em><tt class="descclassname">MMTK.Universe.</tt><tt class="descname">CubicPeriodicUniverse</tt><big>(</big><em>size=None</em>, <em>forcefield=None</em>, <em>**properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#CubicPeriodicUniverse"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.CubicPeriodicUniverse" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Universe.OrthorhombicPeriodicUniverse" title="MMTK.Universe.OrthorhombicPeriodicUniverse"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Universe.OrthorhombicPeriodicUniverse</span></tt></a></p>
<p>Periodic universe with cubic elementary cell.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>size</strong> &#8211; a sequence of length three specifying the edge
lengths along the x, y, and z directions</li>
<li><strong>forcefield</strong> (<tt class="xref py py-class docutils literal"><span class="pre">ForceField</span></tt>) &#8211; a force field, or None for no force field</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.Universe.CubicPeriodicUniverse.setSize">
<tt class="descname">setSize</tt><big>(</big><em>size</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#CubicPeriodicUniverse.setSize"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.CubicPeriodicUniverse.setSize" title="Permalink to this definition">¶</a></dt>
<dd><p>Set the edge length to a given value.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>size</strong> (<em>float</em>) &#8211; the new size</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Universe.InfiniteUniverse">
<em class="property">class </em><tt class="descclassname">MMTK.Universe.</tt><tt class="descname">InfiniteUniverse</tt><big>(</big><em>forcefield=None</em>, <em>**properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#InfiniteUniverse"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.InfiniteUniverse" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Universe.Universe" title="MMTK.Universe.Universe"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Universe.Universe</span></tt></a></p>
<p>Infinite (unbounded and nonperiodic) universe.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>forcefield</strong> (<tt class="xref py py-class docutils literal"><span class="pre">ForceField</span></tt>) &#8211; a force field, or None for no force field</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="MMTK.Universe.OrthorhombicPeriodicUniverse">
<em class="property">class </em><tt class="descclassname">MMTK.Universe.</tt><tt class="descname">OrthorhombicPeriodicUniverse</tt><big>(</big><em>size=None</em>, <em>forcefield=None</em>, <em>**properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#OrthorhombicPeriodicUniverse"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.OrthorhombicPeriodicUniverse" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.Universe.Periodic3DUniverse</span></tt></p>
<p>Periodic universe with orthorhombic elementary cell.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>size</strong> &#8211; a sequence of length three specifying the edge
lengths along the x, y, and z directions</li>
<li><strong>forcefield</strong> (<tt class="xref py py-class docutils literal"><span class="pre">ForceField</span></tt>) &#8211; a force field, or None for no force field</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.Universe.OrthorhombicPeriodicUniverse.scaleSize">
<tt class="descname">scaleSize</tt><big>(</big><em>factor</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#OrthorhombicPeriodicUniverse.scaleSize"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.OrthorhombicPeriodicUniverse.scaleSize" title="Permalink to this definition">¶</a></dt>
<dd><p>Multiplies all edge lengths by a factor.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>factor</strong> (<em>float</em>) &#8211; the scale factor</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Universe.ParallelepipedicPeriodicUniverse">
<em class="property">class </em><tt class="descclassname">MMTK.Universe.</tt><tt class="descname">ParallelepipedicPeriodicUniverse</tt><big>(</big><em>shape=None</em>, <em>forcefield=None</em>, <em>**properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#ParallelepipedicPeriodicUniverse"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.ParallelepipedicPeriodicUniverse" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">MMTK.Universe.Periodic3DUniverse</span></tt></p>
<p>Periodic universe with parallelepipedic elementary cell.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>shape</strong> (<em>sequence of Scientific.Geometry.Vector</em>) &#8211; the basis vectors</li>
<li><strong>forcefield</strong> (<tt class="xref py py-class docutils literal"><span class="pre">ForceField</span></tt>) &#8211; a force field, or None for no force field</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="MMTK.Universe.ParallelepipedicPeriodicUniverse.scaleSize">
<tt class="descname">scaleSize</tt><big>(</big><em>factor</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#ParallelepipedicPeriodicUniverse.scaleSize"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.ParallelepipedicPeriodicUniverse.scaleSize" title="Permalink to this definition">¶</a></dt>
<dd><p>Multiplies all edge lengths by a factor.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>factor</strong> (<em>float</em>) &#8211; the scale factor</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="MMTK.Universe.Universe">
<em class="property">class </em><tt class="descclassname">MMTK.Universe.</tt><tt class="descname">Universe</tt><big>(</big><em>forcefield</em>, <em>properties</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Collections.GroupOfAtoms</span></tt></a>, <a class="reference internal" href="#MMTK.Visualization.Viewable" title="MMTK.Visualization.Viewable"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.Visualization.Viewable</span></tt></a></p>
<p>Universe</p>
<p>A universe represents a complete model of a chemical system, i.e.
the molecules, their environment (topology, boundary conditions,
thermostats, etc.), and optionally a force field.</p>
<p>The class Universe is an abstract base class that defines
properties common to all kinds of universes. To create universe
objects, use one of its subclasses.</p>
<p>In addition to the methods listed below, universe objects support
the following operations (u is any universe object, o is any
chemical object):</p>
<blockquote>
<div><ul class="simple">
<li>len(u) yields the number of chemical objects in the universe</li>
<li>u[i] returns object number i</li>
<li>u.name = o adds o to the universe and also makes it accessible as
an attribute</li>
<li>del u.name removes the object that was assigned to u.name from
the universe</li>
</ul>
</div></blockquote>
<dl class="method">
<dt id="MMTK.Universe.Universe.acquireConfigurationChangeLock">
<tt class="descname">acquireConfigurationChangeLock</tt><big>(</big><em>waitflag=True</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.acquireConfigurationChangeLock"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.acquireConfigurationChangeLock" title="Permalink to this definition">¶</a></dt>
<dd><p>Acquire the configuration change lock. This lock should be
acquired before starting an algorithm that changes the
configuration continuously, e.g. minimization or molecular dynamics
algorithms. This guarantees the proper order of execution when
several such operations are started in succession. For example,
when a minimization should be followed by a dynamics run,
the use of this flag permits both operations to be started
as background tasks which will be executed one after the other,
permitting other threads to run in parallel.</p>
<p>The configuration change lock should not be confused with
the universe state lock. The former guarantees the proper
sequence of long-running algorithms, whereas the latter
guarantees the consistency of the data. A dynamics algorithm,
for example, keeps the configuration change lock from the
beginning to the end, but acquires the universe state lock
only immediately before modifying configuration and velocities,
and releases it immediately afterwards.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>waitflag</strong> (<em>bool</em>) &#8211; if true, the method waits until the lock
becomes available; this is the most common mode.
If false, the method returns immediately even
if another thread holds the lock.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a flag indicating if the lock was successfully
acquired (1) or not (0).</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.acquireReadStateLock">
<tt class="descname">acquireReadStateLock</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.acquireReadStateLock"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.acquireReadStateLock" title="Permalink to this definition">¶</a></dt>
<dd><p>Acquire the universe read state lock. Any application that
uses threading must acquire this lock prior to accessing the
current state of the universe, in particular its configuration
(particle positions). This guarantees the consistency of the
data; while any thread holds the read state lock, no other
thread can obtain the write state lock that permits modifying
the state. The read state lock should be released as soon as
possible.</p>
<p>The read state lock can be acquired only if no thread holds
the write state lock. If the read state lock cannot be
acquired immediately, the thread will be blocked until
it becomes available. Any number of threads can acquire
the read state lock simultaneously.</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.acquireWriteStateLock">
<tt class="descname">acquireWriteStateLock</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.acquireWriteStateLock"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.acquireWriteStateLock" title="Permalink to this definition">¶</a></dt>
<dd><p>Acquire the universe write state lock. Any application that
uses threading must acquire this lock prior to modifying the
current state of the universe, in particular its configuration
(particle positions). This guarantees the consistency of the
data; while any thread holds the write state lock, no other
thread can obtain the read state lock that permits accessing
the state. The write state lock should be released as soon as
possible.</p>
<p>The write state lock can be acquired only if no other thread
holds either the read state lock or the write state lock. If
the write state lock cannot be acquired immediately, the
thread will be blocked until it becomes available.</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.addObject">
<tt class="descname">addObject</tt><big>(</big><em>object</em>, <em>steal=False</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.addObject"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.addObject" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds object to the universe. If object is a Collection,
all elements of the Collection are added to the universe.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>object</strong> &#8211; the object (chemical or environment) to be added</li>
<li><strong>steal</strong> (<em>bool</em>) &#8211; if True, permit stealing the object from another
universe, otherwise the object must not yet be
attached to any universe.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.addToConfiguration">
<tt class="descname">addToConfiguration</tt><big>(</big><em>displacement</em>, <em>block=True</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.addToConfiguration"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.addToConfiguration" title="Permalink to this definition">¶</a></dt>
<dd><p>Update the current configuration of the universe by adding
the given displacement vector.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>displacement</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>) &#8211; the displacement vector for each atom</li>
<li><strong>block</strong> (<em>bool</em>) &#8211; if True, the operation blocks other threads
from accessing the configuration before the update
is completed. If False, it is assumed that the
caller takes care of locking.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.adjustVelocitiesToConstraints">
<tt class="descname">adjustVelocitiesToConstraints</tt><big>(</big><em>velocities=None</em>, <em>block=True</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.adjustVelocitiesToConstraints"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.adjustVelocitiesToConstraints" title="Permalink to this definition">¶</a></dt>
<dd><p>Modifies the velocities to be compatible with
the distance constraints, i.e. projects out the velocity
components along the constrained distances.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>velocities</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>) &#8211; the velocities in which the
constraints are enforced
(None for current velocities)</li>
<li><strong>block</strong> (<em>bool</em>) &#8211; if True, the operation blocks other threads
from accessing the configuration before the update
is completed. If False, it is assumed that the
caller takes care of locking.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.angle">
<tt class="descname">angle</tt><big>(</big><em>p1</em>, <em>p2</em>, <em>p3</em>, <em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.angle"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.angle" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>p1</strong> &#8211; a vector or a chemical object whose position is taken</li>
<li><strong>p2</strong> &#8211; a vector or a chemical object whose position is taken</li>
<li><strong>p3</strong> &#8211; a vector or a chemical object whose position is taken</li>
<li><strong>conf</strong> &#8211; a configuration (None for the current configuration)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the angle between the distance vectors p1-p2 and p3-p2</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">float</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.atomList">
<tt class="descname">atomList</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.atomList"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.atomList" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a list of all atoms in the universe</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.basisVectors">
<tt class="descname">basisVectors</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.basisVectors"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.basisVectors" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the basis vectors of the elementary cell of a periodic
universe, or None for a non-periodic universe</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">NoneType or list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.boxToRealCoordinates">
<tt class="descname">boxToRealCoordinates</tt><big>(</big><em>vector</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.boxToRealCoordinates"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.boxToRealCoordinates" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>vector</strong> &#8211; a point in the universe expressed in box coordinates</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the real-space equivalent of vector</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Vector</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.cartesianToFractional">
<tt class="descname">cartesianToFractional</tt><big>(</big><em>vector</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.cartesianToFractional"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.cartesianToFractional" title="Permalink to this definition">¶</a></dt>
<dd><p>Fractional coordinates are defined only for periodic universes;
their components have values between 0. and 1.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>vector</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; a point in the universe</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the fractional coordinate equivalent of vector</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Scientific.N.array_type</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.cellVolume">
<tt class="descname">cellVolume</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.cellVolume"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.cellVolume" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the volume of the elementary cell of a periodic
universe, None for a non-periodic universe</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">NoneType or float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.charges">
<tt class="descname">charges</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.charges"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.charges" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the atomic charges defined by the universe&#8217;s
force field.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the charges of all atoms in the universe</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.configuration">
<tt class="descname">configuration</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.configuration"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.configuration" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the configuration object describing the current
configuration of the universe. Note that this is not a
copy of the current state, but a reference: the positions
in the configuration object will change when coordinate
changes are applied to the universe in whatever way.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.configurationDifference">
<tt class="descname">configurationDifference</tt><big>(</big><em>conf1</em>, <em>conf2</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.configurationDifference"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.configurationDifference" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>conf1</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a>) &#8211; a configuration</li>
<li><strong>conf2</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a>) &#8211; a configuration</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the difference vector between the two configurations
for each atom, taking into account the universe
topology (e.g. minimum-image convention).</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last"><a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.contiguousObjectConfiguration">
<tt class="descname">contiguousObjectConfiguration</tt><big>(</big><em>objects=None</em>, <em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.contiguousObjectConfiguration"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.contiguousObjectConfiguration" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>objects</strong> (<em>list</em>) &#8211; a list of chemical objects, or None for all
objects in the universe</li>
<li><strong>conf</strong> &#8211; a configuration (None for the current configuration)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">configuration conf (default: current configuration)
corrected by the contiguous object offsets for that
configuration.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last"><a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.contiguousObjectOffset">
<tt class="descname">contiguousObjectOffset</tt><big>(</big><em>objects=None</em>, <em>conf=None</em>, <em>box_coordinates=False</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.contiguousObjectOffset"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.contiguousObjectOffset" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>objects</strong> (<em>list</em>) &#8211; a list of chemical objects, or None for all
objects in the universe</li>
<li><strong>conf</strong> &#8211; a configuration (None for the current configuration)</li>
<li><strong>box_coordinates</strong> (<em>bool</em>) &#8211; use box coordinates rather than real ones</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">a set of displacement vectors relative to
the conf which, when added to the configuration,
create a configuration in which none of the objects
is split across the edge of the elementary cell.
For nonperiodic universes the return value is None.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last"><a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.copyConfiguration">
<tt class="descname">copyConfiguration</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.copyConfiguration"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.copyConfiguration" title="Permalink to this definition">¶</a></dt>
<dd><p>This operation is thread-safe; it won&#8217;t return inconsistent
data even when another thread is modifying the configuration.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a copy of the current configuration</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.dihedral">
<tt class="descname">dihedral</tt><big>(</big><em>p1</em>, <em>p2</em>, <em>p3</em>, <em>p4</em>, <em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.dihedral"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.dihedral" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>p1</strong> &#8211; a vector or a chemical object whose position is taken</li>
<li><strong>p2</strong> &#8211; a vector or a chemical object whose position is taken</li>
<li><strong>p3</strong> &#8211; a vector or a chemical object whose position is taken</li>
<li><strong>p4</strong> &#8211; a vector or a chemical object whose position is taken</li>
<li><strong>conf</strong> &#8211; a configuration (None for the current configuration)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the dihedral angle between the plane containing the
distance vectors p1-p2 and p3-p2 and the plane containing
the distance vectors p2-p3 and p4-p3</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">float</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.distance">
<tt class="descname">distance</tt><big>(</big><em>p1</em>, <em>p2</em>, <em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.distance"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.distance" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>p1</strong> &#8211; a vector or a chemical object whose position is taken</li>
<li><strong>p2</strong> &#8211; a vector or a chemical object whose position is taken</li>
<li><strong>conf</strong> &#8211; a configuration (None for the current configuration)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the distance between p1 and p2, i.e. the length
of the distance vector</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">float</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.distanceConstraintList">
<tt class="descname">distanceConstraintList</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.distanceConstraintList"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.distanceConstraintList" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the list of distance constraints</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.distanceVector">
<tt class="descname">distanceVector</tt><big>(</big><em>p1</em>, <em>p2</em>, <em>conf=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.distanceVector"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.distanceVector" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>p1</strong> &#8211; a vector or a chemical object whose position is taken</li>
<li><strong>p2</strong> &#8211; a vector or a chemical object whose position is taken</li>
<li><strong>conf</strong> &#8211; a configuration (None for the current configuration)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">the distance vector between p1 and p2 (i.e. the
vector from p1 to p2) in the configuration conf,
taking into account the universe&#8217;s topology.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.energy">
<tt class="descname">energy</tt><big>(</big><em>subset1=None</em>, <em>subset2=None</em>, <em>small_change=False</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.energy"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.energy" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>subset1</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.ChemicalObject" title="MMTK.ChemicalObjects.ChemicalObject"><tt class="xref py py-class docutils literal"><span class="pre">ChemicalObject</span></tt></a>) &#8211; a subset of a universe, or None</li>
<li><strong>subset2</strong> (<a class="reference internal" href="#MMTK.ChemicalObjects.ChemicalObject" title="MMTK.ChemicalObjects.ChemicalObject"><tt class="xref py py-class docutils literal"><span class="pre">ChemicalObject</span></tt></a>) &#8211; a subset of a universe, or None</li>
<li><strong>small_change</strong> (<em>bool</em>) &#8211; if True, algorithms optimized for small
configurational changes relative to the last
evaluation may be used.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the potential energy of interaction between the atoms
in subset1 and the atoms in subset2. If subset2 is None,
the interactions within subset1 are calculated. It both
subsets are None, the potential energy of the whole
universe is returned.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">float</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.energyAndForceConstants">
<tt class="descname">energyAndForceConstants</tt><big>(</big><em>subset1=None</em>, <em>subset2=None</em>, <em>small_change=False</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.energyAndForceConstants"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.energyAndForceConstants" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the energy and the force constants</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">(float, <tt class="xref py py-class docutils literal"><span class="pre">SymmetricPairTensor</span></tt>)</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.energyAndGradients">
<tt class="descname">energyAndGradients</tt><big>(</big><em>subset1=None</em>, <em>subset2=None</em>, <em>small_change=False</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.energyAndGradients"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.energyAndGradients" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the energy and the energy gradients</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">(float, <a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>)</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.energyGradientsAndForceConstants">
<tt class="descname">energyGradientsAndForceConstants</tt><big>(</big><em>subset1=None</em>, <em>subset2=None</em>, <em>small_change=False</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.energyGradientsAndForceConstants"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.energyGradientsAndForceConstants" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the energy, its gradients, and the force constants</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">(float, <a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>,
<tt class="xref py py-class docutils literal"><span class="pre">SymmetricPairTensor</span></tt>)</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.energyTerms">
<tt class="descname">energyTerms</tt><big>(</big><em>subset1=None</em>, <em>subset2=None</em>, <em>small_change=False</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.energyTerms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.energyTerms" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a dictionary containing the energy values for each
energy term separately. The energy terms are defined by the
force field.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.enforceConstraints">
<tt class="descname">enforceConstraints</tt><big>(</big><em>configuration=None</em>, <em>velocities=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.enforceConstraints"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.enforceConstraints" title="Permalink to this definition">¶</a></dt>
<dd><p>Enforces the previously defined distance constraints
by modifying the configuration and velocities.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>configuration</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a>) &#8211; the configuration in which the
constraints are enforced
(None for current configuration)</li>
<li><strong>velocities</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>) &#8211; the velocities in which the
constraints are enforced
(None for current velocities)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.environmentObjectList">
<tt class="descname">environmentObjectList</tt><big>(</big><em>klass=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.environmentObjectList"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.environmentObjectList" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>klass</strong> (<em>class</em>) &#8211; an optional class argument</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a list of all environment objects in the universe.
If klass is given, only objects that are instances
of klass are returned.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.forcefield">
<tt class="descname">forcefield</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.forcefield"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.forcefield" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the force field</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><tt class="xref py py-class docutils literal"><span class="pre">ForceField</span></tt></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.fractionalToCartesian">
<tt class="descname">fractionalToCartesian</tt><big>(</big><em>array</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.fractionalToCartesian"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.fractionalToCartesian" title="Permalink to this definition">¶</a></dt>
<dd><p>Fractional coordinates are defined only for periodic universes;
their components have values between 0. and 1.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>array</strong> (<em>Scientific.N.array_type</em>) &#8211; an array of fractional coordinates</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the real-space equivalent of vector</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Vector</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.getAtomBooleanArray">
<tt class="descname">getAtomBooleanArray</tt><big>(</big><em>name</em><big>)</big><a class="headerlink" href="#MMTK.Universe.Universe.getAtomBooleanArray" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>name</strong> (<em>str</em>) &#8211; the name of an atom attribute</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the values of the boolean attribute &#8216;name&#8217; for each atom
in the universe, or False for atoms that do not have
the attribute.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.getAtomScalarArray">
<tt class="descname">getAtomScalarArray</tt><big>(</big><em>name</em>, <em>datatype='d'</em><big>)</big><a class="headerlink" href="#MMTK.Universe.Universe.getAtomScalarArray" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>name</strong> (<em>str</em>) &#8211; the name of an atom attribute</li>
<li><strong>datatype</strong> &#8211; the datatype of the array allocated to hold the data</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the values of the attribute &#8216;name&#8217; for each atom
in the universe.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last"><a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.getParticleBoolean">
<tt class="descname">getParticleBoolean</tt><big>(</big><em>name</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.getParticleBoolean"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.getParticleBoolean" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>name</strong> (<em>str</em>) &#8211; the name of an atom attribute</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the values of the boolean attribute &#8216;name&#8217; for each atom
in the universe, or False for atoms that do not have
the attribute.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.getParticleScalar">
<tt class="descname">getParticleScalar</tt><big>(</big><em>name</em>, <em>datatype='d'</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.getParticleScalar"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.getParticleScalar" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>name</strong> (<em>str</em>) &#8211; the name of an atom attribute</li>
<li><strong>datatype</strong> &#8211; the datatype of the array allocated to hold the data</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the values of the attribute &#8216;name&#8217; for each atom
in the universe.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last"><a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.initializeVelocitiesToTemperature">
<tt class="descname">initializeVelocitiesToTemperature</tt><big>(</big><em>temperature</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.initializeVelocitiesToTemperature"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.initializeVelocitiesToTemperature" title="Permalink to this definition">¶</a></dt>
<dd><p>Generate random velocities for all atoms from a Boltzmann
distribution.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>temperature</strong> (<em>float</em>) &#8211; the reference temperature for the Boltzmann
distribution</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.largestDistance">
<tt class="descname">largestDistance</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.largestDistance"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.largestDistance" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the largest possible distance between any two points
that can be represented independent of orientation, i.e. the
radius of the largest sphere that fits into the simulation
cell. Returns None if no such upper limit exists.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">NoneType or float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.map">
<tt class="descname">map</tt><big>(</big><em>function</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.map"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.map" title="Permalink to this definition">¶</a></dt>
<dd><p>Apply a function to all objects in the universe and
return the list of the results. If the results are chemical
objects, a Collection object is returned instead of a list.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>function</strong> (<em>callable</em>) &#8211; the function to be applied</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the list or collection of the results</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.masses">
<tt class="descname">masses</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.masses"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.masses" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the masses of all atoms in the universe</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.numberOfDistanceConstraints">
<tt class="descname">numberOfDistanceConstraints</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.numberOfDistanceConstraints"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.numberOfDistanceConstraints" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the number of distance constraints</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.objectList">
<tt class="descname">objectList</tt><big>(</big><em>klass=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.objectList"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.objectList" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>klass</strong> (<em>class</em>) &#8211; an optional class argument</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">a list of all chemical objects in the universe.
If klass is given, only objects that are instances
of klass are returned.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.randomPoint">
<tt class="descname">randomPoint</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.randomPoint"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.randomPoint" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a random point from a uniform distribution within
the universe. This operation is defined only for
finite-volume universes, e.g. periodic universes.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Vector</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.realToBoxCoordinates">
<tt class="descname">realToBoxCoordinates</tt><big>(</big><em>vector</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.realToBoxCoordinates"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.realToBoxCoordinates" title="Permalink to this definition">¶</a></dt>
<dd><p>Box coordinates are defined only for periodic universes;
their components have values between -0.5 and 0.5; these
extreme values correspond to the walls of the simulation box.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>vector</strong> &#8211; a point in the universe</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the box coordinate equivalent of vector, or the original
vector if no box coordinate system exists</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Scientific.Geometry.Vector</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.reciprocalBasisVectors">
<tt class="descname">reciprocalBasisVectors</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.reciprocalBasisVectors"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.reciprocalBasisVectors" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the reciprocal basis vectors of the elementary cell of
a periodic universe, or None for a non-periodic universe</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">NoneType or list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.releaseConfigurationChangeLock">
<tt class="descname">releaseConfigurationChangeLock</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.releaseConfigurationChangeLock"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.releaseConfigurationChangeLock" title="Permalink to this definition">¶</a></dt>
<dd><p>Releases the configuration change lock.</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.releaseReadStateLock">
<tt class="descname">releaseReadStateLock</tt><big>(</big><em>write=False</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.releaseReadStateLock"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.releaseReadStateLock" title="Permalink to this definition">¶</a></dt>
<dd><p>Release the universe read state lock.</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.releaseWriteStateLock">
<tt class="descname">releaseWriteStateLock</tt><big>(</big><em>write=False</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.releaseWriteStateLock"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.releaseWriteStateLock" title="Permalink to this definition">¶</a></dt>
<dd><p>Release the universe write state lock.</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.removeDistanceConstraints">
<tt class="descname">removeDistanceConstraints</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.removeDistanceConstraints"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.removeDistanceConstraints" title="Permalink to this definition">¶</a></dt>
<dd><p>Removes all distance constraints.</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.removeObject">
<tt class="descname">removeObject</tt><big>(</big><em>object</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.removeObject"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.removeObject" title="Permalink to this definition">¶</a></dt>
<dd><p>Removes object from the universe. If object is a Collection,
each of its elements is removed. The object to be removed must
be in the universe.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>object</strong> &#8211; the object (chemical or environment) to be removed</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.scaleVelocitiesToTemperature">
<tt class="descname">scaleVelocitiesToTemperature</tt><big>(</big><em>temperature</em>, <em>block=True</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.scaleVelocitiesToTemperature"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.scaleVelocitiesToTemperature" title="Permalink to this definition">¶</a></dt>
<dd><p>Scale all velocities by a common factor in order to obtain
the specified temperature.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>temperature</strong> (<em>float</em>) &#8211; the reference temperature</li>
<li><strong>block</strong> (<em>bool</em>) &#8211; if True, the operation blocks other threads
from accessing the configuration before the update
is completed. If False, it is assumed that the
caller takes care of locking.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.selectBox">
<tt class="descname">selectBox</tt><big>(</big><em>p1</em>, <em>p2</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.selectBox"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.selectBox" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>p1</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; one corner of a box in space</li>
<li><strong>p2</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; the other corner of a box in space</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">a Collection of all objects in the universe that lie
within the box whose diagonally opposite corners are
given by p1 and p2.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.selectShell">
<tt class="descname">selectShell</tt><big>(</big><em>point</em>, <em>r1</em>, <em>r2=0.0</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.selectShell"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.selectShell" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>point</strong> (<em>Scientific.Geometry.Vector</em>) &#8211; a point in space</li>
<li><strong>r1</strong> (<em>float</em>) &#8211; one of the radii of a spherical shell</li>
<li><strong>r2</strong> (<em>float</em>) &#8211; the other of the two radii of a spherical shell</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">a Collection of all objects in the universe whose
distance from point lies between r1 and r2.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.setBondConstraints">
<tt class="descname">setBondConstraints</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.setBondConstraints"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.setBondConstraints" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets distance constraints for all bonds.</p>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.setConfiguration">
<tt class="descname">setConfiguration</tt><big>(</big><em>configuration</em>, <em>block=True</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.setConfiguration"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.setConfiguration" title="Permalink to this definition">¶</a></dt>
<dd><p>Update the current configuration of the universe by copying
the given input configuration.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>configuration</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a>) &#8211; the new configuration</li>
<li><strong>block</strong> (<em>bool</em>) &#8211; if True, the operation blocks other threads
from accessing the configuration before the update
is completed. If False, it is assumed that the
caller takes care of locking.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.setForceField">
<tt class="descname">setForceField</tt><big>(</big><em>forcefield</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.setForceField"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.setForceField" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>forcefield</strong> (<tt class="xref py py-class docutils literal"><span class="pre">ForceField</span></tt>) &#8211; the new forcefield for this universe</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.setFromTrajectory">
<tt class="descname">setFromTrajectory</tt><big>(</big><em>trajectory</em>, <em>step=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.setFromTrajectory"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.setFromTrajectory" title="Permalink to this definition">¶</a></dt>
<dd><p>Set the state of the universe to the one stored in a trajectory.
This operation is thread-safe; it blocks other threads that
want to access the configuration or velocities while the data is
being updated.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>trajectory</strong> (<a class="reference internal" href="#MMTK.Trajectory.Trajectory" title="MMTK.Trajectory.Trajectory"><tt class="xref py py-class docutils literal"><span class="pre">Trajectory</span></tt></a>) &#8211; a trajectory object for this universe</li>
<li><strong>step</strong> (<em>int</em>) &#8211; a step number, or None for the default step
(0 for a standard trajectory, the last written
step for a restart trajectory)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.setVelocities">
<tt class="descname">setVelocities</tt><big>(</big><em>velocities</em>, <em>block=True</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.setVelocities"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.setVelocities" title="Permalink to this definition">¶</a></dt>
<dd><p>Update the current velocities of the universe by copying
the given input velocities.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>velocities</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a>) &#8211; the new velocities, or None to remove
the velocity definition from the universe</li>
<li><strong>block</strong> (<em>bool</em>) &#8211; if True, the operation blocks other threads
from accessing the configuration before the update
is completed. If False, it is assumed that the
caller takes care of locking.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.universe">
<tt class="descname">universe</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.universe"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.universe" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the universe itself</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="MMTK.Universe.Universe.velocities">
<tt class="descname">velocities</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#Universe.velocities"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.Universe.velocities" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">the current velocities of all atoms, or None if
no velocities are defined. Note that this is not a
copy of the current state but a reference to it;
its data will change whenever any changes are made
to the current velocities.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#MMTK.ParticleProperties.ParticleVector" title="MMTK.ParticleProperties.ParticleVector"><tt class="xref py py-class docutils literal"><span class="pre">ParticleVector</span></tt></a></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="MMTK.Universe.isUniverse">
<tt class="descclassname">MMTK.Universe.</tt><tt class="descname">isUniverse</tt><big>(</big><em>object</em><big>)</big><a class="reference internal" href="_modules/MMTK/Universe.html#isUniverse"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Universe.isUniverse" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>object</strong> &#8211; any Python object</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">True if object is a universe.</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.Visualization">
<span id="mmtk-visualization"></span><h2>MMTK.Visualization<a class="headerlink" href="#module-MMTK.Visualization" title="Permalink to this headline">¶</a></h2>
<p>Visualization of chemical objects, including animation</p>
<p>This module provides visualization of chemical objects and animated
visualization of normal modes and sequences of configurations, including
trajectories. Visualization depends on external visualization programs.
On Unix systems, these programs are defined by environment variables.
Under Windows NT, the system definitions for files with extension
&#8220;pdb&#8221; and &#8220;wrl&#8221; are used.</p>
<p>A viewer for PDB files can be defined by the environment variable
&#8216;PDBVIEWER&#8217;. For showing a PDB file, MMTK will execute a command
consisting of the value of this variable followed by a space
and the name of the PDB file.</p>
<p>A viewer for VRML files can be defined by the environment variable
&#8216;VRMLVIEWER&#8217;. For showing a VRML file, MMTK will execute a command
consisting of the value of this variable followed by a space
and the name of the VRML file.</p>
<p>Since there is no standard for launching viewers for animation,
MMTK supports only two programs: VMD and XMol. MMTK detects
these programs by inspecting the value of the environment variable
&#8216;PDBVIEWER&#8217;. This value must be the file name of the executable,
and must give &#8220;vmd&#8221; or &#8220;xmol&#8221; after stripping off an optional
directory specification.</p>
<dl class="class">
<dt id="MMTK.Visualization.Viewable">
<em class="property">class </em><tt class="descclassname">MMTK.Visualization.</tt><tt class="descname">Viewable</tt><a class="reference internal" href="_modules/MMTK/Visualization.html#Viewable"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Visualization.Viewable" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Any viewable chemical object</p>
<p>This is a mix-in class that defines a general
visualization method for all viewable objects, i.e. chemical
objects (atoms, molecules, etc.), collections, and universes.</p>
<dl class="method">
<dt id="MMTK.Visualization.Viewable.graphicsObjects">
<tt class="descname">graphicsObjects</tt><big>(</big><em>**options</em><big>)</big><a class="reference internal" href="_modules/MMTK/Visualization.html#Viewable.graphicsObjects"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Visualization.Viewable.graphicsObjects" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>configuration</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.Configuration" title="MMTK.ParticleProperties.Configuration"><tt class="xref py py-class docutils literal"><span class="pre">Configuration</span></tt></a>) &#8211; the configuration in which the objects
are drawn (default: the current configuration)</li>
<li><strong>model</strong> (<em>str</em>) &#8211; the graphical representation to be used (one of
&#8220;wireframe&#8221;, &#8220;tube&#8221;, &#8220;ball_and_stick&#8221;, &#8220;vdw&#8221; and
&#8220;vdw_and_stick&#8221;).  The vdw models use balls
with the radii taken from the atom objects.
Default is &#8220;wireframe&#8221;.</li>
<li><strong>ball_radius</strong> (<em>float</em>) &#8211; the radius of the balls representing the atoms
in a ball_and_stick model, default: 0.03
This is also used in vdw and vdw_and_stick when
an atom does not supply a radius.</li>
<li><strong>stick_radius</strong> (<em>float</em>) &#8211; the radius of the sticks representing the bonds
in a ball_and_stick, vdw_and_stick or tube model.
Default: 0.02 for the tube model, 0.01 for the
ball_and_stick and vdw_and_stick models</li>
<li><strong>graphics_module</strong> (<em>module</em>) &#8211; the module in which the elementary graphics
objects are defined
(default: Scientific.Visualization.VRML)</li>
<li><strong>color_values</strong> (<a class="reference internal" href="#MMTK.ParticleProperties.ParticleScalar" title="MMTK.ParticleProperties.ParticleScalar"><tt class="xref py py-class docutils literal"><span class="pre">ParticleScalar</span></tt></a>) &#8211; a color value for each atom which defines
the color via the color scale object specified
by the option color_scale. If no value is
given, the atoms&#8217; colors are taken from the
attribute &#8216;color&#8217; of each atom object (default
values for each chemical element are provided
in the chemical database).</li>
<li><strong>color_scale</strong> (<em>callable</em>) &#8211; an object that returns a color object (as defined
in the module Scientific.Visualization.Color)
when called with a number argument. Suitable
objects are defined by
Scientific.Visualization.Color.ColorScale and
Scientific.Visualization.Color.SymmetricColorScale.
The object is used only when the option
color_values is specified as well. The default
is a blue-to-red color scale that covers the
range of the values given in color_values.</li>
<li><strong>color</strong> &#8211; a color name predefined in the module
Scientific.Visualization.Color. The corresponding
color is applied to all graphics objects that are
returned.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">a list of graphics objects that represent
the object for which the method is called.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">list</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="MMTK.Visualization.definePDBViewer">
<tt class="descclassname">MMTK.Visualization.</tt><tt class="descname">definePDBViewer</tt><big>(</big><em>progname</em>, <em>exec_path</em><big>)</big><a class="reference internal" href="_modules/MMTK/Visualization.html#definePDBViewer"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Visualization.definePDBViewer" title="Permalink to this definition">¶</a></dt>
<dd><p>Define the program used to view PDB files.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>progname</strong> (<em>str</em>) &#8211; the canonical name of the PDB viewer. If it is
a known one (one of &#8220;vmd&#8221;, &#8220;xmol&#8221;, &#8220;imol&#8221;),
special features such as animation may be
available.</li>
<li><strong>exec_path</strong> (<em>str</em>) &#8211; the path to the executable program</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.Visualization.defineVRMLiewer">
<tt class="descclassname">MMTK.Visualization.</tt><tt class="descname">defineVRMLiewer</tt><big>(</big><em>progname</em>, <em>exec_path</em><big>)</big><a class="reference internal" href="_modules/MMTK/Visualization.html#defineVRMLiewer"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Visualization.defineVRMLiewer" title="Permalink to this definition">¶</a></dt>
<dd><p>Define the program used to view VRML files.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>progname</strong> (<em>str</em>) &#8211; the canonical name of the VRML viewer</li>
<li><strong>exec_path</strong> (<em>str</em>) &#8211; the path to the executable program</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.Visualization.view">
<tt class="descclassname">MMTK.Visualization.</tt><tt class="descname">view</tt><big>(</big><em>object</em>, <em>*parameters</em><big>)</big><a class="reference internal" href="_modules/MMTK/Visualization.html#view"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Visualization.view" title="Permalink to this definition">¶</a></dt>
<dd><p>Equivalent to object.view(parameters).</p>
</dd></dl>

<dl class="function">
<dt id="MMTK.Visualization.viewSequence">
<tt class="descclassname">MMTK.Visualization.</tt><tt class="descname">viewSequence</tt><big>(</big><em>object</em>, <em>conf_list</em>, <em>periodic=False</em>, <em>label=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Visualization.html#viewSequence"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Visualization.viewSequence" title="Permalink to this definition">¶</a></dt>
<dd><p>Launches an animation using an external viewer.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>object</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; the object for which the animation is displayed.</li>
<li><strong>conf_list</strong> (<em>sequence</em>) &#8211; a sequence of configurations that define the animation</li>
<li><strong>periodic</strong> &#8211; if True, turn animation into a loop</li>
<li><strong>label</strong> (<em>str</em>) &#8211; an optional text string that some interfaces
use to pass a description of the object to the
visualization system.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="MMTK.Visualization.viewTrajectory">
<tt class="descclassname">MMTK.Visualization.</tt><tt class="descname">viewTrajectory</tt><big>(</big><em>trajectory</em>, <em>first=0</em>, <em>last=None</em>, <em>skip=1</em>, <em>subset=None</em>, <em>label=None</em><big>)</big><a class="reference internal" href="_modules/MMTK/Visualization.html#viewTrajectory"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.Visualization.viewTrajectory" title="Permalink to this definition">¶</a></dt>
<dd><p>Launches an animation based on a trajectory using an external viewer.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>trajectory</strong> (<a class="reference internal" href="#MMTK.Trajectory.Trajectory" title="MMTK.Trajectory.Trajectory"><tt class="xref py py-class docutils literal"><span class="pre">Trajectory</span></tt></a>) &#8211; the trajectory</li>
<li><strong>first</strong> (<em>int</em>) &#8211; the first trajectory step to be used</li>
<li><strong>last</strong> (<em>int</em>) &#8211; the first trajectory step NOT to be used</li>
<li><strong>skip</strong> (<em>int</em>) &#8211; the distance between two consecutive steps shown</li>
<li><strong>subset</strong> (<a class="reference internal" href="#MMTK.Collections.GroupOfAtoms" title="MMTK.Collections.GroupOfAtoms"><tt class="xref py py-class docutils literal"><span class="pre">GroupOfAtoms</span></tt></a>) &#8211; the subset of the universe that is shown
(default: the whole universe)</li>
<li><strong>label</strong> (<em>str</em>) &#8211; an optional text string that some interfaces
use to pass a description of the object to the
visualization system.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-MMTK.XML">
<span id="mmtk-xml"></span><h2>MMTK.XML<a class="headerlink" href="#module-MMTK.XML" title="Permalink to this headline">¶</a></h2>
<p>XML format for describing molecular systems</p>
<p>Note: this format is not used by any other program at the moment. It should
be considered experimental and subject to change.</p>
<dl class="class">
<dt id="MMTK.XML.XMLMoleculeFactory">
<em class="property">class </em><tt class="descclassname">MMTK.XML.</tt><tt class="descname">XMLMoleculeFactory</tt><big>(</big><em>file</em><big>)</big><a class="reference internal" href="_modules/MMTK/XML.html#XMLMoleculeFactory"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#MMTK.XML.XMLMoleculeFactory" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#MMTK.MoleculeFactory.MoleculeFactory" title="MMTK.MoleculeFactory.MoleculeFactory"><tt class="xref py py-class docutils literal"><span class="pre">MMTK.MoleculeFactory.MoleculeFactory</span></tt></a></p>
<p>XML molecule factory</p>
<p>An XML molecule factory reads an XML specification of a molecular
system and builds the molecule objects and universe described
therein. The universe can be obtained through the attribute
<em>universe</em>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>file</strong> &#8211; the name of an XML file, or a file object</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Module Reference</a><ul>
<li><a class="reference internal" href="#module-MMTK">MMTK</a></li>
<li><a class="reference internal" href="#module-MMTK.Biopolymers">MMTK.Biopolymers</a></li>
<li><a class="reference internal" href="#module-MMTK.Bonds">MMTK.Bonds</a></li>
<li><a class="reference internal" href="#module-MMTK.ChargeFit">MMTK.ChargeFit</a></li>
<li><a class="reference internal" href="#module-MMTK.ChemicalObjects">MMTK.ChemicalObjects</a></li>
<li><a class="reference internal" href="#module-MMTK.Collections">MMTK.Collections</a></li>
<li><a class="reference internal" href="#module-MMTK.ConfigIO">MMTK.ConfigIO</a></li>
<li><a class="reference internal" href="#module-MMTK.DCD">MMTK.DCD</a></li>
<li><a class="reference internal" href="#module-MMTK.Deformation">MMTK.Deformation</a></li>
<li><a class="reference internal" href="#module-MMTK.Dynamics">MMTK.Dynamics</a></li>
<li><a class="reference internal" href="#module-MMTK.Environment">MMTK.Environment</a></li>
<li><a class="reference internal" href="#module-MMTK.Field">MMTK.Field</a></li>
<li><a class="reference internal" href="#mmtk-forcefields">MMTK.ForceFields</a><ul>
<li><a class="reference internal" href="#module-MMTK.ForceFields.Amber.AmberForceField">MMTK.ForceFields.AmberForceField</a><ul>
<li><a class="reference internal" href="#mmtk-forcefields-amber94forcefield">MMTK.ForceFields.Amber94ForceField</a></li>
<li><a class="reference internal" href="#mmtk-forcefields-amber99forcefield">MMTK.ForceFields.Amber99ForceField</a></li>
<li><a class="reference internal" href="#mmtk-forcefields-oplsforcefield">MMTK.ForceFields.OPLSForceField</a></li>
</ul>
</li>
<li><a class="reference internal" href="#mmtk-forcefields-anisotropicnetworkforcefield">MMTK.ForceFields.AnisotropicNetworkForceField</a></li>
<li><a class="reference internal" href="#mmtk-forcefields-calphaforcefield">MMTK.ForceFields.CalphaForceField</a></li>
<li><a class="reference internal" href="#mmtk-forcefields-deformationforcefield">MMTK.ForceFields.DeformationForceField</a></li>
<li><a class="reference internal" href="#mmtk-forcefields-harmonicforcefield">MMTK.ForceFields.HarmonicForceField</a></li>
<li><a class="reference internal" href="#mmtk-forcefields-lennardjonesforcefield">MMTK.ForceFields.LennardJonesForceField</a></li>
<li><a class="reference internal" href="#mmtk-forcefields-spceforcefield">MMTK.ForceFields.SPCEForceField</a></li>
<li><a class="reference internal" href="#module-MMTK.ForceFields.ForceFieldTest">MMTK.ForceFields.ForceFieldTest</a></li>
<li><a class="reference internal" href="#module-MMTK.ForceFields.Restraints">MMTK.ForceFields.Restraints</a></li>
</ul>
</li>
<li><a class="reference internal" href="#module-MMTK.FourierBasis">MMTK.FourierBasis</a></li>
<li><a class="reference internal" href="#module-MMTK.Geometry">MMTK.Geometry</a></li>
<li><a class="reference internal" href="#module-MMTK.InternalCoordinates">MMTK.InternalCoordinates</a></li>
<li><a class="reference internal" href="#module-MMTK.Minimization">MMTK.Minimization</a></li>
<li><a class="reference internal" href="#module-MMTK.MolecularSurface">MMTK.MolecularSurface</a></li>
<li><a class="reference internal" href="#module-MMTK.MoleculeFactory">MMTK.MoleculeFactory</a></li>
<li><a class="reference internal" href="#module-MMTK.NormalModes">MMTK.NormalModes</a><ul>
<li><a class="reference internal" href="#module-MMTK.NormalModes.BrownianModes">MMTK.NormalModes.BrownianModes</a></li>
<li><a class="reference internal" href="#module-MMTK.NormalModes.EnergeticModes">MMTK.NormalModes.EnergeticModes</a></li>
<li><a class="reference internal" href="#module-MMTK.NormalModes.VibrationalModes">MMTK.NormalModes.VibrationalModes</a></li>
</ul>
</li>
<li><a class="reference internal" href="#module-MMTK.NucleicAcids">MMTK.NucleicAcids</a></li>
<li><a class="reference internal" href="#module-MMTK.PDB">MMTK.PDB</a><ul>
<li><a class="reference internal" href="#module-MMTK.PDBMoleculeFactory">MMTK.PDBMoleculeFactory</a></li>
</ul>
</li>
<li><a class="reference internal" href="#module-MMTK.ParticleProperties">MMTK.ParticleProperties</a></li>
<li><a class="reference internal" href="#module-MMTK.ProteinFriction">MMTK.ProteinFriction</a></li>
<li><a class="reference internal" href="#module-MMTK.Proteins">MMTK.Proteins</a></li>
<li><a class="reference internal" href="#module-MMTK.Random">MMTK.Random</a></li>
<li><a class="reference internal" href="#module-MMTK.Solvation">MMTK.Solvation</a></li>
<li><a class="reference internal" href="#module-MMTK.Subspace">MMTK.Subspace</a></li>
<li><a class="reference internal" href="#module-MMTK.Trajectory">MMTK.Trajectory</a></li>
<li><a class="reference internal" href="#module-MMTK.Units">MMTK.Units</a></li>
<li><a class="reference internal" href="#module-MMTK.Universe">MMTK.Universe</a></li>
<li><a class="reference internal" href="#module-MMTK.Visualization">MMTK.Visualization</a></li>
<li><a class="reference internal" href="#module-MMTK.XML">MMTK.XML</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="examples.html"
                        title="previous chapter">Code Examples</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/modules.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="examples.html" title="Code Examples"
             >previous</a> |</li>
        <li><a href="index.html">MMTK User Guide 2.7.7 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2010, Konrad Hinsen.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>